<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Ulisse">
<title>Ulisse pipeline: crosstalk analysis of single cell data • Ulisse</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Ulisse pipeline: crosstalk analysis of single cell data">
<meta property="og:description" content="Ulisse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Ulisse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Ulisse_pipeline.html">Ulisse pipeline: crosstalk analysis of single cell data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Ulisse pipeline: crosstalk analysis of single cell data</h1>
                        <h4 data-toc-skip class="author">Alice
Chiodi</h4>
            
      
      
      <div class="d-none name"><code>Ulisse_pipeline.Rmd</code></div>
    </div>

    
    
<p>In this vignette we will describe the usage of Ulisse. We will use
already processed data from <span class="citation">(Pal et al.
2021)</span> and some Seurat v4 <span class="citation">(Hao et al.
2021)</span> functions to obtain input data. As you will see, the usage
of Seurat is not mandatory, as the same data can be obtained with other
tools. Moreover, the same workflow can be applied to other data (i.e.:
expression, mutation, combination of the two, etc…), or even applied
only partially (i.e.: intracellular crosstalk only on bulk data).</p>
<p>This vignette will describe Ulisse usage to analyse communications
among cell types, intracellular crosstalk in Cancer-Associated
Fibroblasts (CAF), and integrated crosstalk of cancer-CAF clusters. In
general, crosstalk calculation needs:</p>
<ul>
<li>gene sets</li>
<li>gene scores</li>
<li>gene-gene interaction network.</li>
</ul>
<p>Gene-sets definition depends on the test used: in cell-cell
communication the gene sets are the cell types (or clusters), while in
intracellular analysis they represents cellular functions or mechanisms,
defined by using public (pathway) databases. In our example, we will use
MSigDB Hallmarks database.</p>
<p>The definition of gene scores is up to the user, the data under
analysis and the focus of the study. Here we will use a score calculated
over Differentially Expressed Genes (DEG) testing, to use a statistic
that summarize both the significance and the importance of the gene in a
cell type over the entire dataset. It is important to note, that the
analysis can be done without providing gene weights, by setting them =
1.</p>
<p>Equal to the gene sets, the definition of interaction network depends
on the test: communication analysis needs a communication network, while
other gene/protein interaction network can be suitable for intracellular
analysis. Here, we will use Omnipath <span class="citation">(Türei et
al. 2021)</span> and String <span class="citation">(Szklarczyk et al.
2021)</span> networks, respectively.</p>
<p>The code used in this vignette replicates the one used for Ulisse
publication.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Ulisse</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RGLab/MAST/" class="external-link">MAST</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/OmnipathR/" class="external-link">OmnipathR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="data-uploading-preparation-and-differential-expression-testing">Data uploading, preparation and Differential Expression testing<a class="anchor" aria-label="anchor" href="#data-uploading-preparation-and-differential-expression-testing"></a>
</h2>
<p>As an example we will use Triple Negative Breast Cancer(TNBC) data
from <span class="citation">(Pal et al. 2021)</span>. Already processed
data can be obtained at <span class="citation">(Chen and Smyth
2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tnbc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"17058077/SeuratObject_TNBC.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">tnbc</span>, group.by <span class="op">=</span> <span class="st">"integrated_snn_res.0.1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="data/tsne_cluster.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>The dataset is composed by 8 samples integrated with Seurat method.
Processing of the data, described in detail in the article, resulted in
9 cluster. In the paper they provide cluster annotation, which is not
present in the <code>tnbc</code> object. We used
<code>SeuratObject_TNBCSub.rds</code> and paper plots to reconstruct
cluster annotation (not shown).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tnbc_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Cancer_epi"</span>, <span class="st">"Tcells"</span>, <span class="st">"Cancer_epi_cycling"</span>, <span class="st">"TAM"</span>, <span class="st">"Plasma_cells"</span>, <span class="st">"CAF"</span>, <span class="st">"Bcells"</span>, <span class="st">"DC"</span>, <span class="st">"Endothelial"</span><span class="op">)</span>, <span class="fl">0</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span></span>
<span><span class="va">tnbc</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">&lt;-</span> <span class="va">tnbc_ct</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">tnbc</span><span class="op">$</span><span class="va">integrated_snn_res.0.1</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">ct_pal</span> <span class="op">&lt;-</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">trubetskoy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">tnbc_ct</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ct_pal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tnbc_ct</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">tnbc</span>, group.by <span class="op">=</span> <span class="st">"cell_type"</span>, cols <span class="op">=</span> <span class="va">ct_pal</span><span class="op">)</span></span></code></pre></div>
<p><img src="data/tsne_ct.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>Once obtained cluster annotation, we test for differential expression
(DE) by using <code>MAST</code> algorithm. DE results will then be used
to calculate a cluster-specific gene score, which is defined as <span class="math inline">\((log_{2}(Fold
Change))(-log_{10}(adjusted-p-value))\)</span>. To reduce noise, only
scores of genes with: <span class="math inline">\(adjusted-p-value &lt;
0.05\)</span> &amp; <span class="math inline">\(log2(Fold Change) &gt;
0.5\)</span> will be considered, all the other are set to 0. We will
normalize the score over the maximum value to obtain more intelligible
values.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#this may take a while</span></span>
<span><span class="co">#to reduce change the test, the results will be different</span></span>
<span><span class="va">DEG_tnbc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">tnbc</span>, test <span class="op">=</span> <span class="st">"MAST"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_val</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span><span class="op">[</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span><span class="op">[</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">*</span> <span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">score</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">score</span><span class="op">)</span> <span class="co">#normalize score values</span></span>
<span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">score</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">p_adj_BH</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span> <span class="co">#denoising</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="intracellular-crosstalk">Intracellular crosstalk<a class="anchor" aria-label="anchor" href="#intracellular-crosstalk"></a>
</h2>
<p>In this section we will demonstrate the usage of Ulisse to analyse
intracellular crosstalk. To do so, we will focus on Cancer epithelial
cell type.</p>
<div class="section level3">
<h3 id="inputs">Inputs<a class="anchor" aria-label="anchor" href="#inputs"></a>
</h3>
<p>We need:</p>
<ul>
<li>biological network: here we will use String v12, which is provided
within NPAtools package</li>
<li>gene scores: we will use “Cancer_epi” DEGs</li>
<li>gene sets: here we will use MsigDB hallmarks, downloaded from
msigdbr package</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#biological network</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NAPtools</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span> <span class="co">#stringv12 names are entry id, but here we need symbols. </span></span>
<span><span class="co">#Without this code, the adjacency matrix rownames and colnames will be the entry ids</span></span>
<span><span class="va">g.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/as_adjacency_matrix.html" class="external-link">as_adjacency_matrix</a></span><span class="op">(</span><span class="va">stringv12</span>, sparse <span class="op">=</span> <span class="cn">F</span>, type <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#gene set database, mapped to  string genes</span></span>
<span><span class="va">hallmarks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">ptw.H</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">hallmarks</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"gene_symbol"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">ptw.H</span> <span class="op">&lt;-</span> <span class="va">ptw.H</span><span class="op">[</span><span class="va">ptw.H</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span>,<span class="op">]</span></span></code></pre></div>
<p>Once obtained gene set and gene scores, these needs to be coupled
(via <code><a href="../reference/build_S_list.html">build_S_list()</a></code>) to build a weighted gene set list. At
this point, this list can be provided to Ulisse, together with the
network, to perform the analysis with <code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code>
function. There are two way to use <code><a href="../reference/build_S_list.html">build_S_list()</a></code>: one is
shown here, where the gene sets are provided as a table, together with a
named vector for weights; the other is shown in later in the
vignette.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g.w</span> <span class="op">&lt;-</span> <span class="va">DEG_tnbc</span><span class="op">[</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"Cancer_epi"</span>,<span class="op">]</span></span>
<span><span class="va">g.w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">g.w</span><span class="op">$</span><span class="va">score</span>, <span class="va">g.w</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gs_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_S_list.html">build_S_list</a></span><span class="op">(</span>S_tab <span class="op">=</span> <span class="va">ptw.H</span>, g_u <span class="op">=</span> <span class="va">g.w</span>,  universe <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span>, min_size <span class="op">=</span> <span class="cn">NULL</span>, max_size <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="crosstalk-calculation">crosstalk calculation<a class="anchor" aria-label="anchor" href="#crosstalk-calculation"></a>
</h3>
<p>Now we have everything ready for intracellular crosstalk calculation.
Here, we will set <code>ct_type = "intracellular"</code>, so the
function will not consider overlapping genes, as they are the same and
shouldn’t be considered to avoid using gene-set internal links. This
return a crosstalk table</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">can_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>S_list <span class="op">=</span> <span class="va">gs_list</span>,</span>
<span>                        A <span class="op">=</span> <span class="va">g.adj</span>, </span>
<span>                        k <span class="op">=</span> <span class="fl">99</span>, </span>
<span>                        ct_type <span class="op">=</span> <span class="st">"intracellular"</span>,</span>
<span>                        mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, perm_link <span class="op">=</span> <span class="st">"degree"</span>, perm_weights <span class="op">=</span> <span class="st">"degree"</span>, cut_par <span class="op">=</span> <span class="cn">NULL</span>, mc_cores_ct <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">can_ct</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;                        S1_name                         S2_name           c</span></span>
<span><span class="co">#&gt; 1        HALLMARK_ADIPOGENESIS HALLMARK_ESTROGEN_RESPONSE_LATE 0.011211030</span></span>
<span><span class="co">#&gt; 2        HALLMARK_ADIPOGENESIS      HALLMARK_KRAS_SIGNALING_DN 0.025489284</span></span>
<span><span class="co">#&gt; 3        HALLMARK_ADIPOGENESIS       HALLMARK_MTORC1_SIGNALING 0.006278995</span></span>
<span><span class="co">#&gt; 4        HALLMARK_ADIPOGENESIS         HALLMARK_MYC_TARGETS_V1 0.042996954</span></span>
<span><span class="co">#&gt; 5 HALLMARK_ALLOGRAFT_REJECTION      HALLMARK_ANDROGEN_RESPONSE 0.019436802</span></span>
<span><span class="co">#&gt;   S1_size S2_size S1_S2_size S2_S1_size dL  L  r_c        u1         u2</span></span>
<span><span class="co">#&gt; 1      29      54          2          1  2  8 0.25 0.1429003 0.07845351</span></span>
<span><span class="co">#&gt; 2      33      15          1          1  1  1 1.00 0.1129081 0.22575250</span></span>
<span><span class="co">#&gt; 3      32      48          1          1  1 20 0.05 0.0741845 0.08464026</span></span>
<span><span class="co">#&gt; 4      31      53          1          4  4 16 0.25 0.0741845 0.57959484</span></span>
<span><span class="co">#&gt; 5      61      28          1          1  1  5 0.20 0.1308974 0.14848887</span></span>
<span><span class="co">#&gt;              S1                   S2           s    pA    pU            p</span></span>
<span><span class="co">#&gt; 1 ATP5PO;NDUFB7                COX6C 0.014607631 0.111 0.078 0.0497771929</span></span>
<span><span class="co">#&gt; 2          MYLK               CALML5 0.079114437 0.015 0.005 0.0007873517</span></span>
<span><span class="co">#&gt; 3        ATP5PO                 PPA1 0.003674212 0.248 0.288 0.2599206020</span></span>
<span><span class="co">#&gt; 4        ATP5PO RPLP0;RPS2;RPS3;RPS5 0.078286265 0.211 0.010 0.0151098521</span></span>
<span><span class="co">#&gt; 5        CDKN2A                CCND1 0.045286442 0.050 0.011 0.0046780758</span></span></code></pre>
<p>Results can be visualized as scatterplot, by plotting the score
(<code>s</code>) and the crosstalk saturation (`r_c``) affected over the
possible. This visualization allows to prioritize genes on the
percentage of affected crosstalks.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">can_ct</span><span class="op">$</span><span class="va">sign</span> <span class="op">&lt;-</span> <span class="st">"M_A | M_U &gt; 0.01"</span></span>
<span><span class="va">can_ct</span><span class="op">$</span><span class="va">sign</span><span class="op">[</span><span class="va">can_ct</span><span class="op">$</span><span class="va">pA</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">|</span> <span class="va">can_ct</span><span class="op">$</span><span class="va">pU</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"M_A | M_U &lt; 0.01"</span></span>
<span><span class="va">can_ct</span><span class="op">$</span><span class="va">sign</span><span class="op">[</span><span class="va">can_ct</span><span class="op">$</span><span class="va">pA</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">&amp;</span> <span class="va">can_ct</span><span class="op">$</span><span class="va">pU</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"M_A &amp; M_U &lt; 0.01"</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">can_ct</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span>  <span class="va">r_c</span>, y <span class="op">=</span> <span class="va">s</span>, shape <span class="op">=</span> <span class="va">sign</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">21</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span><span class="st">""</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"gray70"</span>, <span class="st">"gray30"</span><span class="op">)</span>, </span>
<span>                    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M_A | M_U &gt; 0.01"</span>, <span class="st">"M_A | M_U &lt; 0.01"</span>, <span class="st">"M_A &amp; M_U &lt; 0.01"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_test</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p><img src="data/pct_scatter.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>Otherwise, we can visualize results as a network. Here, we will
consider only significant interactions and set different color for fully
(<code>r_c = 1</code>) or partially (<code>r_c &lt; 1</code>) affected
crosstalks.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">can_ct</span> <span class="op">&lt;-</span> <span class="va">can_ct</span><span class="op">[</span><span class="va">can_ct</span><span class="op">$</span><span class="va">pA</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">|</span> <span class="va">can_ct</span><span class="op">$</span><span class="va">pU</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, <span class="op">]</span></span>
<span><span class="va">g.ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/graph_from_data_frame.html" class="external-link">graph_from_data_frame</a></span><span class="op">(</span><span class="va">can_ct</span>, directed <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">ly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html" class="external-link">create_layout</a></span><span class="op">(</span><span class="va">g.ct</span>, <span class="st">"nicely"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html" class="external-link">ggraph</a></span><span class="op">(</span><span class="va">g.ct</span>, layout <span class="op">=</span> <span class="va">ly</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_link.html" class="external-link">geom_edge_link</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>color <span class="op">=</span>  <span class="va">sign</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/scale_edge_colour.html" class="external-link">scale_edge_color_manual</a></span><span class="op">(</span><span class="st">""</span>, values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gray70"</span>, <span class="st">"gray30"</span><span class="op">)</span>, </span>
<span>                          breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M_A | M_U &lt; 0.01"</span>, <span class="st">"M_A &amp; M_U &lt; 0.01"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html" class="external-link">geom_node_point</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">21</span>, fill <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggraph.data-imaginist.com/reference/theme_graph.html" class="external-link">theme_graph</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p><img src="data/ct_net.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="crosstalk-gene-classification">Crosstalk gene classification<a class="anchor" aria-label="anchor" href="#crosstalk-gene-classification"></a>
</h3>
<p>Gene classification analysis is performed by extracting crosstalk and
interaction data for genes occurring in significant crosstalks, and
comparing them to a reference model, to calculate interactor and
crosstalk diversity and saturation. The general reference model should
be built on the same data of the case study, but without considering
weight differences among genes (so, all genes should have u = 1). In
this way, we can normalize crosstalk and interactor diversity of each
gene affected by the condition, over all possibilities. To do so, we
need to perform crosstalk analysis and gene classification of the
reference model, to then use it in our cancer cell gene classification
analysis, as described below. Note that here,
<code><a href="../reference/build_S_list.html">build_S_list()</a></code> is differently, as gene sets are provided as
a three column table: gene set names, gene names and weights. This
format should be used when genes have different weights in the gene sets
(as in intercellular crosstalk).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ptw.H</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="va">gs_list_general</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_S_list.html">build_S_list</a></span><span class="op">(</span>S_tab <span class="op">=</span> <span class="va">ptw.H</span>, universe <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span>, min_size <span class="op">=</span> <span class="cn">NULL</span>, max_size <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="va">general</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>S_list <span class="op">=</span> <span class="va">gs_list_general</span>, </span>
<span>                         A <span class="op">=</span> <span class="va">g.adj</span>, k <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                         ct_type <span class="op">=</span> <span class="st">"intracellular"</span>,</span>
<span>                         mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, perm_link <span class="op">=</span> <span class="st">"degree"</span>, </span>
<span>                         perm_weights <span class="op">=</span> <span class="st">"degree"</span>, cut_par <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                         mc_cores_ct <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">general_cl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_classification.html">gene_classification</a></span><span class="op">(</span>gs_ct <span class="op">=</span> <span class="va">general</span>, A <span class="op">=</span> <span class="va">g.adj</span>, n_cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">can_class</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_classification.html">gene_classification</a></span><span class="op">(</span>gs_ct <span class="op">=</span> <span class="va">caf_ct</span>,<span class="co">#only significant ones</span></span>
<span>                                 A <span class="op">=</span> <span class="va">g.adj</span>, </span>
<span>                                 ct_ref <span class="op">=</span> <span class="va">general_cl</span>, <span class="co">#general model for normalization</span></span>
<span>                                 n_cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">can_class</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;      gene        r_q        r_d d_q d_d</span></span>
<span><span class="co">#&gt; 1     APP 0.15625000 0.11904762   5   5</span></span>
<span><span class="co">#&gt; 2 ATP5F1C 0.05263158 0.14285714   1   4</span></span>
<span><span class="co">#&gt; 3  ATP5PO 0.06250000 0.14814815   1   4</span></span>
<span><span class="co">#&gt; 4  CALML5 0.13043478 0.04166667   3   1</span></span>
<span><span class="co">#&gt; 5   CCND1 0.15625000 0.10000000   5   3</span></span>
<span><span class="co">#&gt;                                                                                                                                    d_q_S</span></span>
<span><span class="co">#&gt; 1 HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_MTORC1_SIGNALING;HALLMARK_HYPOXIA;HALLMARK_ANDROGEN_RESPONSE</span></span>
<span><span class="co">#&gt; 2                                                                                                                HALLMARK_MYC_TARGETS_V1</span></span>
<span><span class="co">#&gt; 3                                                                                                                HALLMARK_MYC_TARGETS_V1</span></span>
<span><span class="co">#&gt; 4                                                   HALLMARK_MYOGENESIS;HALLMARK_ADIPOGENESIS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</span></span>
<span><span class="co">#&gt; 5 HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_P53_PATHWAY;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UV_RESPONSE_UP;HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">#&gt;                  d_d_gene  q  d</span></span>
<span><span class="co">#&gt; 1 DBI;KLK11;LSR;CLU;GAPDH 32 42</span></span>
<span><span class="co">#&gt; 2    RPLP0;RPS2;RPS3;RPS5 19 28</span></span>
<span><span class="co">#&gt; 3    RPLP0;RPS2;RPS3;RPS5 16 27</span></span>
<span><span class="co">#&gt; 4                    MYLK 23 24</span></span>
<span><span class="co">#&gt; 5    CDKN2A;CTNNB1;CDKN2B 32 30</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              q_S</span></span>
<span><span class="co">#&gt; 1 HALLMARK_APICAL_JUNCTION;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_GLYCOLYSIS;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MTORC1_SIGNALING;HALLMARK_NOTCH_SIGNALING;HALLMARK_OXIDATIVE_PHOSPHORYLATION;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY;HALLMARK_SPERMATOGENESIS;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE</span></span>
<span><span class="co">#&gt; 2                                                                                                                                                                                                                                                                                                                                                                                                           HALLMARK_P53_PATHWAY;HALLMARK_PROTEIN_SECRETION;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_INTERFERON_ALPHA_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_P53_PATHWAY;HALLMARK_PROTEIN_SECRETION;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_UP;HALLMARK_XENOBIOTIC_METABOLISM</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                                                                                                                                                       HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYOGENESIS;HALLMARK_P53_PATHWAY;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_PROTEIN_SECRETION;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_DN;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_GLYCOLYSIS;HALLMARK_HYPOXIA</span></span>
<span><span class="co">#&gt; 5                                                HALLMARK_ANGIOGENESIS;HALLMARK_APICAL_JUNCTION;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2;HALLMARK_MYOGENESIS;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                d_gene</span></span>
<span><span class="co">#&gt; 1 APLP2;APOE;ACHE;APBB1;TGFB2;DBI;TNFRSF21;CNTN1;GNAI2;CLU;CASP6;PSEN1;APOA1;TTR;APOC2;APOC3;F11;APOC1;GNG2;PRSS3;PLA2G7;APOA4;PRSS2;MAPT;LSR;KLK11;HSD17B10;SNCA;FOXO3;HEBP1;GAPDH;PRNP;SELENOS;CALR;APH1A;PSENEN;IDE;APOM;APBB2;MAPK8IP2;LCAT;NCSTN</span></span>
<span><span class="co">#&gt; 2                                                                                   CMPK1;MRPL3;RPL3L;RPL9;RPS9;CD74;ABCG4;AK3;POLR2C;POLR1C;HPRT1;SEC61A1;CDCA8;LHPP;ABCG2;ATP6V1B1;PPA1;RPS2;RPS3;RPS5;RPLP0;TUFM;KARS1;MRTO4;RPS14;IMP3;KDM6B;EI24</span></span>
<span><span class="co">#&gt; 3                                                                             MRPL3;RPL9;RPS9;HPRT1;SEC61A1;CDCA8;COX6C;SDHD;ATP6V0A1;ACKR3;ATP6V1B1;ATP5MC1;ATP6V1D;PPA1;RPS2;RPS3;RPS5;RPLP0;TUFM;COX5A;KARS1;ATP6V1F;RPS14;IMP3;SLC25A3;RER1;KDM6B</span></span>
<span><span class="co">#&gt; 4                                                                                                MYLK;NOS2;CAMKK2;HRAS;PLCG1;CAV1;PPP3R1;CALM3;ADCY1;HSP90AA1;PYGB;PYGL;PHKA2;CACNA1H;PYGM;HSP90B1;HSP90AB1;CAMK2B;CAV3;CAMK4;PPP1CA;ITPR2;CAV2;PHKG2</span></span>
<span><span class="co">#&gt; 5                                                                    CDKN2C;CCND2;CDKN2A;CCND3;CDK6;SYMPK;CTNNB1;CDK2;CDKN1B;CDKN1A;AR;PCNA;CDK1;CDC25A;CDK4;CKS1B;FOS;CUL1;KAT2B;CDKN1C;STAT3;FBXO4;TOR1AIP2;GSK3B;RB1;CDKN2B;HDAC3;CDKN2D;SKP2;ESR1</span></span></code></pre>
<p>Here we can visualize saturations of crosstalk diversity and
interactor diversity (<code>r_q</code> and <code>r_d</code>,
respectively), which are measure normalized on the general model,
compared to their absolute values (<code>d_q</code> and
<code>d_d</code>, respectively). This visualization allows to identify
genes whose interactors or crosstalks diversity are fully or only
partially affected. Here we can also color genes by their initial value,
to identify interesting genes (in terms of <code>r_q</code> and
<code>r_d</code>) which might not be prioritized by DEG analysis.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">can_class</span><span class="op">$</span><span class="va">u</span> <span class="op">&lt;-</span> <span class="va">g.w</span><span class="op">[</span><span class="va">can_class</span><span class="op">$</span><span class="va">gene</span><span class="op">]</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">can_class</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span>  <span class="va">r_q</span>, y <span class="op">=</span> <span class="va">d_q</span>, label <span class="op">=</span> <span class="va">gene</span>, colour <span class="op">=</span> <span class="va">u</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colours <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/brewer.html" class="external-link">brewer.greys</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_test</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">can_class</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">r_d</span>, y <span class="op">=</span> <span class="va">d_d</span>, label <span class="op">=</span> <span class="va">gene</span>, colour <span class="op">=</span> <span class="va">u</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colours <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/brewer.html" class="external-link">brewer.greys</a></span><span class="op">(</span><span class="fl">6</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_test</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">'collect'</span><span class="op">)</span></span></code></pre></div>
<p><img src="data/fcl_scatter.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="intercellular-communication">Intercellular Communication<a class="anchor" aria-label="anchor" href="#intercellular-communication"></a>
</h2>
<div class="section level3">
<h3 id="input-omnipath-communication-network">input: omnipath communication network<a class="anchor" aria-label="anchor" href="#input-omnipath-communication-network"></a>
</h3>
<p>As described before, we need a communication network for
communication analysis. Here we will use OmnipathR package to download
the intercellular network. The data reported here are the ones used in
the publication (downloaded in September, 2024). Later updates of
Omnipath database may yield different results.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">intercell_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html" class="external-link">import_intercell_network</a></span><span class="op">(</span>interactions_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"datasets"</span> <span class="op">=</span> <span class="st">"omnipath"</span><span class="op">)</span>,</span>
<span>                                          transmitter_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>                                            categories <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ligand'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                          receiver_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>                                            categories <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'receptor'</span><span class="op">)</span><span class="op">)</span>, entity_types <span class="op">=</span> <span class="st">"protein"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g.intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/graph_from_edgelist.html" class="external-link">graph_from_edgelist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">intercell_net</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source_genesymbol"</span>, <span class="st">"target_genesymbol"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                   directed <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">g.adj_intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/as_adjacency_matrix.html" class="external-link">as_adjacency_matrix</a></span><span class="op">(</span><span class="va">g.intercell</span>, sparse <span class="op">=</span> <span class="cn">F</span>, type <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cell-cell-communication-analysis">cell-cell communication analysis<a class="anchor" aria-label="anchor" href="#cell-cell-communication-analysis"></a>
</h3>
<p>Once obtained the input, we can build the weighted gene-set list. As
described before, here we should use <code><a href="../reference/build_S_list.html">build_S_list()</a></code> by
providing the gene sets as a three column table. In this way, we can
build a gene set for each cell type, with different weights for the same
genes.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tnbc_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_S_list.html">build_S_list</a></span><span class="op">(</span>S_tab <span class="op">=</span> <span class="va">DEG_tnbc</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="st">"gene"</span>, <span class="st">"score"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                          universe <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g.intercell</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                          min_size <span class="op">=</span> <span class="cn">NULL</span>, max_size <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tnbcCCC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>S_list <span class="op">=</span> <span class="va">tnbc_list</span>, </span>
<span>                         A <span class="op">=</span> <span class="va">g.adj_intercell</span>, </span>
<span>                         k <span class="op">=</span> <span class="fl">99</span>, </span>
<span>                         ct_type <span class="op">=</span> <span class="st">"communication"</span>,</span>
<span>                         mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, perm_link <span class="op">=</span> <span class="st">"degree"</span>, perm_weights <span class="op">=</span> <span class="st">"degree"</span>, </span>
<span>                         cut_par <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                         mc_cores_ct <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p>By specifying <code>ct_type = "communication"</code>, the
<code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code> function will consider genes shared among
gene sets in the calculation, as they are genes in different cell types
with different score, and return also a list of interacting genes
tables, one for each cell type pair. The output will then be a list with
two object: * the results of crosstalk calculation</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tnbcCCC</span><span class="op">$</span><span class="va">ct_res</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   S1_name            S2_name          c S1_size S2_size S1_S2_size S2_S1_size</span></span>
<span><span class="co">#&gt; 1  Bcells                CAF 0.62820171     246     312         11         23</span></span>
<span><span class="co">#&gt; 2  Bcells         Cancer_epi 0.06259519     246     240          5          4</span></span>
<span><span class="co">#&gt; 3  Bcells Cancer_epi_cycling 0.26639117     246     201         14          3</span></span>
<span><span class="co">#&gt; 4  Bcells                 DC 0.58259489     246     286         23         22</span></span>
<span><span class="co">#&gt; 5  Bcells        Endothelial 0.19706308     246     354         13         30</span></span>
<span><span class="co">#&gt;   dL   L        r_c        u1        u2</span></span>
<span><span class="co">#&gt; 1 25 418 0.05980861 1.8159851 5.2910707</span></span>
<span><span class="co">#&gt; 2  6 425 0.01411765 0.7316357 0.3078850</span></span>
<span><span class="co">#&gt; 3 14 367 0.03814714 2.6078636 0.2809384</span></span>
<span><span class="co">#&gt; 4 61 475 0.12842105 3.5281022 1.1899466</span></span>
<span><span class="co">#&gt; 5 33 523 0.06309751 1.4711100 2.4289122</span></span>
<span><span class="co">#&gt;                                                                                                                                                                     S1</span></span>
<span><span class="co">#&gt; 1                                                                                                  TGFB1;SELL;CXCR4;CD69;PTPRC;HLA-DRA;EZR;PTPN6;HLA-DPA1;HLA-DRB1;LTB</span></span>
<span><span class="co">#&gt; 2                                                                                                                                       TGFB1;CD74;CXCR4;TNFRSF14;CD69</span></span>
<span><span class="co">#&gt; 3                                                        HLA-DQB1;HLA-DPB1;HLA-DRA;HLA-DPA1;HLA-DRB1;HLA-DQA1;HLA-DOB;HLA-DMA;HLA-DMB;HLA-DRB5;CD74;CXCR4;TNFRSF14;EZR</span></span>
<span><span class="co">#&gt; 4 HLA-DMA;HLA-DMB;PTPN6;CD37;CD19;CD53;HLA-DQB1;HLA-DPB1;HLA-DRA;HLA-DPA1;HLA-DRB1;HLA-DQA1;HLA-DOB;HLA-DRB5;IL16;CD40;TGFB1;TNFRSF13C;TNFRSF13B;CD74;CXCR4;CSK;ADGRE5</span></span>
<span><span class="co">#&gt; 5                                                                                            PTPN6;CSK;SELL;PLCG2;TGFB1;HLA-DRA;HLA-DRB1;EZR;LTB;CXCR4;PTPRC;CCR6;CD53</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                 S2</span></span>
<span><span class="co">#&gt; 1                                            DCN;MMP2;FSTL1;VCAN;CXCL12;MYL9;THBS1;LGALS1;CD63;MXRA5;ITGB1;VCAM1;SDC2;ITGAV;IL6ST;CXCL14;ITGB5;LPP;TGFBR2;GAL;CD81;TNFRSF1A;TGFBR3</span></span>
<span><span class="co">#&gt; 2                                                                                                                                                            APP;MIF;CXCL14;S100A8</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                   CTSV;MIF;ICAM1</span></span>
<span><span class="co">#&gt; 4                                            CD74;CSF2RB;HLA-DRA;CD4;LTB;HLA-DRB1;CD37;CTSS;CXCR4;LYN;TNFSF13B;CD53;HLA-DMB;SYK;TNFSF9;HLA-DMA;HLA-C;LGMN;TGFBR2;TGFB1;PECAM1;CD55</span></span>
<span><span class="co">#&gt; 5 PECAM1;CD34;INSR;KDR;ACVRL1;TGFBR2;IL6ST;CD81;MMP2;FSTL1;ITGB1;APP;LIFR;VCAM1;TNFRSF1A;CXCL12;TGFBR3;THBS1;CSF2RB;ITGAV;LGALS9;TGFB1;CD63;JAK1;BCR;TNFSF10;HLA-C;LPP;CD109;ICAM1</span></span>
<span><span class="co">#&gt;            s    pA    pU            p</span></span>
<span><span class="co">#&gt; 1 0.61295845 0.954 0.023 1.057462e-01</span></span>
<span><span class="co">#&gt; 2 0.02719497 0.913 0.128 3.677411e-01</span></span>
<span><span class="co">#&gt; 3 0.96603045 0.020 0.001 2.363956e-04</span></span>
<span><span class="co">#&gt; 4 2.81351583 0.001 0.001 1.481551e-05</span></span>
<span><span class="co">#&gt; 5 0.02875915 0.998 0.348 7.145970e-01</span></span></code></pre>
<ul>
<li>a list, with a table for each cell type pair with the detail of the
interactions.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tnbcCCC</span><span class="op">$</span><span class="va">ct_info</span><span class="op">$</span><span class="va">`CAF|Cancer_epi_cycling`</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   S1_name S1_gene            S2_name S2_gene         u12</span></span>
<span><span class="co">#&gt; 1     CAF    DDR1 Cancer_epi_cycling    CDK1 0.000000000</span></span>
<span><span class="co">#&gt; 2     CAF    DDR1 Cancer_epi_cycling  TM4SF1 0.000000000</span></span>
<span><span class="co">#&gt; 3     CAF    PLAT Cancer_epi_cycling   ANXA2 0.016909091</span></span>
<span><span class="co">#&gt; 4     CAF  LGALS3 Cancer_epi_cycling   ANXA2 0.000000000</span></span>
<span><span class="co">#&gt; 5     CAF    APOE Cancer_epi_cycling     LSR 0.002431027</span></span></code></pre>
<p>The communication network can be visualized overlapped to the map of
the cells by using <code><a href="../reference/plot_communication_network.html">plot_communication_network()</a></code>. Here we
will use only significant communication (<span class="math inline">\(\alpha\)</span> &lt; 0.01).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccc_res</span> <span class="op">&lt;-</span> <span class="va">tnbcCCC</span><span class="op">$</span><span class="va">ct_res</span></span>
<span><span class="va">ccc_res</span> <span class="op">&lt;-</span> <span class="va">ccc_res</span><span class="op">[</span><span class="va">ccc_res</span><span class="op">$</span><span class="va">pA</span> <span class="op">&lt;</span> <span class="fl">0.01</span> <span class="op">|</span> <span class="va">ccc_res</span><span class="op">$</span><span class="va">pU</span> <span class="op">&lt;</span> <span class="fl">0.01</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">cl_cell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cell <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tnbc</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>,</span>
<span>                      cluster <span class="op">=</span> <span class="va">tnbc</span><span class="op">$</span><span class="va">cell_type</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_communication_network.html">plot_communication_network</a></span><span class="op">(</span>cell_emb <span class="op">=</span> <span class="va">tnbc</span><span class="op">@</span><span class="va">reductions</span><span class="op">$</span><span class="va">tsne</span><span class="op">@</span><span class="va">cell.embeddings</span>, </span>
<span>                           cl_res <span class="op">=</span> <span class="va">ccc_res</span>, </span>
<span>                           cl_cell <span class="op">=</span> <span class="va">cl_cell</span>, </span>
<span>                           palette <span class="op">=</span> <span class="va">ct_pal</span>, <span class="co">#palette built at the beginning of the vignette </span></span>
<span>                           e_scale <span class="op">=</span> <span class="fl">9</span>, label <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                           save_file <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="data/ccc_net.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="ct-gene-classification">CT gene classification<a class="anchor" aria-label="anchor" href="#ct-gene-classification"></a>
</h3>
<p>Here, in communication analysis, reference model should be built by
using all cell type, and all tested genes but without weights (u = 1).
In this way, we can normalize the number of crosstalk and interactor of
each gene affected by the condition, over all possibilities. To do so,
we need to perform crosstalk analysis and gene classification of the
reference model, to then use it in our TNBC communication gene
classification analysis, as described below.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG_tnbc_mod</span> <span class="op">&lt;-</span> <span class="va">DEG_tnbc</span></span>
<span><span class="va">DEG_tnbc_mod</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="co">#to consider all genes with same weigths</span></span>
<span></span>
<span><span class="va">tnbc_general</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_S_list.html">build_S_list</a></span><span class="op">(</span>S_tab <span class="op">=</span> <span class="va">DEG_tnbc_mod</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="st">"gene"</span>, <span class="st">"score"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                             universe <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g.intercell</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>, min_size <span class="op">=</span> <span class="cn">NULL</span>, max_size <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="va">general_tnbcCCC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>S_list <span class="op">=</span> <span class="va">tnbc_general</span>, </span>
<span>                                 A <span class="op">=</span> <span class="va">g.adj_intercell</span>, </span>
<span>                                 k <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                                 ct_type <span class="op">=</span> <span class="st">"communication"</span>,</span>
<span>                                 mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, mc_cores_ct <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                 perm_link <span class="op">=</span> <span class="st">"degree"</span>, perm_weights <span class="op">=</span> <span class="st">"degree"</span>, cut_par <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="co">#geneal CCC model</span></span>
<span></span>
<span><span class="va">general_cl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_classification.html">gene_classification</a></span><span class="op">(</span>gs_ct <span class="op">=</span> <span class="va">general_tnbcCCC</span><span class="op">$</span><span class="va">ct_res</span>, </span>
<span>                                  A <span class="op">=</span> <span class="va">g.adj_intercell</span>, n_cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="co">#general gene classification of CCC genes</span></span>
<span></span>
<span><span class="va">fr.tnbc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_classification.html">gene_classification</a></span><span class="op">(</span>gs_ct <span class="op">=</span> <span class="va">ccc_res</span>, <span class="co">#only significant communications</span></span>
<span>                               A <span class="op">=</span> <span class="va">g.adj_intercell</span>, </span>
<span>                               ct_ref <span class="op">=</span> <span class="va">general_cl</span>, <span class="co">#general gene classification to calculate degree ratio and crosstalk activity</span></span>
<span>                               n_cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">fr.tnbc</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;     gene       r_q r_d d_q d_d             d_q_S</span></span>
<span><span class="co">#&gt; 1  ACKR1 0.3750000   1   3  14    TAM;Tcells;CAF</span></span>
<span><span class="co">#&gt; 2  ACTR2 0.3333333   1   1   1       Endothelial</span></span>
<span><span class="co">#&gt; 3 ACVRL1 0.4444444   1   4   2 TAM;Tcells;CAF;DC</span></span>
<span><span class="co">#&gt; 4 ADAM15 0.1250000   1   1   1            Tcells</span></span>
<span><span class="co">#&gt; 5 ADGRE5 0.5000000   1   2   1         DC;Bcells</span></span>
<span><span class="co">#&gt;                                                                             d_d_gene</span></span>
<span><span class="co">#&gt; 1 CXCL9;CXCL10;CCL2;CXCL2;CXCL1;CXCL11;CCL13;CCL7;CXCL8;CCL5;CXCL13;CCL18;CXCL3;CCL8</span></span>
<span><span class="co">#&gt; 2                                                                               LDLR</span></span>
<span><span class="co">#&gt; 3                                                                        TGFB1;TGFB3</span></span>
<span><span class="co">#&gt; 4                                                                                FYN</span></span>
<span><span class="co">#&gt; 5                                                                               CD55</span></span>
<span><span class="co">#&gt;   q  d</span></span>
<span><span class="co">#&gt; 1 8 14</span></span>
<span><span class="co">#&gt; 2 3  1</span></span>
<span><span class="co">#&gt; 3 9  2</span></span>
<span><span class="co">#&gt; 4 8  1</span></span>
<span><span class="co">#&gt; 5 4  1</span></span>
<span><span class="co">#&gt;                                                                               q_S</span></span>
<span><span class="co">#&gt; 1             Plasma_cells;TAM;Tcells;Bcells;CAF;Cancer_epi;Cancer_epi_cycling;DC</span></span>
<span><span class="co">#&gt; 2                                                 Endothelial;Plasma_cells;Bcells</span></span>
<span><span class="co">#&gt; 3 Cancer_epi;Cancer_epi_cycling;DC;Endothelial;Plasma_cells;TAM;Tcells;Bcells;CAF</span></span>
<span><span class="co">#&gt; 4     Cancer_epi;Cancer_epi_cycling;DC;Endothelial;Plasma_cells;TAM;Tcells;Bcells</span></span>
<span><span class="co">#&gt; 5                                                      DC;Plasma_cells;TAM;Bcells</span></span>
<span><span class="co">#&gt;                                                                               d_gene</span></span>
<span><span class="co">#&gt; 1 CCL5;CXCL2;CXCL9;CXCL10;CCL2;CXCL1;CXCL8;CXCL11;CCL13;CCL7;CCL8;CXCL13;CCL18;CXCL3</span></span>
<span><span class="co">#&gt; 2                                                                               LDLR</span></span>
<span><span class="co">#&gt; 3                                                                        TGFB1;TGFB3</span></span>
<span><span class="co">#&gt; 4                                                                                FYN</span></span>
<span><span class="co">#&gt; 5                                                                               CD55</span></span></code></pre>
<p>Now we can plot saturation of interactor diversity and crosstalk
diversity compared to their absolute values, to prioritize genes, as
done before in Intracellular crosstalk section</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">fr.tnbc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">r_q</span>, y <span class="op">=</span> <span class="va">d_q</span>, label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_test</a></span><span class="op">(</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">fr.tnbc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">r_d</span>, y <span class="op">=</span> <span class="va">d_d</span>, label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span>seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_test</a></span><span class="op">(</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> </span></code></pre></div>
<p><img src="data/fr_ccc.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="integrated-crosstalk">Integrated crosstalk<a class="anchor" aria-label="anchor" href="#integrated-crosstalk"></a>
</h2>
<p>Lastly, we can perform integrated crosstalk analysis, which study the
interaction between genes involved in a communication with a cell type
and the intracellular mechanisms (pathways). Here, we will study cancer
- CAF communication and the crosstalk with Hallmarks pathways. We can
perform this analysis in two ways:</p>
<ul>
<li>Single integrated analysis: focused only on cancer genes involved in
the communication with CAFs, and their crosstalks with Hallmarks
pathways involving altered genes</li>
<li>Integrated analysis: focused on multiple inter-intracellular
crosstalks. Here we will apply this analysis on cancer-CAF
communication, to obtain results in both directions</li>
</ul>
<p>In both analysis, we can again filter the results to maintain only
the significant ones, as done for intercellular and intracellular.</p>
<div class="section level3">
<h3 id="single-integrated-analysis">Single integrated analysis<a class="anchor" aria-label="anchor" href="#single-integrated-analysis"></a>
</h3>
<p>To perform this analysis we simply need the data already used for
intracellular and intercellular crosstalks. In detail, we need the
cancer communication gene set, but considering as altered (with u &gt;
0) only the genes involved in the communication with CAFs, and the
pathway gene set list created for intracellular communication.
<code><a href="../reference/single_integrated_cross_talk.html">single_integrated_cross_talk()</a></code> takes as input a gene set
list, composed by both these inputs, the name of the communicating gene
set (<code>ref</code>), and a biological network (here we will use
String, as done in intrecellular section). The function returns a table
with the same structure of the ones obtained with
<code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code>. Here, S1 is always the communication gene
set (<code>"Cancer_epi"</code> in this case), as this function do not
calculate the the crosstalk between each possible pair, to not rerun
intracellular analysis and do not complicate null models building.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ccc_info</span> <span class="op">&lt;-</span> <span class="va">tnbcCCC</span><span class="op">$</span><span class="va">ct_info</span><span class="op">[[</span><span class="st">"CAF|Cancer_epi"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">can_ccc</span> <span class="op">&lt;-</span> <span class="va">tnbc_list</span><span class="op">$</span><span class="va">Cancer_epi</span></span>
<span><span class="va">can_ccc</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">can_ccc</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ccc_info</span><span class="op">$</span><span class="va">S2_gene</span><span class="op">[</span><span class="va">ccc_info</span><span class="op">$</span><span class="va">u12</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">gs_list_int</span> <span class="op">&lt;-</span> <span class="va">gs_list</span></span>
<span><span class="va">gs_list_int</span><span class="op">[[</span><span class="st">"Cancer_epi"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">can_ccc</span></span>
<span></span>
<span><span class="va">integr_can</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/single_integrated_cross_talk.html">single_integrated_cross_talk</a></span><span class="op">(</span>S_list <span class="op">=</span> <span class="va">gs_list_int</span>, <span class="co">#Hallmark gene set list built for intracellular analysis</span></span>
<span>                                           ref <span class="op">=</span> <span class="st">"Cancer_epi"</span>, <span class="co">#name of the communication gene set</span></span>
<span>                                           A <span class="op">=</span> <span class="va">g.adj</span>, <span class="co">#String used for intracellular analysis</span></span>
<span>                                           k <span class="op">=</span> <span class="fl">99</span>, </span>
<span>                                           perm_link <span class="op">=</span> <span class="st">"degree"</span>, perm_weights <span class="op">=</span> <span class="st">"degree"</span>, </span>
<span>                                           cut_par <span class="op">=</span> <span class="cn">NULL</span>, mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, mc_cores_ct <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">integr_can</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;       S1_name                          S2_name          c S1_size S2_size</span></span>
<span><span class="co">#&gt; 11 Cancer_epi     HALLMARK_ALLOGRAFT_REJECTION 0.01775875     200      31</span></span>
<span><span class="co">#&gt; 12 Cancer_epi       HALLMARK_ANDROGEN_RESPONSE 0.01993079     229      27</span></span>
<span><span class="co">#&gt; 13 Cancer_epi               HALLMARK_APOPTOSIS 0.01055460     213      38</span></span>
<span><span class="co">#&gt; 14 Cancer_epi HALLMARK_CHOLESTEROL_HOMEOSTASIS 0.02550742     229      13</span></span>
<span><span class="co">#&gt; 15 Cancer_epi             HALLMARK_COAGULATION 0.01495281     225      18</span></span>
<span><span class="co">#&gt;    S1_S2_size S2_S1_size dL  L        r_c         u1         u2        S1</span></span>
<span><span class="co">#&gt; 11          1          2  2 66 0.03030303 0.08581545 0.20694122      RPSA</span></span>
<span><span class="co">#&gt; 12          2          1  2 29 0.06896552 0.19987201 0.09971779   APP;LSR</span></span>
<span><span class="co">#&gt; 13          1          1  1 39 0.02564103 0.13183863 0.08005698     PTPRF</span></span>
<span><span class="co">#&gt; 14          2          2  2 16 0.12500000 0.25920987 0.19745251 APP;PTPRF</span></span>
<span><span class="co">#&gt; 15          1          1  1 28 0.03571429 0.12737124 0.11739553       APP</span></span>
<span><span class="co">#&gt;             S2          s    pA    pU            p</span></span>
<span><span class="co">#&gt; 11 RPL39;RPS19 0.04759216 0.111 0.002 2.089649e-03</span></span>
<span><span class="co">#&gt; 12         DBI 0.06587219 0.045 0.001 4.953982e-04</span></span>
<span><span class="co">#&gt; 13      CTNNB1 0.02288445 0.140 0.006 6.788971e-03</span></span>
<span><span class="co">#&gt; 14  CLU;CTNNB1 0.10891385 0.004 0.001 5.371686e-05</span></span>
<span><span class="co">#&gt; 15         CLU 0.03774268 0.083 0.004 2.991445e-03</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="integrated-analysis">Integrated analysis<a class="anchor" aria-label="anchor" href="#integrated-analysis"></a>
</h3>
<p>To perform this analysis we need:</p>
<ul>
<li>the results of cell-cell communication, in a form of a list with the
detail of all the communication we want to study (here only CAF - Cancer
cycling)</li>
<li>weighted gene set lists: we need the gene set built for cell-cell
communication, as also intracellular gene sets. Here so far we have
analyzed only cancer intracellular crosstalk, so we need to obtain CAF
list.</li>
<li>a biological network: here we will again use string network, as
before. <code><a href="../reference/integrated_cross_talk.html">integrated_cross_talk()</a></code> function is a wrapper of
<code><a href="../reference/single_integrated_cross_talk.html">single_integrated_cross_talk()</a></code>, that run this analysis on
each communication present in <code>ccc_list</code>, in both directions,
and returns a table with all integrated analysis performed. To
discriminate different communicating gene sets, the first column
contains the name fo the <code>ccc_list</code> object used</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1) communication results: a list with details of all communications we want to investigate</span></span>
<span><span class="va">ccc_info</span> <span class="op">&lt;-</span> <span class="va">tnbcCCC</span><span class="op">$</span><span class="va">ct_info</span><span class="op">[</span><span class="st">"CAF|Cancer_epi"</span><span class="op">]</span> </span>
<span></span>
<span><span class="co"># 2) gene set list: we are missing Cancer cycling ones, so we built it as done for CAF</span></span>
<span><span class="va">g.wCaf</span> <span class="op">&lt;-</span> <span class="va">DEG_tnbc</span><span class="op">[</span><span class="va">DEG_tnbc</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="st">"CAF"</span>,<span class="op">]</span></span>
<span><span class="va">g.wCaf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">g.wCaf</span><span class="op">$</span><span class="va">score</span>, <span class="va">g.wCaf</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gs_listCaf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_S_list.html">build_S_list</a></span><span class="op">(</span>S_tab <span class="op">=</span> <span class="va">ptw.H</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, g_u <span class="op">=</span> <span class="va">g.wCaf</span>,  universe <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">stringv12</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span>, min_size <span class="op">=</span> <span class="cn">NULL</span>, max_size <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#here is the list both intracellular gene set lists</span></span>
<span><span class="va">gs_list_tot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>CAF <span class="op">=</span> <span class="va">gs_listCaf</span>,</span>
<span>                    Cancer_epi <span class="op">=</span> <span class="va">gs_list</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">integrated_tnbc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integrated_cross_talk.html">integrated_cross_talk</a></span><span class="op">(</span>cl_list <span class="op">=</span> <span class="va">tnbc_list</span>, <span class="co">#CCC gene sets list</span></span>
<span>                                         ccc_list <span class="op">=</span> <span class="va">ccc_info</span>, <span class="co">#CCC detailed results of the communications of interest</span></span>
<span>                                         gs_list <span class="op">=</span> <span class="va">gs_list_tot</span>, <span class="co">#intracellular gene set lists</span></span>
<span>                                         A <span class="op">=</span> <span class="va">g.adj</span>, <span class="co">#biological network</span></span>
<span>                                         k <span class="op">=</span> <span class="fl">99</span>, mc_cores_perm <span class="op">=</span> <span class="fl">2</span>, mc_cores_ct <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                         perm_link <span class="op">=</span> <span class="st">"degree"</span>, perm_weights <span class="op">=</span> <span class="st">"degree"</span>, cut_par <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span>                                   </span>
<span></span>
<span><span class="va">integrated_tnbc</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;                       ccc S1_name                      S2_name          c</span></span>
<span><span class="co">#&gt; 52 CAF|Cancer_epi_cycling     CAF        HALLMARK_ADIPOGENESIS 0.14920848</span></span>
<span><span class="co">#&gt; 53 CAF|Cancer_epi_cycling     CAF HALLMARK_ALLOGRAFT_REJECTION 0.52238288</span></span>
<span><span class="co">#&gt; 54 CAF|Cancer_epi_cycling     CAF   HALLMARK_ANDROGEN_RESPONSE 0.06300081</span></span>
<span><span class="co">#&gt; 55 CAF|Cancer_epi_cycling     CAF        HALLMARK_ANGIOGENESIS 1.92839471</span></span>
<span><span class="co">#&gt; 56 CAF|Cancer_epi_cycling     CAF     HALLMARK_APICAL_JUNCTION 0.26458631</span></span>
<span><span class="co">#&gt;    S1_size S2_size S1_S2_size S2_S1_size dL  L       r_c        u1        u2</span></span>
<span><span class="co">#&gt; 52     301      44          4          2  4 30 0.1333333 0.8481891 0.2531009</span></span>
<span><span class="co">#&gt; 53     276      21          7          2  7 38 0.1842105 1.2819973 0.6212551</span></span>
<span><span class="co">#&gt; 54     302      34          3          1  3 29 0.1034483 0.3717228 0.1694833</span></span>
<span><span class="co">#&gt; 55     293       4          8          3  9 15 0.6000000 2.4254667 1.7978163</span></span>
<span><span class="co">#&gt; 56     283      44          5          5  6 55 0.1090909 0.9155698 0.6608048</span></span>
<span><span class="co">#&gt;                                                   S1</span></span>
<span><span class="co">#&gt; 52                            ITGB1;MMP2;ITGAV;ITGA5</span></span>
<span><span class="co">#&gt; 53        ITGB1;ITGB5;ITGA5;MMP2;CD63;MMP14;TNFRSF1A</span></span>
<span><span class="co">#&gt; 54                                 ITGB1;ITGB5;ITGA5</span></span>
<span><span class="co">#&gt; 55 DCN;COL6A3;ITGB1;COL16A1;MMP2;CD63;MMP14;TNFRSF1A</span></span>
<span><span class="co">#&gt; 56                       ITGB5;ITGA5;LUM;APOE;TUBA1A</span></span>
<span><span class="co">#&gt;                             S2          s   pA   pU           p</span></span>
<span><span class="co">#&gt; 52                CD151;COL4A1 0.23232204 0.43 0.01 0.027731303</span></span>
<span><span class="co">#&gt; 53                  FLNA;TIMP1 1.42112231 0.02 0.01 0.001903439</span></span>
<span><span class="co">#&gt; 54                       ACTN1 0.08860062 0.65 0.01 0.039233695</span></span>
<span><span class="co">#&gt; 55         COL3A1;COL5A2;TIMP1 5.76775101 0.01 0.01 0.001021034</span></span>
<span><span class="co">#&gt; 56 ACTN1;FBN1;PARVA;TGFBI;TJP1 0.37063518 0.33 0.02 0.039736525</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Chen2022" class="csl-entry">
Chen, Yunshun, and Gordon Smyth. 2022. <span>“<span class="nocase">Data,
R code and output Seurat Objects for single cell RNA-seq analysis of
human breast tissues</span>,”</span> February. <a href="https://doi.org/10.6084/m9.figshare.17058077.v1" class="external-link">https://doi.org/10.6084/m9.figshare.17058077.v1</a>.
</div>
<div id="ref-Hao2021" class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erica Andersen-Nissen, William M. Mauck,
Shiwei Zheng, Andrew Butler, Maddie J. Lee, et al. 2021. <span>“<span class="nocase">Integrated analysis of multimodal single-cell
data</span>.”</span> <em>Cell</em> 184 (13): 3573–3587.e29. <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">https://doi.org/10.1016/j.cell.2021.04.048</a>.
</div>
<div id="ref-pal2021single" class="csl-entry">
Pal, Bhupinder, Yunshun Chen, François Vaillant, Bianca D Capaldo,
Rachel Joyce, Xiaoyu Song, Vanessa L Bryant, et al. 2021. <span>“A
Single-Cell RNA Expression Atlas of Normal, Preneoplastic and
Tumorigenic States in the Human Breast.”</span> <em>The EMBO
Journal</em> 40 (11): e107333.
</div>
<div id="ref-szklarczyk2021" class="csl-entry">
Szklarczyk, Damian, Annika L Gable, Katerina C Nastou, David Lyon,
Rebecca Kirsch, Sampo Pyysalo, Nadezhda T Doncheva, et al. 2021.
<span>“The STRING Database in 2021: Customizable Protein–Protein
Networks, and Functional Characterization of User-Uploaded
Gene/Measurement Sets.”</span> <em>Nucleic Acids Research</em> 49 (D1):
D605–12.
</div>
<div id="ref-turei2021" class="csl-entry">
Türei, Dénes, Alberto Valdeolivas, Lejla Gul, Nicolàs Palacio-Escat,
Michal Klein, Olga Ivanova, Márton Ölbei, et al. 2021. <span>“Integrated
Intra-and Intercellular Signaling Knowledge for Multicellular Omics
Analysis.”</span> <em>Molecular Systems Biology</em> 17 (3): e9923.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alice Chiodi, Ettore Mosca.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
