<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Ulisse">
<title>Ulisse workflow 1: Pathway Cross-Talk analysis of bulk data • Ulisse</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Ulisse workflow 1: Pathway Cross-Talk analysis of bulk data">
<meta property="og:description" content="Ulisse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Ulisse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Ulisse_bulk.html">Ulisse workflow 1: Pathway Cross-Talk analysis of bulk data</a>
    <a class="dropdown-item" href="../articles/Ulisse_sc.html">Ulisse workflow 2: Cell-cell cross-talk analysis on single cell data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Ulisse_bulk_files/kePrint-0.0.1/kePrint.js"></script><link href="Ulisse_bulk_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Ulisse workflow 1: Pathway Cross-Talk analysis of bulk data</h1>
            
            <h4 data-toc-skip class="date">2024-03-29</h4>
      
      
      <div class="d-none name"><code>Ulisse_bulk.Rmd</code></div>
    </div>

    
    
<p>In this vignette we will show the usage of Ulisse on gene-sets or
pathway data. As an example, we will apply cross-talk analysis on
mutational bulk data of breast cancer obtained from TCGA. The very same
pipeline can be used to analyse gene of interest of cell clusters in
single cell data. In this case, the only difference is that this
pipeline can be applied to only a cluster at time.</p>
<div class="section level2">
<h2 id="downloading-and-preparing-the-data">Downloading and preparing the data<a class="anchor" aria-label="anchor" href="#downloading-and-preparing-the-data"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Ulisse</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks" class="external-link">TCGAbiolinks</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">STRINGdb</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<p>Ulisse needs mainly two inputs: a gene list, that can be or not
ranked, and a biological network.</p>
<p>Breast cancer mutational data obtained from TCGA will be used to
obtain the ranked gene list. The biological network will be obtained
from STRING database.</p>
<div class="section level3">
<h3 id="biological-network">Biological network<a class="anchor" aria-label="anchor" href="#biological-network"></a>
</h3>
<p>In the code below, we obtain STRING data from STRINGdb <span class="citation">(Szklarczyk et al. 2021)</span> and build the network
from all the genes present in the database. The gene names are formatted
with STRING identifiers, so we need to convert it to gene symbol.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">string_db</span> <span class="op">&lt;-</span> <span class="va"><a href="https://rdrr.io/pkg/STRINGdb/man/STRINGdb-class.html" class="external-link">STRINGdb</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span> version<span class="op">=</span><span class="st">"11.0"</span>, </span>
<span>                           species<span class="op">=</span><span class="fl">9606</span>, </span>
<span>                           score_threshold<span class="op">=</span><span class="fl">700</span>, </span>
<span>                           input_directory<span class="op">=</span><span class="st">"."</span> <span class="op">)</span></span>
<span><span class="va">string_proteins</span> <span class="op">&lt;-</span> <span class="va">string_db</span><span class="op">$</span><span class="fu">get_proteins</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">PPI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">string_db</span><span class="op">$</span><span class="fu">get_interactions</span><span class="op">(</span><span class="va">string_proteins</span><span class="op">$</span><span class="va">protein_external_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">PPI.g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/graph_from_edgelist.html" class="external-link">graph_from_edgelist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">PPI</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, directed <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">PPI.g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="va">string_proteins</span><span class="op">$</span><span class="va">preferred_name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">PPI.g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>, <span class="va">string_proteins</span><span class="op">$</span><span class="va">protein_external_id</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">PPI.g</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH a991c72 UN-- 17185 420534 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c)</span></span>
<span><span class="co">#&gt; + edges from a991c72 (vertex names):</span></span>
<span><span class="co">#&gt;  [1] ARF5--SPTBN2    ARF5--KIF13B    ARF5--AP1B1     ARF5--KIF21A   </span></span>
<span><span class="co">#&gt;  [5] ARF5--TMED7     ARF5--ARFGAP1   ARF5--ANK2      ARF5--KLC1     </span></span>
<span><span class="co">#&gt;  [9] ARF5--COPZ2     ARF5--KIF15     ARF5--DCTN5     ARF5--KIF2B    </span></span>
<span><span class="co">#&gt; [13] ARF5--KIF16B    ARF5--KIF25     ARF5--KDELR2    ARF5--RAB1B    </span></span>
<span><span class="co">#&gt; [17] ARF5--TMED10    ARF5--KIF3B     ARF5--RACGAP1   ARF5--SPTBN4   </span></span>
<span><span class="co">#&gt; [21] ARF5--GBF1      ARF5--DYNC1I2   ARF5--INS       ARF5--CYTH3    </span></span>
<span><span class="co">#&gt; [25] ARF5--CAPZB     ARF5--RAB11FIP3 ARF5--ACAP2     ARF5--KIF3C    </span></span>
<span><span class="co">#&gt; [29] ARF5--DYNC1LI1  ARF5--KIFAP3    ARF5--COPA      ARF5--YKT6     </span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="ranked-gene-list">Ranked gene list<a class="anchor" aria-label="anchor" href="#ranked-gene-list"></a>
</h3>
<p>Then, we will download the data from TGCA by using TCGAbiolinks
package <span class="citation">Mounir et al. (2019)</span>. Here, we
want to prioritize the genes considering the frequency of mutations
observed among the samples. So, we handle the data to obtain a gene by
sample matrix, with the number of mutation observed for each gene. By
making the row mean we will obtain the frequency of mutation for each
gene in the samples, and we will use this value as a rank, cosnidering
the top 200 genes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCquery.html" class="external-link">GDCquery</a></span><span class="op">(</span>project <span class="op">=</span> <span class="st">"TCGA-BRCA"</span>, </span>
<span>                  data.category <span class="op">=</span> <span class="st">"Simple Nucleotide Variation"</span>,</span>
<span>                  data.type <span class="op">=</span> <span class="st">"Masked Somatic Mutation"</span><span class="op">)</span></span>
<span>                  <span class="co">#)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCdownload.html" class="external-link">GDCdownload</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span> <span class="co">#have to be done only the first time</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">TCGAbiolinks</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TCGAbiolinks/man/GDCprepare.html" class="external-link">GDCprepare</a></span><span class="op">(</span><span class="va">query</span>, summarizedExperiment <span class="op">=</span> <span class="cn">F</span>, save <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="op">!</span><span class="va">data</span><span class="op">$</span><span class="va">Entrez_Gene_Id</span> <span class="op">==</span> <span class="fl">0</span>,<span class="op">]</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">SYMBOL</span>, <span class="va">data</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/cast.html" class="external-link">acast</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">Var1</span> <span class="op">~</span> <span class="va">Var2</span><span class="op">)</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">mean</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">mean</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">gene_weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>,</span>
<span>                      weights <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">mean</span>,</span>
<span>                      stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">target_gene_weights</span> <span class="op">&lt;-</span> <span class="va">gene_weights</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathway-data">Pathway Data<a class="anchor" aria-label="anchor" href="#pathway-data"></a>
</h3>
<p>Now that we have the two needed input, we can proceed in the
pipeline. The subsequent step will be:</p>
<ul>
<li><p>group genes into a (ranked) gene-sets (or pathways) to obtain a
gene-set list</p></li>
<li><p>use the gene-set list and the adjacency matrix of the biological
network to calculate cross-talk.</p></li>
</ul>
<p>As an example we will use MSigDB Hallmark pathways <span class="citation">(Liberzon et al. 2015)</span> to obtain the gene-set
list by using <code><a href="../reference/preparing_msigdb_list.html">preparing_msigdb_list()</a></code>. This function
downloads a defines gene-set database and uses it to group the gene in
the gene list. The function will return a list composed by a named
vector for each pathway. The vectors are composed by the ranks named
after the genes in the gene-set. It returns only the pathways that are
composed by at least <code>min_size</code> and maximum
<code>max_size</code> genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">ptw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparing_msigdb_list.html">preparing_msigdb_list</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, </span>
<span>                             category <span class="op">=</span> <span class="st">"H"</span>, </span>
<span>                             type <span class="op">=</span> <span class="st">"gene_symbol"</span>, min_size <span class="op">=</span> <span class="fl">1</span>, max_size <span class="op">=</span> <span class="fl">500</span>, </span>
<span>                             gene <span class="op">=</span> <span class="va">target_gene_weights</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>                             weights <span class="op">=</span> <span class="va">target_gene_weights</span><span class="op">$</span><span class="va">weights</span><span class="op">)</span></span>
<span><span class="va">ptw</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">#&gt;       AKT1      BRCA1       FLNA </span></span>
<span><span class="co">#&gt; 0.02938197 0.02532928 0.02938197 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $HALLMARK_ANDROGEN_RESPONSE</span></span>
<span><span class="co">#&gt;       AKT1 </span></span>
<span><span class="co">#&gt; 0.02938197 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $HALLMARK_APICAL_JUNCTION</span></span>
<span><span class="co">#&gt;       CDH1       FBN1       MYH9        NF1       PTEN </span></span>
<span><span class="co">#&gt; 0.13880446 0.02938197 0.03444782 0.04457953 0.05876393</span></span></code></pre>
<p>Ulisse contains also a function, <code><a href="../reference/preparing_gs_list.html">preparing_gs_list()</a></code> to
build gene-set list from custom resources. To exemplify the usage of
this function, here we download Hallmark db from MSigDB by using msigdbr
package <span class="citation">(Dolgalev 2021)</span> to then build the
very same gene-set list.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msig_out</span> <span class="op">&lt;-</span> <span class="fu">msigdbr</span><span class="fu">::</span><span class="fu"><a href="https://igordot.github.io/msigdbr/reference/msigdbr.html" class="external-link">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span><span class="va">ptw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparing_gs_list.html">preparing_gs_list</a></span><span class="op">(</span>gs_names <span class="op">=</span>  <span class="va">msig_out</span><span class="op">$</span><span class="va">gs_name</span>,</span>
<span>                         gs_genes <span class="op">=</span> <span class="va">msig_out</span><span class="op">$</span><span class="va">gene_symbol</span>, </span>
<span>                         weights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">target_gene_weights</span><span class="op">$</span><span class="va">weights</span>, <span class="va">target_gene_weights</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>, </span>
<span>                         min_size <span class="op">=</span> <span class="fl">1</span>, max_size <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="cross-talk">Cross-talk<a class="anchor" aria-label="anchor" href="#cross-talk"></a>
</h2>
<p>Now we have all the needed input for calculating cross-talk. The
biological network will be provided to <code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code>
function as an adjacency matrix. Cross-talk calculation can be
parallelized by increasing the <code>mc_cores_pct</code> and
<code>mc_cores_perm</code> values: the first parallelize the calculation
of the cross-talk on the gene-sets, the second all the permutations used
for p-value calculation, thus multiplying <code>mc_cores_pct</code> in
some part of the function. The number of permutations <code>k</code>
corresponds to the number of permutations needed that, together with the
original matrix, are used for the calculation of p-value and FDR. Here
we set the number to 49, which means that the p-value and FDR are
calculated on 49 permuted matrices + the original one, so 50 matrix in
total (and thus the minimal p-value will be 1/50 = 0.02). Other inputs
parameter are:</p>
<ul>
<li><p><code>shared</code> which should be set to <code>FALSE</code> in
gene-set cross-talk to avoid considering shared genes. Pathways are
defined to group genes that occur to a particular molecular function.
However, they are not mutually exclusive, as a gene may be involved in
multiple functions. Considering shared genes will lead to consider in
cross-talk calculations even links internal to the pathway. Thus, when
calculating cross-talk on gene-sets or pathways this parameter should be
set to <code>FALSE</code>;</p></li>
<li><p><code>hash</code>: logical, used to speed-up calculation when
lots of gene-sets are used. For pathway cross-talk we suggest to set it
to <code>TRUE</code>;</p></li>
<li><p><code>ct_info</code>: logical, if detailS of gene-gene
interaction in the gene-set should be returned. This might be highly
important in cell-cell cross-talk. Here we set it to
<code>FALSE</code>.</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adj.m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/as_adjacency_matrix.html" class="external-link">as_adjacency_matrix</a></span><span class="op">(</span><span class="va">PPI.g</span>, sparse <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">pct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">ptw</span>, </span>
<span>                     gene_network_adj <span class="op">=</span> <span class="va">adj.m</span>, </span>
<span>                     k <span class="op">=</span> <span class="fl">49</span>, shared <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                     hash <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                     ct_info <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                     mc_cores_perm <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     mc_cores_ct <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">pct</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:right;">
ct_score
</th>
<th style="text-align:right;">
ngenes_gs1
</th>
<th style="text-align:right;">
ngenes_gs2
</th>
<th style="text-align:right;">
nlink
</th>
<th style="text-align:right;">
p_value_link
</th>
<th style="text-align:right;">
FDR_link
</th>
<th style="text-align:right;">
p_adj_BH
</th>
<th style="text-align:right;">
weight_gs1
</th>
<th style="text-align:right;">
weight_gs2
</th>
<th style="text-align:left;">
genes_gs1
</th>
<th style="text-align:left;">
genes_gs2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_ANDROGEN_RESPONSE
</td>
<td style="text-align:right;">
0.0016075
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0547112
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:left;">
BRCA1;FLNA
</td>
<td style="text-align:left;">
AKT1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0107445
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0622889
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0840932
</td>
<td style="text-align:right;">
0.2765957
</td>
<td style="text-align:left;">
AKT1;FLNA;BRCA1
</td>
<td style="text-align:left;">
CDH1;MYH9;NF1;PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE
</td>
<td style="text-align:right;">
0.0045844
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:right;">
0.1560284
</td>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:left;">
BRCA1;GATA3
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS
</td>
<td style="text-align:right;">
0.0015182
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:right;">
0.0516717
</td>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:left;">
BRCA1;ERBB2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_COMPLEMENT
</td>
<td style="text-align:right;">
0.0150929
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:right;">
0.5136778
</td>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:left;">
GATA3;PIK3CA
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_DNA_REPAIR
</td>
<td style="text-align:right;">
0.0192903
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0547112
</td>
<td style="text-align:right;">
0.3525836
</td>
<td style="text-align:left;">
AKT1;BRCA1
</td>
<td style="text-align:left;">
TP53
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_E2F_TARGETS
</td>
<td style="text-align:right;">
0.0132174
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0655673
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:right;">
0.4498480
</td>
<td style="text-align:left;">
AKT1;FLNA
</td>
<td style="text-align:left;">
BRCA1;BRCA2;PRKDC;TP53
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:right;">
0.0016671
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0688075
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:right;">
0.0567376
</td>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:left;">
FLNA;LAMA1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_ESTROGEN_RESPONSE_LATE
</td>
<td style="text-align:right;">
0.0040783
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.1088180
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0293820
</td>
<td style="text-align:right;">
0.1388045
</td>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:left;">
CDH1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
HALLMARK_G2M_CHECKPOINT
</td>
<td style="text-align:right;">
0.0031134
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0655673
</td>
<td style="text-align:right;">
0.0283761
</td>
<td style="text-align:right;">
0.0547112
</td>
<td style="text-align:right;">
0.0911854
</td>
<td style="text-align:left;">
BRCA1;FLNA
</td>
<td style="text-align:left;">
ATRX;BRCA2;KIF4A
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="cross-talk-visualization">Cross-talk visualization<a class="anchor" aria-label="anchor" href="#cross-talk-visualization"></a>
</h3>
<p>Cross-talk results can be filtered to maintain only the significant
ones. The results can be visualized as a network
(<code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code>) or as an heatmap
(<code><a href="../reference/ct_heatmap.html">ct_heatmap()</a></code>) by using Ulisse package functions.
Considering the network, the <code>filtering</code> argument is used to
control which CT have to be visualized in the network. If set to
<code>TRUE</code>, then <code>p_val</code>, <code>FDR</code> and
<code>ct_val</code> are used to identify the significant CT and plot
these with a solid line, whereas all the others will be plotted with a
dashed line. Otherwise, the whole cross-talk result table will be used.
<code>community</code> can be logical (if the community should be
calculated by using <code><a href="https://r.igraph.org/reference/cluster_fast_greedy.html" class="external-link">igraph::fastgreedy.community()</a></code>), or a
community object as calculated with igraph package.
<code>vertex_label</code> can be either logical (if the label of the
vertices should be plotted), or a vector with the label that should be
assigned to each vertex (explained more later in the vignette).
<code>edge_col_by</code> and <code>edge_width</code> parameters control
the value used to color the edges and if their width should be
proportional to that value. <code>edge_adj_col</code> is used to control
the transparency of the edges: as in pathway cross-talk there can be
lots of links it can e useful to control the transparency.
<code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code> function returns the igraph cross-talk
network with the communities under “comm_id” vertex attribute (if
calculated). <code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code> function uses
<code>ggraph</code> package functions <span class="citation">(Pedersen
2022)</span>, which are ggplot2-based <span class="citation">(Wickham
2016)</span>. If <code>file_out</code> is set to <code>NULL</code> the
function returns also the ggplot2 network object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pct_f</span> <span class="op">&lt;-</span> <span class="va">pct</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">pct</span><span class="op">$</span><span class="va">p_value_link</span> <span class="op">&lt;=</span><span class="fl">0.05</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">pct_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_network_CT.html">plot_network_CT</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">pct_f</span>, </span>
<span>                           filtering <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                           community <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                           vertex_label <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                           edge_col_by <span class="op">=</span> <span class="st">"ct_score"</span>, edge_width <span class="op">=</span> <span class="cn">T</span>, edge_adj_col <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>                           file_out <span class="op">=</span><span class="cn">NULL</span>, width <span class="op">=</span> <span class="fl">200</span>, height <span class="op">=</span> <span class="fl">200</span>, res <span class="op">=</span> <span class="fl">300</span>, units <span class="op">=</span> <span class="st">"mm"</span><span class="op">)</span> </span>
<span><span class="va">pct_net</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH cb0e5ae UN-- 26 149 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), comm_id (v/c), ct_score (e/n), ngenes_gs1 (e/n),</span></span>
<span><span class="co">#&gt; | ngenes_gs2 (e/n), nlink (e/n), p_value_link (e/n), FDR_link (e/n),</span></span>
<span><span class="co">#&gt; | p_adj_BH (e/n), weight_gs1 (e/n), weight_gs2 (e/n), genes_gs1 (e/c),</span></span>
<span><span class="co">#&gt; | genes_gs2 (e/c)</span></span>
<span><span class="co">#&gt; + edges from cb0e5ae (vertex names):</span></span>
<span><span class="co">#&gt; [1] HALLMARK_ALLOGRAFT_REJECTION--HALLMARK_ANDROGEN_RESPONSE</span></span>
<span><span class="co">#&gt; [2] HALLMARK_ALLOGRAFT_REJECTION--HALLMARK_APICAL_JUNCTION  </span></span>
<span><span class="co">#&gt; [3] HALLMARK_ALLOGRAFT_REJECTION--HALLMARK_APICAL_SURFACE   </span></span>
<span><span class="co">#&gt; [4] HALLMARK_ALLOGRAFT_REJECTION--HALLMARK_APOPTOSIS        </span></span>
<span><span class="co">#&gt; [5] HALLMARK_ALLOGRAFT_REJECTION--HALLMARK_COMPLEMENT       </span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code></pre>
<p><img src="bulk_data/pct_net.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>Here you can see the significant CT as an heatmap. You can choose the
variable to color the heatmap (here: cross-talk score). The color
parameters are used to set the color of the extremes of the color scale,
so if you want to correct for outliers you can choose not to use minimum
and maximum (default) ad provide your own values.
<code>no_ct_color</code> is used to color the cells in the heatmap
corresponding to the pathways pairs that do not shows CT but that are
necessarily present in an heatmap with a 0 score. Having different
colors for minimum and zero help the heatmap readability. Other
parameters can be passed through <code>...</code> argument. See
complexHeatmap package for more parameters <span class="citation">(Gu,
Eils, and Schlesner 2016)</span>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/ct_heatmap.html">ct_heatmap</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">pct_f</span>, </span>
<span>           color_by <span class="op">=</span> <span class="st">"ct_score"</span>, </span>
<span>           color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightyellow"</span>, <span class="st">"red3"</span><span class="op">)</span>, </span>
<span>           no_ct_color <span class="op">=</span> <span class="st">"whitesmoke"</span>, </span>
<span>           rect_gp <span class="op">=</span> <span class="fu">gpar</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>, </span>
<span>           cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           cluster_columns <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           row_names_gp <span class="op">=</span> <span class="fu">gpar</span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>, column_names_gp <span class="op">=</span> <span class="fu">gpar</span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/pct_f_heatmap.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="gene-functional-relevance">Gene functional relevance<a class="anchor" aria-label="anchor" href="#gene-functional-relevance"></a>
</h2>
<p>The significant cross-talk results can be used for gene functional
relevance analysis via <code><a href="../reference/gene_functional_relevance.html">gene_functional_relevance()</a></code>
function. In this analysis the cross-talk data are used to study the
roles of the genes in the cross-talks. It returns:</p>
<ul>
<li><p><span class="math inline">\(n_f\)</span> the functional
diversity, which is the number of different pathways with which a gene
is involved in a cross-talk;</p></li>
<li><p><span class="math inline">\(n_d\)</span> the interactors
diversity, or the number of interactors, which are the genes with which
a gene interacts in a significant cross-talk;</p></li>
<li><p>the functional relevance, which is <span class="math inline">\(\log_2(\frac{n_f}{n_d})\)</span></p></li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">funct_rel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_functional_relevance.html">gene_functional_relevance</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">pct_f</span>, adj <span class="op">=</span> <span class="va">adj.m</span>, method <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span></span>
<span><span class="va">funct_rel</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
functional_diversity
</th>
<th style="text-align:right;">
interactor_diversity
</th>
<th style="text-align:right;">
functional_relevance
</th>
<th style="text-align:right;">
n_gs_gene
</th>
<th style="text-align:left;">
gs_gene
</th>
<th style="text-align:left;">
functional_gs
</th>
<th style="text-align:left;">
interactors_gene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;width: 5cm; ">
0.3785116
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_E2F_TARGETS
</td>
<td style="text-align:left;">
HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_DNA_REPAIR;HALLMARK_G2M_CHECKPOINT;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
AKT1;NF1;PTEN;ERBB2;TP53;BRCA2;ATRX;KIF4A;TEX15;SYNE1
</td>
</tr>
<tr>
<td style="text-align:left;">
FLNA
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;width: 5cm; ">
1.5849625
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_E2F_TARGETS;HALLMARK_G2M_CHECKPOINT;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_HYPOXIA;HALLMARK_MYOGENESIS;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
AKT1;BRCA2;MYH9
</td>
</tr>
<tr>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;width: 5cm; ">
0.3870231
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MYOGENESIS;HALLMARK_P53_PATHWAY;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
BRCA1;FLNA;NF1;PTEN;CDH1;GATA3;ERBB2;PIK3CA;TP53;PRKDC;LAMA1;RELN;RB1
</td>
</tr>
<tr>
<td style="text-align:left;">
NF1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;width: 5cm; ">
1.4594316
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:left;">
BRCA1;AKT1;PTEN;TP53
</td>
</tr>
<tr>
<td style="text-align:left;">
PTEN
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;width: 5cm; ">
1.0000000
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_G2M_CHECKPOINT;HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:left;">
BRCA1;AKT1;NF1;CDH1;ERBB2;PIK3CA;TP53;BRCA2
</td>
</tr>
<tr>
<td style="text-align:left;">
CDH1
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;width: 5cm; ">
0.8930848
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_TGF_BETA_SIGNALING
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
AKT1;PTEN;ERBB2;PIK3CA;TP53;LAMA1;APC
</td>
</tr>
<tr>
<td style="text-align:left;">
FBN1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;width: 5cm; ">
0.0000000
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_COAGULATION;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
HALLMARK_COMPLEMENT
</td>
<td style="text-align:left;">
F5
</td>
</tr>
<tr>
<td style="text-align:left;">
GATA3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;width: 5cm; ">
1.0000000
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE;HALLMARK_COMPLEMENT
</td>
<td style="text-align:left;">
HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE
</td>
<td style="text-align:left;">
AKT1;RUNX1
</td>
</tr>
<tr>
<td style="text-align:left;">
ERBB2
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;width: 5cm; ">
1.0995357
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS;HALLMARK_UV_RESPONSE_DN
</td>
<td style="text-align:left;">
HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_G2M_CHECKPOINT;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
BRCA1;AKT1;PTEN;CDH1;PIK3CA;TP53;BRCA2
</td>
</tr>
<tr>
<td style="text-align:left;">
F8
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;width: 5cm; ">
0.0000000
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HALLMARK_COAGULATION;HALLMARK_COMPLEMENT
</td>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
LRP1
</td>
</tr>
</tbody>
</table>
</div>
<p>Functional relevance analysis results can be visualized through
<code><a href="../reference/plot_functional_relevance.html">plot_functional_relevance()</a></code> function, which takes as input
the result table from <code><a href="../reference/gene_functional_relevance.html">gene_functional_relevance()</a></code>. As
before, if <code>file_name</code> argument is set to <code>NULL</code>
the function will return the ggplot2 object instead of saving it.
<code>plot_names</code> can be used to turn on/off the plotting of gene
names (logical) or to provide a vector with the new labels of the
vertices, instead of their name. It should be a named vector to assure
correspondence between vertices names and new labels It can be used to
change the vertices names or, as demonstrated here, to plot only some
vertices of interest, like the one that have top/least functional and
interactor values and functional relevance scores.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">funct_rel</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel</span><span class="op">$</span><span class="va">functional_diversity</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span>,</span>
<span>           <span class="va">funct_rel</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel</span><span class="op">$</span><span class="va">interactor_diversity</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span>,</span>
<span>           <span class="va">funct_rel</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel</span><span class="op">$</span><span class="va">functional_relevance</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">names</span><span class="op">)</span></span>
<span><span class="va">funct_rel_names</span> <span class="op">&lt;-</span> <span class="va">funct_rel</span><span class="op">$</span><span class="va">gene</span></span>
<span><span class="va">funct_rel_names</span><span class="op">[</span><span class="op">!</span><span class="va">funct_rel_names</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">names</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_functional_relevance.html">plot_functional_relevance</a></span><span class="op">(</span>fr <span class="op">=</span> <span class="va">funct_rel</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"count"</span>, </span>
<span>                          plot_names <span class="op">=</span> <span class="va">funct_rel_names</span>,</span>
<span>                          pal <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/funct_rel.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>We implemented two ways to perform functional relevance analysis:
<code>method = "count"</code> is used to obtain the results shown above,
that list the numbers of pathways and interactors;
<code>method = "relative"</code> is used to calculate normalized
functional relevance by using a general model. In this case, we have to
provide to the function cross-talk results obtained from a general model
built by using all pathway data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h.ptw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparing_msigdb_list.html">preparing_msigdb_list</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, </span>
<span>                               category <span class="op">=</span> <span class="st">"H"</span>,</span>
<span>                               type <span class="op">=</span> <span class="st">"gene_symbol"</span>, min_size <span class="op">=</span> <span class="fl">1</span>, max_size <span class="op">=</span> <span class="fl">500</span>, </span>
<span>                               gene <span class="op">=</span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">PPI.g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,</span>
<span>                               weights <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">null.pct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">h.ptw</span>, </span>
<span>                          gene_network_adj <span class="op">=</span> <span class="va">adj.m</span>, </span>
<span>                          shared <span class="op">=</span> <span class="cn">F</span>, hash <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                          ct_info <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                          mc_cores_ct <span class="op">=</span> <span class="fl">1</span>, mc_cores_perm <span class="op">=</span> <span class="fl">1</span>, k <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></code></pre></div>
<p>With the code above we have run a cross-talk analysis on the whole
Hallmark database. We set <code>weight</code> argument to
<code>NULL</code> to avoid using any weights, as we have none. In this
case the cross-talk is equal to the number of links. Now we can
calculate the normalized functional relevance</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">funct_rel_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_functional_relevance.html">gene_functional_relevance</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">pct_f</span>, </span>
<span>                                            adj <span class="op">=</span> <span class="va">adj.m</span>, </span>
<span>                                            method <span class="op">=</span> <span class="st">"relative"</span>, </span>
<span>                                            ct_null <span class="op">=</span> <span class="va">null.pct</span><span class="op">)</span></span>
<span></span>
<span><span class="va">funct_rel_norm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
relative_functional_diversity
</th>
<th style="text-align:right;">
relative_interactor_diversity
</th>
<th style="text-align:right;">
relative_functional_relevance
</th>
<th style="text-align:right;">
functional_diversity
</th>
<th style="text-align:right;">
interactor_diversity
</th>
<th style="text-align:right;">
functional_relevance
</th>
<th style="text-align:right;">
n_gs_gene
</th>
<th style="text-align:left;">
gs_gene
</th>
<th style="text-align:left;">
functional_gs
</th>
<th style="text-align:left;">
interactors_gene
</th>
<th style="text-align:right;">
functional_diversity_null
</th>
<th style="text-align:right;">
interactor_diversity_null
</th>
<th style="text-align:left;">
functional_gs_null
</th>
<th style="text-align:left;">
interactors_gene_null
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
AKT1
</td>
<td style="text-align:right;">
0.3469388
</td>
<td style="text-align:right;">
0.0400000
</td>
<td style="text-align:right;width: 5cm; ">
3.1166092
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.3870231
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MYOGENESIS;HALLMARK_P53_PATHWAY;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
BRCA1;FLNA;NF1;PTEN;CDH1;GATA3;ERBB2;PIK3CA;TP53;PRKDC;LAMA1;RELN;RB1
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
325
</td>
<td style="text-align:left;">
HALLMARK_ANGIOGENESIS;HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_G2M_CHECKPOINT;HALLMARK_GLYCOLYSIS;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_ALPHA_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2;HALLMARK_MYOGENESIS;HALLMARK_NOTCH_SIGNALING;HALLMARK_OXIDATIVE_PHOSPHORYLATION;HALLMARK_P53_PATHWAY;HALLMARK_PANCREAS_BETA_CELLS;HALLMARK_PEROXISOME;HALLMARK_PROTEIN_SECRETION;HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE
</td>
<td style="text-align:left;">
CAT;ACLY;ADCY6;ADIPOQ;ANGPT1;CHUK;DECR1;LEP;PDCD4;PFKL;SOD1;STAT5A;YWHAG;LCK;TGFB1;TIMP1;ABI1;BCL3;BRCA1;CCL2;CD28;CD40;CD40LG;CD80;CD86;CDKN2A;FASLG;FLNA;HIF1A;ICAM1;IKBKB;IL10;IL18;IL2;IL2RA;IL2RB;IL2RG;IL6;LYN;MMP9;NOS2;NPM1;STAT1;TLR2;TNF;UBE2D1;GNAI3;CCND1;NKX3-1;FKBP5;NDRG1;VEGFA;SPP1;VAV2;CDH1;GNAI1;GNAI2;PIK3R3;NF1;PTEN;VCAM1;PLCG1;SRC;MMP2;MAPK14;MAPK11;HRAS;INPPL1;IRS1;PIK3CB;SHC1;AKT2;AKT3;YWHAH;IKBKG;ACTB;ITGB4;LAMC2;MSN;NF2;MAPK13;CD274;RAC2;PARVA;TSC1;MDK;GATA3;SLC2A4;HSPB1;GHRL;CREBBP;CD44;JUN;RELA;BCL2L1;CASP3;CASP8;CAV1;CTNNB1;ERBB2;ERBB3;SMAD7;HMOX1;SOD2;PAK1;PDGFRB;PPP2R5B;CYLD;RARA;BCL2L11;CASP9;MCL1;XIAP;HGF;RHOB;BMP2;BAX;BCL2L2;DIABLO;LMNA;CDKN1B;CFLAR;TXNIP;CDKN1A;PEA15;PFKM;AR;RXRA;NEDD4;RXRG;RBP1;ANXA5;FASN;TRIB3;RAC1;GNG12;ITGB3;RAPGEF3;MST1;GNB2;SERPINE1;SIRT2;FN1;FYN;PDGFB;PLEK;HSPA5;GNB4;GNG2;GNGT2;NOTCH4;CBLB;PPP2CB;BRPF3;PIK3CA;SIRT6;GRB2;ZEB1;RAF1;RHOG;PIK3CG;USP8;PIK3R5;TP53;AURKA;MYC;DNMT1;EZH2;PRKDC;CCNE1;LMNB1;CXCL8;FGF2;SNAI2;BDNF;CDH2;IGFBP3;VEGFC;LAMA1;VIM;CXCL12;GJA1;LOX;ADCY9;DLC1;GAB2;DEPTOR;RHOD;NRIP1;NCOR2;FOS;FKBP4;SFN;SLC2A1;KLF4;BCL2;SLC9A3R1;MAPT;JAK1;HSP90AA1;EGF;CCNA2;PML;E2F1;SMAD3;DKC1;CXCR4;IRS2;HK2;PFKP;VLDLR;RRAGD;SHH;TNS1;MKRN1;ABCG2;FOXO3;NR3C1;GATA1;GAPDH;ACKR3;VHL;HK1;PDK1;EFNA1;CTLA4;ITGA6;SELL;SELP;TNFSF11;CSF2;MAP3K8;STAT3;TNFRSF1A;PTPN11;PTPN1;IFNAR1;NFKB1;CSF3;NFKBIA;EDN1;SPHK1;PTGS2;NOS1;TCL1A;PDK2;RELN;IGF2;NGF;GNG11;ADAM17;STRN;CDC42;CTTN;YWHAE;PXN;SOS1;PREX1;ARF6;EZR;RICTOR;CCDC88A;HSP90B1;HSPA4;HSPA9;GSK3B;RPS6;HSP90AB1;EIF4E;PHB2;YWHAQ;PTGES3;IGF1;MAPK12;RB1;GNAO1;MEF2C;HSPB2;TSC2;FOXO4;SOD3;MEF2A;MEF2D;NOTCH1;MYOG;WNT2;ARRB1;CYCS;PDK4;MDM2;SP1;RALGDS;APAF1;TRAF4;ZFP36L1;INS;GCG;FOXO1;FOXA2;ALB;CTBP1;ESR2;MAPK1;LAMTOR5;SIRT1;MTOR;TOPBP1;RHOA;FKBP1A;EGR1;NFE2L2;NR4A1;YWHAZ;KHSRP;EIF4EBP1;ATF4;KIT;PIK3CD;ARRB2;RXRB;SPR;WNT1;SKP2;ESR1;PINK1
</td>
</tr>
<tr>
<td style="text-align:left;">
APC
</td>
<td style="text-align:right;">
0.0465116
</td>
<td style="text-align:right;">
0.0136986
</td>
<td style="text-align:right;width: 5cm; ">
1.7635598
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_ESTROGEN_RESPONSE_LATE
</td>
<td style="text-align:left;">
CDH1
</td>
<td style="text-align:right;">
43
</td>
<td style="text-align:right;">
73
</td>
<td style="text-align:left;">
HALLMARK_G2M_CHECKPOINT;HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_OXIDATIVE_PHOSPHORYLATION;HALLMARK_P53_PATHWAY;HALLMARK_PEROXISOME;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_UP;HALLMARK_MYOGENESIS;HALLMARK_NOTCH_SIGNALING;HALLMARK_PANCREAS_BETA_CELLS;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_SPERMATOGENESIS;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_ALPHA_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
UBC;PSMB10;CCND1;CDH1;CTNNA1;DLG1;JUP;CASP3;CTNNB1;PPP2R5B;LEF1;RAC1;PSMB9;PPP2CB;RBX1;XPO1;CDH2;PSME1;CUL1;TLE3;PSMC4;TLE1;PSMD9;BTRC;FZD5;PSMA3;PSMB8;PSME2;PSMA2;PSMB2;TCF7L1;KMT2D;HNF1A;CDC42;CSNK1D;MAPRE1;PSMB5;PSMC2;PSMD12;PSMD13;PSME3;GSK3B;CACYBP;PSMA4;PSMC6;PSMD14;PSMD1;PSMA1;PSMA6;PSMA7;PSMB3;PSMD3;PSMD7;PSMD8;NOTCH1;WNT2;SKP1;TCF7L2;FZD1;CTBP1;SIAH1;PPP2R1B;TLE4;HDAC1;FZD2;PSMC3;AXIN1;AXIN2;WNT1;TCF7;DVL2;CSNK1E;FRAT1
</td>
</tr>
<tr>
<td style="text-align:left;">
ATRX
</td>
<td style="text-align:right;">
0.4090909
</td>
<td style="text-align:right;">
0.2352941
</td>
<td style="text-align:right;width: 5cm; ">
0.7979562
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1.1699250
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
HALLMARK_G2M_CHECKPOINT;HALLMARK_UV_RESPONSE_DN
</td>
<td style="text-align:left;">
HALLMARK_P53_PATHWAY;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_G2M_CHECKPOINT;HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
BRCA1;TP53;PRKDC;BRCA2
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_OXIDATIVE_PHOSPHORYLATION;HALLMARK_P53_PATHWAY;HALLMARK_PEROXISOME;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_G2M_CHECKPOINT
</td>
<td style="text-align:left;">
IDH1;BRCA1;XRCC5;XRCC6;RAD51;RAD52;TP53;BRCA2;EZH2;SMC3;PRKDC;PNN;XRCC3;ARID4A;WRN;TRIM28;ARID4B
</td>
</tr>
<tr>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:right;">
0.3170732
</td>
<td style="text-align:right;">
0.0675676
</td>
<td style="text-align:right;width: 5cm; ">
2.2304130
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.3785116
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_E2F_TARGETS
</td>
<td style="text-align:left;">
HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_DNA_REPAIR;HALLMARK_G2M_CHECKPOINT;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
AKT1;NF1;PTEN;ERBB2;TP53;BRCA2;ATRX;KIF4A;TEX15;SYNE1
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:left;">
HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_G2M_CHECKPOINT;HALLMARK_GLYCOLYSIS;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2;HALLMARK_NOTCH_SIGNALING;HALLMARK_OXIDATIVE_PHOSPHORYLATION;HALLMARK_P53_PATHWAY;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_PROTEIN_SECRETION;HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
UBC;GADD45A;EGFR;CDKN2A;UBE2D1;UBE2N;AKT1;CCND1;PIAS1;UBE2I;XRCC5;XRCC6;CDK8;GTF2F1;NF1;PTEN;TUBG1;VCAM1;JUN;CASP3;ERBB2;CDK2;CCNA1;TOP2A;AR;KIF2A;RNF4;RAD51;RFC4;RAD52;ERCC1;ERCC4;POLD4;POLH;RFC5;ELL;ERCC2;ERCC3;GTF2H1;GTF2H3;GTF2H5;NELFB;NELFCD;NELFE;POLR2A;POLR2C;POLR2D;POLR2E;POLR2F;POLR2G;POLR2H;POLR2I;POLR2J;POLR2K;SUPT4H1;SUPT5H;TP53;PCNA;POLD1;POLE4;RFC2;RFC3;RPA2;RPA3;SSRP1;POLD3;CDK1;AURKA;HMMR;MYC;KIF2C;KIF4A;BRCA2;CDK4;RPA1;EZH2;MSH2;CHEK1;PLK1;POLD2;POLE;RAD21;SMC1A;SMC3;UBE2S;CCNE1;HUS1;NBN;STAG1;BARD1;RAD51C;RFC1;LMNB1;PGR;RBBP8;XRCC3;CDC6;BUB1;CCNA2;CDC27;CDC45;EXO1;KIF20B;KIF23;PRC1;TPX2;UBE2C;CCNT1;MNAT1;RAD54L;TFDP1;ATRX;ABL1;WRN;FANCC;EWSR1;PAXIP1;BACH1;TCEA1;FOSL2;POU2F1;MSH5;TEX15;STAG3;CBX8;ANLN;CENPJ;MARK4;TUBGCP5;TUBGCP6;SUN2;UBE2D3;ACACA;HNRNPA2B1;POLE3;HNRNPA1;MDM2;CCNK;RAD9A;CTBP1;CDK7;RAD17;TOPBP1;PARP2;DMC1;CDK9;SYNE1;ESR1
</td>
</tr>
<tr>
<td style="text-align:left;">
BRCA2
</td>
<td style="text-align:right;">
0.4444444
</td>
<td style="text-align:right;">
0.1176471
</td>
<td style="text-align:right;width: 5cm; ">
1.9175378
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.5849625
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_E2F_TARGETS;HALLMARK_G2M_CHECKPOINT;HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_DN;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
BRCA1;FLNA;PTEN;ERBB2;TP53;PRKDC;ATRX;TEX15
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2;HALLMARK_P53_PATHWAY;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_SPERMATOGENESIS;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_APICAL_JUNCTION;HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_COAGULATION;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_G2M_CHECKPOINT
</td>
<td style="text-align:left;">
UBC;BRCA1;FLNA;PTEN;ERBB2;CDK2;USP11;RAD51;RFC4;RAD52;ERCC1;POLD4;POLH;RFC5;TP53;PCNA;POLD1;POLE4;RFC2;RFC3;RPA2;RPA3;POLD3;CDK1;AURKA;CDK4;RPA1;MSH2;AURKB;BUB1B;CHEK1;CHEK2;PLK1;POLD2;POLE;PRKDC;HUS1;RAD1;NBN;RAD50;BARD1;RAD51AP1;RAD51C;RFC1;PDS5B;MLH1;PMS2;PSMC3IP;USP1;PGR;RBBP8;XRCC3;EXO1;RAD54L;ATRX;WRN;FANCC;KAT2B;MSH5;TEX15;POLE3;PSMD3;RAD9A;RAD17;TOPBP1;PARP2;DMC1;ESR1
</td>
</tr>
<tr>
<td style="text-align:left;">
CACNA1A
</td>
<td style="text-align:right;">
0.0714286
</td>
<td style="text-align:right;">
0.0526316
</td>
<td style="text-align:right;width: 5cm; ">
0.4405726
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HALLMARK_UV_RESPONSE_DN
</td>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_DN
</td>
<td style="text-align:left;">
RYR1
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MYOGENESIS;HALLMARK_PANCREAS_BETA_CELLS;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_PROTEIN_SECRETION;HALLMARK_TGF_BETA_SIGNALING
</td>
<td style="text-align:left;">
PRKCB;PRKCG;ATXN2;RAC1;GNB2;CALM1;CALM3;KCNK5;CACNA2D2;RYR1;CACNG1;ARHGAP5;GNAO1;CAV3;INS;CACNA1B;GNGT1;CAV2;RHOA
</td>
</tr>
<tr>
<td style="text-align:left;">
CACNA1F
</td>
<td style="text-align:right;">
0.0909091
</td>
<td style="text-align:right;">
0.0833333
</td>
<td style="text-align:right;width: 5cm; ">
0.1255309
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_DN
</td>
<td style="text-align:left;">
HALLMARK_MYOGENESIS
</td>
<td style="text-align:left;">
RYR1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MYOGENESIS;HALLMARK_PEROXISOME;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_APICAL_JUNCTION;HALLMARK_COMPLEMENT;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_GLYCOLYSIS
</td>
<td style="text-align:left;">
GNAI2;CALM1;CALM3;CACNA2D2;CACNA1H;RYR1;CACNG1;PRKG2;CACNA1B;HSPA2;FKBP1A;KCNMA1
</td>
</tr>
<tr>
<td style="text-align:left;">
CDH1
</td>
<td style="text-align:right;">
0.2765957
</td>
<td style="text-align:right;">
0.0510949
</td>
<td style="text-align:right;width: 5cm; ">
2.4365280
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.8930848
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_TGF_BETA_SIGNALING
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_MITOTIC_SPINDLE;HALLMARK_P53_PATHWAY;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_UV_RESPONSE_DN;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_APICAL_JUNCTION;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
AKT1;PTEN;ERBB2;PIK3CA;TP53;LAMA1;APC
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
137
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE;HALLMARK_APOPTOSIS;HALLMARK_BILE_ACID_METABOLISM;HALLMARK_CHOLESTEROL_HOMEOSTASIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_DNA_REPAIR;HALLMARK_E2F_TARGETS;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION;HALLMARK_ESTROGEN_RESPONSE_EARLY;HALLMARK_ESTROGEN_RESPONSE_LATE;HALLMARK_FATTY_ACID_METABOLISM;HALLMARK_G2M_CHECKPOINT;HALLMARK_GLYCOLYSIS;HALLMARK_HEDGEHOG_SIGNALING;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MITOTIC_SPINDLE;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYC_TARGETS_V1;HALLMARK_MYC_TARGETS_V2;HALLMARK_MYOGENESIS;HALLMARK_NOTCH_SIGNALING;HALLMARK_P53_PATHWAY;HALLMARK_PANCREAS_BETA_CELLS;HALLMARK_PEROXISOME;HALLMARK_PI3K_AKT_MTOR_SIGNALING;HALLMARK_PROTEIN_SECRETION;HALLMARK_SPERMATOGENESIS;HALLMARK_TGF_BETA_SIGNALING;HALLMARK_TNFA_SIGNALING_VIA_NFKB;HALLMARK_UNFOLDED_PROTEIN_RESPONSE;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_WNT_BETA_CATENIN_SIGNALING;HALLMARK_XENOBIOTIC_METABOLISM;HALLMARK_ADIPOGENESIS;HALLMARK_ALLOGRAFT_REJECTION;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_ANGIOGENESIS;HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:left;">
PDCD4;TGFB1;EGFR;CDKN2A;IL6;MMP9;AKT1;ACTN1;CCND1;IQGAP2;KRT19;KRT8;VEGFA;FGFR1;S100A4;CTNNA1;CTNND1;DLG1;JUP;PTEN;SRC;MMP2;ACTA1;HRAS;PKD1;EXOC4;STX4;VCL;ITGB4;LAMC2;TJP1;CDH11;CDH15;CDH3;CDH4;CDH6;CDH8;LIMA1;VASP;ZYX;ADAM9;ADAM15;CLDN4;CLDN7;ADAM10;FLOT2;CD44;JUN;CASP3;CAV1;CTNNB1;ERBB2;TIMP2;HGF;PSEN1;KRT18;AR;ANXA5;RAC1;PROC;MMP3;MMP7;F2RL2;HNF4A;SERPINE1;FN1;PLG;FYN;MMP14;PIK3CA;ZEB1;CDH13;CA2;TP53;MYC;DNMT1;EZH2;CXCL8;FGF2;SNAI2;WNT5A;CDH2;LAMA1;FOXC2;VIM;IGF1R;GFRA1;RET;KLF4;PKP3;EGF;SMAD3;NUP98;SOX9;MET;CLDN3;GLI1;ABCG2;GAPDH;PRKCA;VHL;EFNA1;ITGA6;ITGAE;STAT3;PTGS2;ITGB7;GDNF;SLC6A3;SKIL;KLK7;IGF2;WNT7A;CDC42;CTTN;APC;EZR;GSK3B;HDAC2;IGF1;NOTCH1;KIFC3;EPHA2;FOXA2;ALB;CTBP1;SMAD2;MAPK1;RAB5A;MEP1B;SIRT1;RHOA;MTA1;EPCAM;NUMB;NCSTN;ESR1
</td>
</tr>
<tr>
<td style="text-align:left;">
COL12A1
</td>
<td style="text-align:right;">
0.0833333
</td>
<td style="text-align:right;">
0.0500000
</td>
<td style="text-align:right;width: 5cm; ">
0.7369656
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
HALLMARK_MYOGENESIS
</td>
<td style="text-align:left;">
COL6A3
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_MTORC1_SIGNALING;HALLMARK_MYOGENESIS;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_ADIPOGENESIS;HALLMARK_APICAL_JUNCTION;HALLMARK_COMPLEMENT
</td>
<td style="text-align:left;">
COL15A1;COL3A1;COL5A2;COL17A1;COL9A1;MMP12;COL4A2;COL1A1;COL1A2;SERPINH1;COL11A1;COL6A2;COL6A3;COL5A1;PLOD1;PLOD2;P4HA1;P4HA2;COL6A1;COL2A1
</td>
</tr>
<tr>
<td style="text-align:left;">
COL6A3
</td>
<td style="text-align:right;">
0.0454545
</td>
<td style="text-align:right;">
0.0263158
</td>
<td style="text-align:right;width: 5cm; ">
0.7884959
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
HALLMARK_MYOGENESIS
</td>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
COL12A1
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:left;">
HALLMARK_GLYCOLYSIS;HALLMARK_HEME_METABOLISM;HALLMARK_HYPOXIA;HALLMARK_IL2_STAT5_SIGNALING;HALLMARK_IL6_JAK_STAT3_SIGNALING;HALLMARK_INFLAMMATORY_RESPONSE;HALLMARK_INTERFERON_ALPHA_RESPONSE;HALLMARK_INTERFERON_GAMMA_RESPONSE;HALLMARK_KRAS_SIGNALING_DN;HALLMARK_KRAS_SIGNALING_UP;HALLMARK_MTORC1_SIGNALING;HALLMARK_P53_PATHWAY;HALLMARK_UV_RESPONSE_DN;HALLMARK_UV_RESPONSE_UP;HALLMARK_ADIPOGENESIS;HALLMARK_ANDROGEN_RESPONSE;HALLMARK_ANGIOGENESIS;HALLMARK_APICAL_JUNCTION;HALLMARK_APOPTOSIS;HALLMARK_COAGULATION;HALLMARK_COMPLEMENT;HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
COL4A1;COL15A1;ITGA7;ITGAV;LUM;COL3A1;COL5A2;POSTN;COL17A1;COL9A1;ITGB4;ITGA2;COL16A1;CASP8;DCN;ITGB3;PDGFB;COL7A1;COL1A1;COL1A2;SDC1;ITGA5;SERPINH1;COL11A1;COL12A1;COL5A3;COL8A2;COLGALT1;PLOD3;PPIB;COL5A1;PLOD1;PLOD2;P4HA1;P4HA2;COL6A1;ITGA4;COL2A1
</td>
</tr>
</tbody>
</table>
</div>
<p>As you can see, relative functional relevance returns also the
“count” values, as well as the functional and interactor diversity
calculated on the general model and used for normalization. Now we can
use the same <code><a href="../reference/plot_functional_relevance.html">plot_functional_relevance()</a></code> function to plot
relative functional relevance results.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">functional_diversity</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span>,</span>
<span>           <span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">interactor_diversity</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span>,</span>
<span>           <span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">functional_relevance</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">38</span><span class="op">:</span><span class="fl">40</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">names</span><span class="op">)</span></span>
<span><span class="va">funct_rel_names</span> <span class="op">&lt;-</span> <span class="va">funct_rel_norm</span><span class="op">$</span><span class="va">gene</span></span>
<span><span class="va">funct_rel_names</span><span class="op">[</span><span class="op">!</span><span class="va">funct_rel_names</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">names</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_functional_relevance.html">plot_functional_relevance</a></span><span class="op">(</span>fr <span class="op">=</span> <span class="va">funct_rel_norm</span>, </span>
<span>                          method <span class="op">=</span> <span class="st">"relative"</span>, </span>
<span>                          plot_names <span class="op">=</span> <span class="va">funct_rel_names</span>, pal <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/funct_rel_norm.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>As you can see from the plotting function call, the
<code>method</code> argument is used to define which data you want to
plot (“count” or “relative”, as in functional relevance calculation).
This implies that by using the functional relevance results obtained
with method “relative”, that thus returns both relative and count
results, it is possible to plot both results without re-running the
calculation.</p>
</div>
<div class="section level2">
<h2 id="sub-gene-set-cross-talk">Sub-gene-set cross-talk<a class="anchor" aria-label="anchor" href="#sub-gene-set-cross-talk"></a>
</h2>
<p>Gene interacts inside the gene-sets and groups into connected
components (CC). Each connected components correspond to genes with a
similar biological function. It may be important to study the CC of
gene-sets of interest. <code><a href="../reference/gene_set_cc.html">gene_set_cc()</a></code> assigns a score to
each CC in the gene-sets. Similarly to <code>gs_cross_talk</code>,
<code>mc_cores_cc</code> control the parallelization of the calculation
of the CC score. The function returns a list of two elements: the first
are the details of the CC present in each pathway and the second a table
with the result of the sub-pathway analysis</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pct_CC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_set_cc.html">gene_set_cc</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">ptw</span>, </span>
<span>                      gene_network_adj <span class="op">=</span> <span class="va">adj.m</span>, </span>
<span>                      mc_cores_cc <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">pct_CC</span><span class="op">$</span><span class="va">membership</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt;  AKT1 BRCA1  FLNA </span></span>
<span><span class="co">#&gt;     1     1     1 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; CDH1 FBN1 MYH9  NF1 PTEN </span></span>
<span><span class="co">#&gt;    1    2    3    1    1 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; BRCA1 ERBB2 </span></span>
<span><span class="co">#&gt;     1     1</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">pct_CC</span><span class="op">$</span><span class="va">pathway_cc</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
pathway
</th>
<th style="text-align:left;">
ID
</th>
<th style="text-align:right;">
score
</th>
<th style="text-align:right;">
n_gene
</th>
<th style="text-align:right;">
n_link
</th>
<th style="text-align:left;">
gene
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0189125
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0501520
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0129179
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_COAGULATION
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0192503
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_COMPLEMENT
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0238095
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_E2F_TARGETS
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.1212428
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0157042
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_G2M_CHECKPOINT
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0162107
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_G2M_CHECKPOINT
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:right;">
0.0134245
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
HALLMARK_KRAS_SIGNALING_DN
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
0.0371496
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
2
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="tm-ct">TM-CT<a class="anchor" aria-label="anchor" href="#tm-ct"></a>
</h2>
<p>Network can be studied to identify communities, that are groups of
vertices that are more connected between them than to the rest of the
network. It is said that these communities correspond to genes that have
similar biological function. By aligning the biological network to the
target genes we can obtain the gene network, that can be studied to
identify communities. The genes of the different communities can be
grouped in gene-sets, which should represent the part of pathways that
have a similar biological function. We decided to develop
<code><a href="../reference/TM_CT.html">TM_CT()</a></code> as to be able to study how the gene-sets build on
these gene communities interact between communities. First, we need to
subset the biological network to then identify the gene communities. We
can use <code>find_communites()</code> function to compare different
algorithms</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sub.PPI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/subgraph.html" class="external-link">induced.subgraph</a></span><span class="op">(</span><span class="va">PPI.g</span>, vids <span class="op">=</span> <span class="va">target_gene_weights</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="va">target_gene_weights</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">PPI.g</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sub.PPI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/subgraph.html" class="external-link">induced.subgraph</a></span><span class="op">(</span><span class="va">sub.PPI</span>, <span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">sub.PPI</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/degree.html" class="external-link">degree</a></span><span class="op">(</span><span class="va">sub.PPI</span><span class="op">)</span> <span class="op">==</span><span class="fl">0</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">comm_det</span> <span class="op">&lt;-</span> <span class="fu">find_communities</span><span class="op">(</span><span class="va">sub.PPI</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comm_det</span><span class="op">$</span><span class="va">info</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;    algorithm modularity  n</span></span>
<span><span class="co">#&gt; 1 fastgreedy  0.6529376 12</span></span>
<span><span class="co">#&gt; 2    labprop  0.6177907 17</span></span>
<span><span class="co">#&gt; 3   walktrap  0.6468313 18</span></span>
<span><span class="co">#&gt; 4      eigen  0.6170664 14</span></span>
<span><span class="co">#&gt; 5   multilev  0.6687360 11</span></span>
<span><span class="co">#&gt; 6    infomap  0.6531292 18</span></span></code></pre>
<p>Considering the output of <code>find_communities</code>, the best
algorithm to choose is the one with highest modularity and the lowest
number of communities. Here we choose to use multi-level modularity
optimization algorithm.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comm</span> <span class="op">&lt;-</span> <span class="va">comm_det</span><span class="op">$</span><span class="va">comm</span><span class="op">$</span><span class="va">multilev</span></span>
<span><span class="va">comm_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/communities.html" class="external-link">membership</a></span><span class="op">(</span><span class="va">comm</span><span class="op">)</span></span>
<span><span class="va">sub.PPI</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/set_vertex_attr.html" class="external-link">set_vertex_attr</a></span><span class="op">(</span><span class="va">sub.PPI</span>, <span class="st">"comm_id"</span>, value <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">comm_m</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">sub.PPI</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">comm_m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pal</span> <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">alphabet2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">comm_m</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">comm_m</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggraph</span><span class="op">(</span><span class="va">sub.PPI</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">theme_graph</span><span class="op">(</span><span class="op">)</span>  <span class="op">+</span></span>
<span>        <span class="fu">geom_edge_link</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"grey45"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">geom_node_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color <span class="op">=</span> <span class="va">comm_id</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">scale_color_manual</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pal</span><span class="op">)</span>, values <span class="op">=</span> <span class="va">pal</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/BRCA_TCGA_gene_net_communities.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>Now we have all the inputs needed for TM-CT. As before,
<code>mc_cores_ct</code> and <code>mc_cores_tm</code> controls the
parallelization of the calculation of TM-PCT score in pathway pairs and
combinations of communities, respectively. <code><a href="../reference/TM_CT.html">TM_CT()</a></code> function
returns a list of two object, the first is a pathway list for each
community, the second a table with the results of TM-CT calculation</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tm_pct</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TM_CT.html">TM_CT</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">ptw</span>, </span>
<span>                 gene_network_adj <span class="op">=</span> <span class="va">adj.m</span>,</span>
<span>                 membership <span class="op">=</span> <span class="va">comm_m</span>,</span>
<span>                 mc_cores_ct <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                 mc_cores_tm <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">tm_pct</span><span class="op">[[</span><span class="st">"comm_pathway_list"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; $HALLMARK_ALLOGRAFT_REJECTION</span></span>
<span><span class="co">#&gt;      BRCA1 </span></span>
<span><span class="co">#&gt; 0.02532928 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $HALLMARK_APICAL_JUNCTION</span></span>
<span><span class="co">#&gt;       MYH9        NF1 </span></span>
<span><span class="co">#&gt; 0.03444782 0.04457953 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $HALLMARK_APICAL_SURFACE</span></span>
<span><span class="co">#&gt;      BRCA1      PKHD1 </span></span>
<span><span class="co">#&gt; 0.02532928 0.03343465</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tm_pct</span><span class="op">$</span><span class="va">TM_CT_res</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
commID_1
</th>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
commID_2
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:right;">
ct_score
</th>
<th style="text-align:right;">
ngenes_pathway1
</th>
<th style="text-align:right;">
ngenes_pathway2
</th>
<th style="text-align:right;">
nlink
</th>
<th style="text-align:right;">
weight_pathway1
</th>
<th style="text-align:right;">
weight_pathway2
</th>
<th style="text-align:left;">
gene_pathway1
</th>
<th style="text-align:left;">
gene_pathway2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_E2F_TARGETS
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_HEDGEHOG_SIGNALING
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0026197
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0445795
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
NF1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_MITOTIC_SPINDLE
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:right;">
0.0026197
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0445795
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
NF1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_ALLOGRAFT_REJECTION
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_APICAL_JUNCTION
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:right;">
0.0026197
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0445795
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
NF1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_APICAL_SURFACE
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
HALLMARK_APOPTOSIS
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
HALLMARK_PI3K_AKT_MTOR_SIGNALING
</td>
<td style="text-align:right;">
0.0014884
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.0253293
</td>
<td style="text-align:right;">
0.0587639
</td>
<td style="text-align:left;">
BRCA1
</td>
<td style="text-align:left;">
PTEN
</td>
</tr>
</tbody>
</table>
</div>
<p>TM-PCT results can be visualized as a network or as an heatmap again
by using Ulisse functions. We need to slightly adapt the TM-PCT output
to use it with graphical CT functions. In the code below we will modify
the gs names to be compatible with both <code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code>
and <code><a href="../reference/ct_heatmap.html">ct_heatmap()</a></code>. In the code below we will also create a
<code>vertex</code> object to plot only the vertex names involved in the
highest score TM-CT</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tm_pct_mod</span> <span class="op">&lt;-</span> <span class="va">tm_pct</span><span class="op">$</span><span class="va">TM_CT_res</span></span>
<span><span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">gs1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">commID_1</span>, <span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">gs1</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">gs2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">commID_2</span>, <span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">gs2</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="va">tm_pct_mod</span> <span class="op">&lt;-</span> <span class="va">tm_pct_mod</span><span class="op">[</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">tm_ptw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Map</a></span><span class="op">(</span><span class="va">c</span>, <span class="va">tm_pct_mod</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, <span class="va">tm_pct_mod</span><span class="op">[</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tm_ptw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">tm_ptw</span><span class="op">$</span><span class="va">commID_1</span><span class="op">)</span>, <span class="va">tm_ptw</span><span class="op">$</span><span class="va">gs1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tm_n</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tm_ptw</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tm_ptw</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">target_tm_pct</span> <span class="op">&lt;-</span> <span class="va">tm_pct_mod</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">tm_pct_mod</span><span class="op">$</span><span class="va">ct_score</span>, decreasing <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">target_tm_pct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">tm_pct_mod</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tm_n</span><span class="op">[</span><span class="op">!</span><span class="va">tm_n</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">target_tm_pct</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_network_CT.html">plot_network_CT</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">tm_pct_mod</span>, filtering <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                vertex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"gene_comm"</span>, <span class="va">tm_ptw</span><span class="op">)</span>, </span>
<span>                community <span class="op">=</span> <span class="cn">NULL</span>, edge_col_by <span class="op">=</span> <span class="st">"ct_score"</span>, </span>
<span>                vertex_label <span class="op">=</span> <span class="va">tm_n</span>,</span>
<span>                file_out <span class="op">=</span> <span class="st">"tm_pct_net.jpeg"</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/tm_pct_net2.jpeg" width="600" style="display: block; margin: auto;"></p>
<p>Here, we will use TM-CT results to create heatmap annotations to
highlight the communities to which the pathways belongs to.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">tm_pct.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://r.igraph.org/reference/as_adjacency_matrix.html" class="external-link">as_adjacency_matrix</a></span><span class="op">(</span><span class="va">tm_pct_net</span>, attr <span class="op">=</span> <span class="st">"pct"</span>, sparse <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">tm_pct.adj</span> <span class="op">&lt;-</span> <span class="va">tm_pct.adj</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tm_pct.adj</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tm_pct.adj</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">tm_pct.comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tm_pct.adj</span><span class="op">)</span>, <span class="st">"_"</span>, fixed <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, <span class="st">"[["</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tm_pct.comm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tm_pct.adj</span><span class="op">)</span></span>
<span><span class="va">tm_ptw_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>tm_ptw <span class="op">=</span> <span class="va">tm_ptw</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tm_ptw_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">tm_ptw</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/ct_heatmap.html">ct_heatmap</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">tm_pct_mod</span>, </span>
<span>           color_by <span class="op">=</span> <span class="st">"ct_score"</span>, </span>
<span>           community <span class="op">=</span> <span class="cn">NULL</span>, row_annotation <span class="op">=</span> <span class="va">tm_ptw_df</span>, </span>
<span>           column_annotation <span class="op">=</span> <span class="va">tm_ptw_df</span>, </span>
<span>           color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightyellow"</span>, <span class="st">"red"</span><span class="op">)</span>, no_ct_color <span class="op">=</span> <span class="st">"whitesmoke"</span>, </span>
<span>           filtering <span class="op">=</span> <span class="cn">F</span>, label_size <span class="op">=</span> <span class="fl">5</span>, col_name_side <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>           width <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"cm"</span><span class="op">)</span>, height <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"cm"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="bulk_data/DEGcc_ct_f_heatmap.jpeg" width="600" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-colaprico2016" class="csl-entry">
Colaprico, Antonio, Tiago C Silva, Catharina Olsen, Luciano Garofano,
Claudia Cava, Davide Garolini, Thais S Sabedot, et al. 2016.
<span>“TCGAbiolinks: An r/Bioconductor Package for Integrative Analysis
of TCGA Data.”</span> <em>Nucleic Acids Research</em> 44 (8): e71–71.
</div>
<div id="ref-msigdbr2021" class="csl-entry">
Dolgalev, Igor. 2021. <em>Msigdbr: MSigDB Gene Sets for Multiple
Organisms in a Tidy Data Format</em>. <a href="https://CRAN.R-project.org/package=msigdbr" class="external-link">https://CRAN.R-project.org/package=msigdbr</a>.
</div>
<div id="ref-gu2016" class="csl-entry">
Gu, Zuguang, Roland Eils, and Matthias Schlesner. 2016. <span>“Complex
Heatmaps Reveal Patterns and Correlations in Multidimensional Genomic
Data.”</span> <em>Bioinformatics</em> 32 (18): 2847–49.
</div>
<div id="ref-liberzon2015" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi,
Jill P Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures
Database Hallmark Gene Set Collection.”</span> <em>Cell Systems</em> 1
(6): 417–25.
</div>
<div id="ref-mounir2019" class="csl-entry">
Mounir, Mohamed, Marta Lucchetta, Tiago C Silva, Catharina Olsen,
Gianluca Bontempi, Xi Chen, Houtan Noushmehr, Antonio Colaprico, and
Elena Papaleo. 2019. <span>“New Functionalities in the TCGAbiolinks
Package for the Study and Integration of Cancer Data from GDC and
GTEx.”</span> <em>PLoS Computational Biology</em> 15 (3): e1006701.
</div>
<div id="ref-ggraph2022" class="csl-entry">
Pedersen, Thomas Lin. 2022. <em>Ggraph: An Implementation of Grammar of
Graphics for Graphs and Networks</em>.
</div>
<div id="ref-szklarczyk2021" class="csl-entry">
Szklarczyk, Damian, Annika L Gable, Katerina C Nastou, David Lyon,
Rebecca Kirsch, Sampo Pyysalo, Nadezhda T Doncheva, et al. 2021.
<span>“The STRING Database in 2021: Customizable Protein–Protein
Networks, and Functional Characterization of User-Uploaded
Gene/Measurement Sets.”</span> <em>Nucleic Acids Research</em> 49 (D1):
D605–12.
</div>
<div id="ref-ggplot22016" class="csl-entry">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data
Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alice Chiodi, Ettore Mosca.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
