<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Ulisse">
<title>Ulisse workflow 2: Cell-cell cross-talk analysis on single cell data • Ulisse</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Ulisse workflow 2: Cell-cell cross-talk analysis on single cell data">
<meta property="og:description" content="Ulisse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">Ulisse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Ulisse_bulk.html">Ulisse workflow 1: Pathway Cross-Talk analysis of bulk data</a>
    <a class="dropdown-item" href="../articles/Ulisse_sc.html">Ulisse workflow 2: Cell-cell cross-talk analysis on single cell data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Ulisse_sc_files/kePrint-0.0.1/kePrint.js"></script><link href="Ulisse_sc_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Ulisse workflow 2: Cell-cell cross-talk analysis on single cell data</h1>
            
            <h4 data-toc-skip class="date">2023-08-11</h4>
      
      
      <div class="d-none name"><code>Ulisse_sc.Rmd</code></div>
    </div>

    
    
<p>In this tutorial we will see how to analyse intercelluar cross-talk
by using Ulisse. For this analysis we need two inputs: ranked gene sets
of the cell cluster and a communication network.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/OmnipathR/" class="external-link">OmnipathR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Ulisse</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/circlize" class="external-link">circlize</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="preparation-of-the-inputs">Preparation of the inputs<a class="anchor" aria-label="anchor" href="#preparation-of-the-inputs"></a>
</h2>
<p>Firstly we will see how to prepare the inputs by using Ulisse’s
functions. The cell gene sets derive from the cell type or cell clusters
of a single cell sample (or multiple integrated samples), while the
intercellular network has to be provided.Here, we will use Omnipath
communication network <span class="citation">(Türei et al. 2021)</span>,
but other resources can be used. Only take care that Ulisse uses
undirected biological network and does not considers sub-units.</p>
<div class="section level3">
<h3 id="ligand-receptor-network">Ligand-receptor network<a class="anchor" aria-label="anchor" href="#ligand-receptor-network"></a>
</h3>
<p>With the code below we will download Omnipath intercellualr
communications, that are used to build an intercellualr network by using
<code>igraph</code> package functions.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intercell_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html" class="external-link">import_intercell_network</a></span><span class="op">(</span>interactions_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"datasets"</span> <span class="op">=</span> <span class="st">"omnipath"</span><span class="op">)</span>,</span>
<span>                                          transmitter_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>                                            categories <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ligand'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                          receiver_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>                                            categories <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'receptor'</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                          entity_types <span class="op">=</span> <span class="st">"protein"</span><span class="op">)</span></span>
<span><span class="va">g.intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">intercell_net</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source_genesymbol"</span>, <span class="st">"target_genesymbol"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">g.intercell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_edgelist.html" class="external-link">graph_from_edgelist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">g.intercell</span><span class="op">)</span>, </span>
<span>                                   directed <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pre-processing-with-seurat-pipeline">Pre-processing with Seurat pipeline<a class="anchor" aria-label="anchor" href="#pre-processing-with-seurat-pipeline"></a>
</h3>
<p>To explain the functioning of the intercellular cross-talk
calculation implemented in Ulisse we will use one of the samples
sequenced by Wu et al. <span class="citation">(Wu et al. 2021)</span>.
In this study, the authors analyse 26 single cell of breast cancer
samples, that are composed by tumoral and normal cells composing its
microenvironment. The data can be downloaded at <a href="https://singlecell.broadinstitute.org/single_cell/study/SCP1039" class="external-link uri">https://singlecell.broadinstitute.org/single_cell/study/SCP1039</a>
or at GEO website under accession GSE176078. In particular, we will
study intercellular cross-talk in sample CID4515. The code below can be
used to read the 10x data and pre-process them to obtain dimensionality
reduction through Seurat v4 package <span class="citation">(Hao et al.
2021)</span>. We will study the cell-cell communication among the
cell-types provided by the authors, but the pipeline can be applied also
to the clusters.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadMtx.html" class="external-link">ReadMtx</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="st">"CID4515/count_matrix_sparse.mtx"</span>, </span>
<span>                cells <span class="op">=</span> <span class="st">"CID4515/count_matrix_barcodes.tsv"</span>,</span>
<span>                features <span class="op">=</span> <span class="st">"CID4515/count_matrix_genes.tsv"</span> <span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">5</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">data</span>, min.cells <span class="op">=</span> <span class="fl">0</span>, min.features <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><code>metadata.csv</code> is the file with the cell-types
identification provided by the authors (that can be downloaded again at
<a href="https://singlecell.broadinstitute.org/single_cell/study/SCP1039" class="external-link uri">https://singlecell.broadinstitute.org/single_cell/study/SCP1039</a>).
After uploading it, we can use this information to plot the UMAP by cell
type.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"CID44991/metadata.csv"</span>, sep <span class="op">=</span> <span class="st">","</span>, stringsAsFactors <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">meta</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
X
</th>
<th style="text-align:left;">
orig.ident
</th>
<th style="text-align:right;">
nCount_RNA
</th>
<th style="text-align:right;">
nFeature_RNA
</th>
<th style="text-align:right;">
percent.mito
</th>
<th style="text-align:left;">
subtype
</th>
<th style="text-align:left;">
celltype_subset
</th>
<th style="text-align:left;">
celltype_minor
</th>
<th style="text-align:left;">
celltype_major
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
CID44991_AAACGGGTCCGCATAA
</td>
<td style="text-align:left;">
CID44991
</td>
<td style="text-align:right;">
2648
</td>
<td style="text-align:right;">
1153
</td>
<td style="text-align:right;">
1.435045
</td>
<td style="text-align:left;">
TNBC
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial
</td>
</tr>
<tr>
<td style="text-align:left;">
CID44991_AACCATGAGCTACCTA
</td>
<td style="text-align:left;">
CID44991
</td>
<td style="text-align:right;">
4554
</td>
<td style="text-align:right;">
1795
</td>
<td style="text-align:right;">
9.244620
</td>
<td style="text-align:left;">
TNBC
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial
</td>
</tr>
<tr>
<td style="text-align:left;">
CID44991_CAGATCACAAGGTGTG
</td>
<td style="text-align:left;">
CID44991
</td>
<td style="text-align:right;">
2001
</td>
<td style="text-align:right;">
1032
</td>
<td style="text-align:right;">
12.043978
</td>
<td style="text-align:left;">
TNBC
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial
</td>
</tr>
<tr>
<td style="text-align:left;">
CID44991_CATATTCGTGCACTTA
</td>
<td style="text-align:left;">
CID44991
</td>
<td style="text-align:right;">
3293
</td>
<td style="text-align:right;">
1442
</td>
<td style="text-align:right;">
9.626480
</td>
<td style="text-align:left;">
TNBC
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial
</td>
</tr>
<tr>
<td style="text-align:left;">
CID44991_CCTTTCTCACAGTCGC
</td>
<td style="text-align:left;">
CID44991
</td>
<td style="text-align:right;">
4452
</td>
<td style="text-align:right;">
1884
</td>
<td style="text-align:right;">
4.784367
</td>
<td style="text-align:left;">
TNBC
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial ACKR1
</td>
<td style="text-align:left;">
Endothelial
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">meta.data</span>, <span class="va">meta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>, <span class="va">meta</span><span class="op">$</span><span class="va">X</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pal_ct</span> <span class="op">&lt;-</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">turbo</a></span><span class="op">(</span><span class="fl">9</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pal_ct</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"celltype_major"</span>, label <span class="op">=</span> <span class="cn">T</span>,  cols <span class="op">=</span> <span class="va">pal_ct</span><span class="op">)</span></span></code></pre></div>
<p><img src="sc_data/umap_cell_type.jpeg" width="600"></p>
</div>
</div>
<div class="section level2">
<h2 id="intercellular-cross-talk-analysis">Intercellular cross-talk analysis<a class="anchor" aria-label="anchor" href="#intercellular-cross-talk-analysis"></a>
</h2>
<p>Ulisse provides two methods to study intercellular cross-talk, one
that uses expression data and the other that uses Differentially
Expressed Genes (DEG) data. We exemplify the usage of both approaches,
to then compare the obtained results.</p>
<div class="section level3">
<h3 id="expression-data">Expression data<a class="anchor" aria-label="anchor" href="#expression-data"></a>
</h3>
<p>In this case, cell-type marker genes to be sued for cell-type gene
sets are obtained from the normalized count matrix.
<code>preparing_cl_list()</code> takes as an inputs:</p>
<ul>
<li><p>the normalized gene counts matrix (<code>mtx</code>);</p></li>
<li><p>the cell type affiliation of each cell in the count matrix
(<code>clusters</code>);</p></li>
<li><p>the gene of interest (<code>universe</code>);</p></li>
<li><p>two threshold (<code>mean_t</code> and <code>cell_t</code>) to
filter the data and remove noise.</p></li>
</ul>
<p>The function uses <code>mean_t</code> to identify expressed genes in
the normalized expression matrix: if the gene in a cell has a value
equal or higher of that threshold is set to 1, 0 otherwise.
Subsequently, the function uses the binarized expression matrix to
calculate for each gene in each cell-type the frequency of being
expressed in at least <code>cell_t</code> cells. Only the genes in
<code>universe</code> are considered, which are the genes present in the
LR network.</p>
<p>We suggest identifying <code>mean_t</code> considering the mean
expression of each gene calculated on non-zero values. Instead,
<code>cell_t</code> can be set considering the filtering used to remove
low expressed genes at the beginning of the Seurat pipeline. The code
below can be used to assess the expression threshold.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mean_norm</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span></span>
<span><span class="va">mean_norm</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">mean_norm</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="va">mean_g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">mean_norm</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mean_g</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;  0.1906  0.5091  0.6242  0.7346  0.8295  5.4092</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">mean_g</span>, breaks <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fl">0.65</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="sc_data/hist_mean_expr.jpeg" width="600"></p>
<p>Considering the summary of <code>mean_g</code> and the histogram
above, we decide to use 0.65 as <code>mean_t</code> (the red line in the
histogram). We decided this value as it lays past the high frequency
values in the distribution and thus allows to filter out the low and the
most common values, maintaining only the high ones. As cell threshold we
will use the same threshold used for the pre-processing.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">universe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">g.intercell</span><span class="op">)</span><span class="op">$</span><span class="va">name</span></span>
<span><span class="va">ct_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparing_expr_list.html">preparing_expr_list</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span>, </span>
<span>                               clusters <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span>, </span>
<span>                               mean_t <span class="op">=</span> <span class="fl">0.65</span>, </span>
<span>                               cell_t <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                               universe <span class="op">=</span> <span class="va">universe</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ct_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   TNFRSF4  TNFRSF14      ENO1    AGTRAP  TNFRSF1B     EPHA2 </span></span>
<span><span class="co">#&gt; 0.2377049 0.2049180 0.7704918 0.1885246 0.2459016 0.1147541</span></span></code></pre>
<div class="section level4">
<h4 id="communication-analysis">Communication analysis<a class="anchor" aria-label="anchor" href="#communication-analysis"></a>
</h4>
<p>The required inputs are now ready to analyse intercellular cross-talk
with <code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code> function. This function requires also
additional parameter:</p>
<ul>
<li><p><code>k</code> corresponds to the number of permutation needed
that, together with the original matrix, are used for the calculation of
p-value and FDR. In this case we use 49 permutation that, together with
the real one, correspond to a total of 50 permuted list for each cell
type pair, and thus the minimal p-value that can be observed is 1/50 or
0.02;</p></li>
<li><p><code>shared</code> which should be set to <code>TRUE</code> to
enable possible overlapping in gene-lists of a cell-type pair. Different
cells may express the same genes, moreover when considering clusters,
which may represents different states of the same cell-type. These
shared genes may represents intracellular communication, but as both
cells in a cell-pair express the same genes, is impossible to discern if
these are autocrine or paracrine communication. Thus, it is important to
consider also these in intercellular cross-talk;</p></li>
<li><p><code>hash</code>: logical, used to speed-up calculation when
lots of gene-sets are used. Here, we have low cell-type numbers, so we
suggest to set it to <code>FALSE</code>;</p></li>
<li><p><code>ct_info</code>: logical, if detail of gene-gene interaction
in the gene-set should be returned. This might be highly important in
cell-cell cross-talk, so we set it to <code>TRUE</code>. By enabling
this parameter, the output of the function will be a two-element list,
with the fist (<code>ct_info</code>) a table with all the gene
communications between the two cell types; and the second
(<code>ct_res</code>) a table with the intercellular cross-talk results.
Note that the cumulative score of a cell type pair is equal to the sum
of all the gene pair scores in <code>ct _info</code>.</p></li>
<li><p><code>mc_cores_perm</code> and <code>mc_cores_ct</code> control
the parallelization of the function in intercellular cross-talk
calculation and permutation, respectively. Be aware that
<code>mc_cores_perm</code> multiplies <code>mc_cores_ct</code>.
Parallelization is useful to reduce the computational time for the
calculation but improves the amount of memory needed.</p></li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LR.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/as_adjacency_matrix.html" class="external-link">as_adjacency_matrix</a></span><span class="op">(</span><span class="va">g.intercell</span>, sparse <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expr_ccc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">cl_list</span>,</span>
<span>                          gene_network_adj <span class="op">=</span> <span class="va">LR.adj</span>,</span>
<span>                          k <span class="op">=</span> <span class="fl">49</span>, </span>
<span>                          shared <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                          ct_info <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                          hash <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                          mc_cores_perm <span class="op">=</span> <span class="fl">1</span>, mc_cores_ct <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expr_ccc</span><span class="op">$</span><span class="va">communications_info</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
gs1_gene
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:left;">
gs2_gene
</th>
<th style="text-align:right;">
score
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
MIF
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
TNFRSF14
</td>
<td style="text-align:right;">
0.0337512
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
TNFSF12
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
TNFRSF8
</td>
<td style="text-align:right;">
0.0021916
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
BIRC2
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
TNFRSF1B
</td>
<td style="text-align:right;">
0.0059174
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
TNFRSF1A
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
TNFRSF1B
</td>
<td style="text-align:right;">
0.0170948
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
GRN
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
TNFRSF1B
</td>
<td style="text-align:right;">
0.0175331
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_ccc</span><span class="op">$</span><span class="va">cc_communications</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:right;">
ct_score
</th>
<th style="text-align:right;">
ngenes_gs1
</th>
<th style="text-align:right;">
ngenes_gs2
</th>
<th style="text-align:right;">
nlink
</th>
<th style="text-align:right;">
p_value_link
</th>
<th style="text-align:right;">
FDR_link
</th>
<th style="text-align:right;">
p_adj_BH
</th>
<th style="text-align:right;">
weight_gs1
</th>
<th style="text-align:right;">
weight_gs2
</th>
<th style="text-align:left;">
genes_gs1
</th>
<th style="text-align:left;">
genes_gs2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:right;">
125.23126
</td>
<td style="text-align:right;">
399
</td>
<td style="text-align:right;">
443
</td>
<td style="text-align:right;">
1334
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.1266667
</td>
<td style="text-align:right;">
0.0205714
</td>
<td style="text-align:right;">
110.13115
</td>
<td style="text-align:right;">
119.11230
</td>
<td style="text-align:left;">
MIF;TNFSF12;BIRC2;TNFRSF1A;GRN;EFNA1;TP53;NOTCH1;ITGAV;ITGB1;ITGA5;EDN1;FYN;EFNB1;EFNB2;FGF2;MDK;ANGPTL4;STX3;STX4;NID1;LEPR;IL7R;LIFR;IL6ST;IL15RA;PTPN11;IL4R;TYK2;IFNAR2;JAK1;APOD;JAK2;SOCS2;STAT3;TGFB3;TGFB1;IL6;ITGA9;EZR;ICAM1;SLC7A1;SIRPA;LRP5;FZD4;LRP6;MAGED1;LGALS3BP;DLL1;JAG2;DLL4;JAG1;ITGB4;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;ARF6;CD36;TLR4;MDM2;BTNL9;IL18BP;ABL1;MAPK8;LGALS3;CD74;P2RY6;TEK;ITGA6;DAG1;ITGA1;ITGA2;CD44;ITGA3;ADRB2;LDLR;S100A10;SELP;SELL;SEMA6B;PTPRF;TNC;THBS1;LAMA5;SNCA;LMAN1;ACVR1B;IL1RAP;IL33;CXCL12;INHBB;BMP6;BMP4;CXCL2;CXCL10;CXCL11;PAM;ADA;LAMC1;SPP1;LAMA4;LAMB1;MFGE8;CX3CL1;CALR;PDGFB;TNFSF10;BMP1;SEMA3F;SEMA3G;RARRES1;VEGFC;FLT4;PGF;ADM;ALCAM;GDF15;VCAM1;VWF;LAMB2;PRND;DUSP18;CCL2;CCL14;CCL23;RPSA;LAMA3;PLXNA2;NRP2;NRP1;RYK;PTPRK;FZD6;MCAM;NLK;MET;PTPRG;LGALS9;DHH;DIP2A;TLN1;LTBP3;VCL;CASP8;RIPK1;TNFRSF10B;TNFRSF10D;TNFRSF10A;TNFRSF11B;IL1R1;ROBO1;PLXNA1;ROBO4;APP;ADAM23;PDGFA;PDGFD;SDC3;ACKR1;KDR;SDC2;PLCE1;PRKACA;ACKR3;S1PR1;PTGER4;FGFR1;IL2RG;FLT1;LYVE1;VCAN;FAM3C;SLC6A8;BMPR2;CD47;EPHA2;EPHA4;CXCR4;TGFBR3;COPA;APLP2;NOTCH4;CD81;HLA-C;CTSS;CTSL;CTSF;CD63;LGMN;TNFRSF21;CIRBP;POSTN;NFE2L2;ADAM15;CD55;SOS1;PRKCE;SCN9A;TUBA4A;ITPR1;RAF1;PIK3CA;LPP;TNK2;CASP3;PIK3R1;FLOT1;MAPK14;CDK6;RPS6KA3;PTK2;THY1;SLK;PTPRE;TUBA1A;CDK2;CSK;LAT;PLCG2;PLD2;CD79B;PECAM1;PLCG1;BSG;STK11;PLAUR;CSF2RB;IGF2;FURIN;HRH1;ACVR2A;ACVR1;ACVR2B;BAMBI;S100A4;CMTM8;ARF4;AREG;HBEGF;ANXA1;DCN;ANXA2;S100A9;S100A8;SAA1;PLXND1;PLAU;PCNA;CD99;RARA;LAMP1;NCL;PTPRB;PLXNB2;CCRL2;PILRB;CD200;EPHB4;APLNR;TIE1;FGF12;SPARC;FGF18;RXRA;HMGB1;APOE;CD40;CAPN1;TGS1;MFAP2;YBX1;CHUK;IGF2R;INSR;CALCRL;RAMP3;RAMP2;SCARB1;RSPO3;STAB1;TXLNA;MMRN2;TNFRSF1B;HLA-A;IL15;SERPINF1;NPNT;ANGPTL2;JAM3;LUM;LGALS1;VEGFB;CKAP4;FLRT2;POMC;APLN;IKBKB;NSMAF;HLA-E;HLA-B;CKLF;ACTR2;RARRES2;NRAS;MPZL1;IFNGR1;MAP3K4;KRAS;MAPK3;CSF1;LY96;RHBDL2;TFRC;PIP5K1A;SMAP1;PIP4K2A;PIP5K1C;ACVRL1;MAPK1;TGFBR2;UNC5B;MMP2;PLAT;IL13RA1;STAT6;PLXDC2;TNFRSF12A;PLK3;MCL1;RGS16;PTGS2;ODC1;EIF2AK2;CASP10;CASP6;PLK2;NR3C1;CSNK2B;HSPA1B;TAP1;HSP90AB1;ABCB1;HIPK2;NOS3;RRM2B;CTSD;BIRC3;CASP1;ATM;HSPA8;PTEN;SLC38A2;PRKAB1;LATS2;BDKRB2;VRK1;ULK2;PPM1D;BCL2L1;DNMT1;MMP14;ETV5;KMT2E;PTHLH;ZFP91;VANGL1;AVPR2;MCFD2;CD46;GPRC5B;TNFRSF14;LTBR;GPIHBP1;CDH11;CDH5;PTAFR;PRMT5;PTPN1;PRKD2;IFNAR1;ARPC5;TAP2;PDIA3;SCARF1;GAS6;FSTL1;CD109;LSR;ITGA10;GMFB;ICAM2;ICAM3
</td>
<td style="text-align:left;">
TNFRSF14;TNFRSF8;TNFRSF1B;EPHA2;MFAP2;PLA2G2A;ECE1;EPHB2;SDC3;COL16A1;TXLNA;YBX1;PTPRF;PLK3;JAK1;LEPR;NEGR1;TGFBR3;F3;VCAM1;CSF1;WNT2B;NRAS;NGF;VANGL1;NOTCH2;ECM1;MCL1;CTSS;PIP5K1A;S100A9;S100A4;IL6R;ADAM15;MUC1;COPA;MPZL1;ANGPTL1;RGS16;LAMC1;ARPC5;PRG4;PTGS2;PLA2G4A;CD55;CD46;CD34;PLXNA2;WNT9A;NID1;LGALS8;ODC1;SDC1;EIF2AK2;SOS1;PRKCE;MCFD2;ACTR2;IL1R1;IL1RL1;CXCR4;ACVR2A;ACVR1;DPP4;NFE2L2;ITGAV;CASP8;BMPR2;NRP2;TUBA4A;NCL;ACKR3;COL6A3;RAMP1;CHL1;ITPR1;HRH1;RAF1;KAT2B;TGFBR2;ITGA9;RPSA;CCR1;LAMB2;DAG1;SEMA3B;PRKCD;WNT5A;LRIG1;CNTN3;ALCAM;CD47;BOC;FSTL1;ITGB5;PLXNA1;PODXL2;ACKR4;RARRES1;TNFSF10;PIK3CA;BCL6;LPP;IL1RAP;APOD;TNK2;FGFRL1;SPON2;SLIT2;LGI2;PDGFRA;CXCL8;CXCL6;CXCL1;CXCL3;CXCL2;CXCL9;CXCL10;CXCL11;CXCL13;SPP1;CASP6;FGF2;IL15;EDNRA;PDGFC;VEGFC;CASP3;TLR3;SEMA5A;LIFR;PTGER4;FGF10;ITGA1;ITGA2;FST;IL6ST;PLK2;PIK3R1;THBS4;VCAN;EDIL3;RGMB;EFNA5;HSPA4;CXCL14;HBEGF;FGF1;NR3C1;PDGFRB;CD74;SPARC;SLIT3;RIPK1;HLA-A;FLOT1;HLA-C;CSNK2B;HSPA1B;NOTCH4;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;TAP2;TAP1;HLA-DMA;HLA-DPA1;HLA-DPB1;MAPK14;PI16;TREM1;PTK7;VEGFA;HSP90AB1;SMAP1;CD109;FYN;LAMA4;RSPO3;PTPRK;LAMA2;MAP3K5;IFNGR1;EZR;IGF2R;MAP3K4;THBS2;DLL1;PDGFA;ITGB8;IL6;INHBA;EGFR;CD36;SEMA3C;HGF;SEMA3D;CDK6;PILRB;KMT2E;LAMB1;WNT2;FAM3C;PLXNA4;PTN;HIPK2;RARRES2;CD99;MXRA5;RPS6KA3;AR;EFNB1;IL13RA1;APLN;GPC4;CD99L2;PLXNA3;ANGPT2;BMP1;TNFRSF10B;SCARA5;FGFR1;PLAT;IKBKB;TGS1;NSMAF;LY96;IL7;SDC2;RRM2B;ANGPT1;TNFRSF11B;PTK2;JAK2;IL33;IL11RA;CCL19;TLN1;GNAQ;NTRK2;CTSL;TGFBR1;TNC;ANGPTL2;ABL1;RXRA;NOTCH1;CTSD;IGF2;CD81;ADM;SPON1;SAA1;LGR4;CD59;CD44;MDK;STX3;VEGFB;CAPN1;CD248;CTSF;CLCF1;GAL;P2RY6;BIRC3;BIRC2;CASP1;ATM;THY1;HSPA8;CHEK1;CDON;APLP2;JAM3;IL15RA;PLXDC2;PIP4K2A;BAMBI;ITGB1;NRP1;FZD8;CXCL12;MAPK8;DKK1;UNC5B;PLAU;VCL;BMPR1A;PTEN;FAS;CHUK;SLK;ADRA2A;PTPRE;TNFRSF1A;LAG3;CLEC2D;KLRD1;LRP6;KRAS;PTHLH;SLC38A2;VDR;TUBA1A;ACVRL1;ACVR1B;ITGA5;CD63;GDF11;STAT6;LRP1;MDM2;LUM;DCN;SOCS2;IGF1;CMKLR1;PTPN11;PRKAB1;SCARB1;ULK1;LATS2;SLC7A1;HMGB1;EDNRB;EFNB2;TNFSF13B;LAMP1;ANG;MMP14;PRMT5;ARF6;BMP4;GMFB;LGALS3;PGF;TGFB3;FLRT2;LGMN;BDKRB2;VRK1;HSP90AA1;GREM1;THBS1;PDIA3;FGF7;ANXA2;SEMA7A;CSK;MFGE8;SEMA4B;FURIN;RGMA;IGF1R;TNFRSF12A;VASN;IL4R;MAPK3;STX4;MMP2;CX3CL1;CDH11;CKLF;PLCG2;SERPINF1;PLD2;TNFSF12;CD68;TP53;ULK2;LGALS9;UNC119;CCL2;CCL7;CCL11;CCL8;CCL5;CCL4;ERBB2;RARA;STAT3;CCR10;RAMP2;GRN;MAP3K14;PPM1D;MRC2;PRKCA;LGALS3BP;C1QTNF1;COLEC12;LAMA1;LMAN1;SIRPA;PCNA;JAG1;SRC;PLCG1;PI3;MMP9;PTPN1;AURKA;BSG;STK11;PIP5K1C;CD70;TNFSF14;INSR;ANGPTL4;DNMT1;ICAM1;TYK2;LDLR;CALR;PRKACA;ADGRE5;NOTCH3;GDF15;COMP;FXYD1;MIA;AXL;TGFB1;PLAUR;APOE;PRKD2;CLEC11A;IL11;MAPK1;MIF;CHEK2;LIF;LGALS1;EP300;FBLN1;APP;IFNAR2;IFNAR1;ITGB2;DIP2A
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
PVL
</td>
<td style="text-align:right;">
77.68139
</td>
<td style="text-align:right;">
368
</td>
<td style="text-align:right;">
331
</td>
<td style="text-align:right;">
1056
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.2866667
</td>
<td style="text-align:right;">
0.0205714
</td>
<td style="text-align:right;">
101.29508
</td>
<td style="text-align:right;">
78.27642
</td>
<td style="text-align:left;">
MIF;EDN1;ITGB1;STX3;STX4;EFNB2;NOTCH1;NID1;FYN;GFRA1;TP53;LEPR;IL7R;LIFR;IL6ST;IL15RA;PTPN11;IL4R;TYK2;IFNAR2;TGFB3;TGFB1;IL6;ITGA9;EZR;ICAM1;SLC7A1;SIRPA;MAGED1;LGALS3BP;DLL1;MDK;JAG2;DLL4;JAG1;ITGB4;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;ARF6;MDM2;JAK1;BTNL9;JAK2;IL18BP;STAT3;EPHA2;EPHA4;ABL1;MAPK8;LGALS3;CD74;P2RY6;SEMA3G;TEK;ITGA6;ITGAV;DAG1;ITGA1;ITGA2;CD44;ITGA3;ADRB2;LDLR;PTPRF;LMAN1;ACVR1B;IL1RAP;INHBB;BMP6;BMP4;VCAM1;SPP1;VCAN;CXCL12;VWF;THBS1;LAMC1;FGF2;LAMA4;LAMB1;TNC;MFGE8;CX3CL1;CALR;PDGFB;ADM;BMP1;SEMA3F;RARRES1;VEGFC;FLT4;PGF;GDF15;LAMB2;PRND;DUSP18;RPSA;LAMA3;LAMA5;MET;NRP1;LGALS9;DIP2A;TLN1;LTBP3;VCL;SELL;NRP2;CASP8;RIPK1;TNFRSF10B;TNFRSF10D;TNFRSF10A;TNFRSF11B;IL1R1;IL33;ITGA5;ACKR1;TLR4;S1PR1;PTGER4;PLXNA2;PLXNA1;FAM3C;SLC6A8;KDR;SELP;TGFBR3;FGFR1;IL17RA;PDGFA;PDGFD;COPA;ROBO1;ROBO4;TNFSF10;TNFRSF1A;APLP2;NOTCH4;CD81;CTSS;CTSL;CTSF;CD63;LGMN;POSTN;NFE2L2;FLT1;SDC3;ADAM15;CD55;SOS1;PRKCE;SCN9A;TUBA4A;ITPR1;RAF1;PIK3CA;LPP;TNK2;CASP3;PIK3R1;FLOT1;MAPK14;CD36;CDK6;RPS6KA3;PTK2;MCAM;THY1;SLK;PTPRE;TUBA1A;CDK2;CSK;LAT;PLCG2;PLD2;CD79B;PECAM1;PLCG1;BSG;STK11;PRKACA;PLAUR;CSF2RB;APP;IGF2;CD47;FURIN;HRH1;ACVR2A;ACVR1;ACVR2B;BAMBI;S100A9;S100A8;SAA1;SDC2;PLAU;PCNA;CD99;RARA;LAMP1;NCL;PTPRB;PLXNB2;CCRL2;PILRB;CD200;EPHB4;TIE1;BMPR2;FGF12;SPARC;FGF18;ANXA2;RXRA;HMGB1;ANGPTL4;APOE;POMC;PTHLH;CAPN1;TGS1;MFAP2;YBX1;CHUK;HLA-C;CALCRL;ACKR3;RAMP3;RAMP2;SCARB1;STAB1;HBEGF;APLN;MMRN2;TNFRSF1B;HLA-A;IL15;NPNT;ANGPTL2;JAM3;LUM;LGALS1;ALCAM;VEGFB;CXCR4;FLRT2;CXCL2;CXCL10;CXCL11;ANXA1;CCL23;IKBKB;NSMAF;GRN;RSPO3;CKLF;ACTR2;DCN;S100A4;INSR;NRAS;MPZL1;IFNGR1;MAP3K4;KRAS;MAPK3;LY96;TFRC;CD40;CCL2;PIP5K1A;SMAP1;PIP4K2A;PIP5K1C;PTPRK;TGFBR2;ACVRL1;LRP5;MMP2;PLAT;EFEMP2;SOCS2;TNFSF12;IL2RG;IL13RA1;STAT6;SNCA;TXLNA;LRP6;PLXDC2;TNFRSF12A;PLK3;MCL1;RGS16;PTGS2;ODC1;EIF2AK2;CASP10;CASP6;PLK2;NR3C1;CSNK2B;HSPA1B;TAP1;HSP90AB1;ABCB1;HIPK2;NOS3;RRM2B;CTSD;BIRC3;CASP1;ATM;HSPA8;PTEN;SLC38A2;PRKAB1;LATS2;BDKRB2;VRK1;ULK2;PPM1D;BCL2L1;DNMT1;MMP14;ETV5;AREG;EFNB1;KMT2E;LGALS8;VANGL1;AVPR2;MCFD2;CD46;BCAM;GPIHBP1;CDH11;CDH5;ARPC5;INPP5D;KITLG;NPTN;PTPN1;TAP2;PDIA3;SCARF1;ADAM23;PAM;GAS6;FSTL1;CD109;LSR;ITGA10;GMFB;TNFRSF14;EFNA1
</td>
<td style="text-align:left;">
TNFRSF14;ECE1;COL16A1;TXLNA;RHBDL2;YBX1;PTPRF;ARTN;PLK3;JAK1;TGFBR3;F3;VCAM1;CSF1;NRAS;NGF;VANGL1;NOTCH2;ECM1;MCL1;CTSS;PIP5K1A;S100A4;IL6R;EFNA1;MUC1;COPA;FCGR2A;MPZL1;ANGPTL1;RGS16;LAMC1;ARPC5;CD55;CD46;LAMB3;NID1;LGALS8;ODC1;EIF2AK2;SOS1;MCFD2;ACTR2;IL1R1;ACVR2A;ACVR1;NFE2L2;ITGA4;ITGAV;CALCRL;BMPR2;NRP2;TUBA4A;NCL;COL6A3;RAMP1;ITPR1;RAF1;TGFBR2;RPSA;LAMB2;DAG1;LRIG1;ALCAM;CD47;FSTL1;ITGB5;PLXNA1;PODXL2;RARRES1;TNFSF10;PIK3CA;BCL6;LPP;IL1RAP;SPON2;CXCL3;CXCL2;CXCL9;CXCL10;SPP1;NPNT;CASP6;EDNRA;CASP3;SEMA5A;LIFR;ITGA1;IL6ST;PLK2;PIK3R1;VCAN;EDIL3;RGMB;HSPA4;HBEGF;FGF1;NR3C1;IL17B;PDGFRB;CD74;SPARC;SLIT3;RIPK1;HLA-A;FLOT1;HLA-C;CSNK2B;HSPA1B;HLA-DRA;HLA-DQA1;HLA-DQB1;TAP2;TAP1;HLA-DPA1;HLA-DPB1;MAPK14;PTK7;VEGFA;HSP90AB1;SMAP1;CD109;FYN;LAMA4;PTPRK;MAP3K5;IFNGR1;ESR1;EZR;IGF2R;MAP3K4;THBS2;PDGFA;IL6;INHBA;CD36;HGF;CDK6;PILRB;KMT2E;LAMB1;FAM3C;PTN;HIPK2;RARRES2;CD99;MXRA5;RPS6KA3;AR;EFNB1;IL13RA1;CD99L2;ANGPT2;BMP1;FGFR1;PLAT;IKBKB;TGS1;LY96;SDC2;RRM2B;ANGPT1;TNFRSF11B;PTK2;GPR20;VLDLR;JAK2;CCL19;TLN1;GNAQ;NTRK2;CTSL;TGFBR1;TNC;ANGPTL2;ABL1;RXRA;NOTCH1;CTSD;CD81;ADM;SAA1;CD59;CD44;MDK;APLNR;CAPN1;CD248;CTSF;TRPC6;BIRC3;BIRC2;CASP1;ATM;MCAM;THY1;HSPA8;APLP2;JAM3;IL15RA;ITGA8;PIP4K2A;BAMBI;ITGB1;NRP1;CXCL12;MAPK8;UNC5B;PLAU;VCL;BMPR1A;PTEN;FAS;CHUK;SLK;ADRA2A;PTPRE;TNFRSF1A;LRP6;KRAS;SLC38A2;TUBA1A;ACVRL1;ACVR1B;ITGA5;ITGA7;CD63;GDF11;STAT6;LRP1;AVPR1A;MDM2;LUM;DCN;SOCS2;PTPN11;ULK1;LATS2;HMGB1;EDNRB;TNFSF13B;LAMP1;MMP14;PRMT5;ARF6;LGALS3;PGF;TGFB3;LGMN;VRK1;HSP90AA1;THBS1;PDIA3;SEMA6D;FGF7;ANXA2;CSK;LINGO1;MFGE8;SEMA4B;FURIN;RGMA;IGF1R;TNFRSF12A;VASN;IL4R;MAPK3;STX4;MMP2;CDH11;CKLF;SERPINF1;TNFSF12;TP53;ULK2;LGALS9;CCL2;CCL8;ERBB2;RARA;STAT3;CCR10;GRN;ITGA3;PPM1D;MRC2;BIRC5;LGALS3BP;C1QTNF1;LAMA3;LMAN1;BCL2;PCNA;JAG1;BCL2L1;MMP9;PTPN1;LAMA5;BSG;STK11;INSR;ANGPTL4;DNMT1;ICAM1;LDLR;EPOR;CALR;PRKACA;ADGRE5;NOTCH3;LGI4;FXYD1;LSR;AXL;TGFB1;PLAUR;BCAM;APOE;CLEC11A;MAPK1;MIF;DUSP18;LGALS1;EP300;FBLN1;APP;IFNAR2;IFNAR1;DIP2A;EPHA3;SEMA5B
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
B-cells
</td>
<td style="text-align:right;">
41.40780
</td>
<td style="text-align:right;">
364
</td>
<td style="text-align:right;">
323
</td>
<td style="text-align:right;">
1024
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.2940000
</td>
<td style="text-align:right;">
0.0205714
</td>
<td style="text-align:right;">
101.59016
</td>
<td style="text-align:right;">
41.86842
</td>
<td style="text-align:left;">
MIF;BIRC2;TNFRSF1A;GRN;ITGAV;ITGB1;ITGA5;EDN1;TP53;TYK2;STX3;STX4;FYN;NOTCH1;NID1;LEPR;IL7R;LIFR;IL6ST;IL15RA;PTPN11;IL4R;IFNAR2;SPP1;PDGFB;HLA-C;HLA-DRA;HLA-DRB1;DLL1;MDK;JAG2;DLL4;JAG1;HLA-DRB5;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;ARF6;CD36;TLR4;MDM2;EPHA2;EPHA4;FLT4;EFNB2;CD74;P2RY6;INPP5D;CD34;VCAN;ITGA6;DAG1;ITGA1;ITGA2;CD44;ITGA3;ITGB4;ADRB2;LDLR;LGALS9;LGALS1;IL10RB;RAMP3;RAMP2;SNCA;LMAN1;ACVR1B;CXCL12;TGFB1;FGF12;LAMC1;LAMB2;LAMA4;LAMB1;THBS1;LAMA3;LAMA5;DUSP18;VCAM1;VWF;FGF2;TNC;MFGE8;CX3CL1;CALR;TNFSF10;BMP6;BMP1;BMP4;SEMA3F;SEMA3G;RARRES1;VEGFC;PGF;LRP5;LRP6;TBXA2R;TGFB3;GDF15;PRND;CCL2;CCL14;CCL23;CYTL1;NRP1;DIP2A;NRP2;CASP8;RIPK1;TNFRSF10B;TNFRSF10D;TNFRSF10A;TNFRSF11B;IL1R1;IL33;PECAM1;ACKR1;S1PR1;ITGA9;PTGER4;IL2RG;JAK1;PIK3R1;IL7;IL6;JAK2;STAT3;KDR;TEK;ARPC5;POMC;ACTR2;ADM;VEGFB;HSPA8;PTHLH;COPA;FGFR1;APLP2;RGS2;TM4SF1;C6orf15;CDK2;PLCG1;NOTCH4;CD81;TNFRSF14;TNFRSF1B;LTBR;CSF1;SEMA4C;NFE2L2;TNFRSF21;CHUK;TBK1;HMOX1;CD40;BAG6;CTSS;CTSL;CTSF;CD63;LGMN;FLT1;SIRPA;SDC3;PTPRF;ADAM15;CD55;PLXNA2;SOS1;PRKCE;SCN9A;TUBA4A;ITPR1;RAF1;PIK3CA;LPP;TNK2;CASP3;FLOT1;MAPK14;CDK6;RPS6KA3;PTK2;ABL1;MCAM;THY1;SLK;PTPRE;TUBA1A;CSK;LAT;PLCG2;PLD2;CD79B;BSG;STK11;PRKACA;PLAUR;CSF2RB;APP;LGALS3;ICAM1;IGF2;HRH1;PCNA;CD99;BTN3A2;RARA;LAMP1;EFNA1;EFNB1;PILRB;IL15;CXCL10;CXCL11;TIE1;SPARC;FGF18;RXRA;HMGB1;TLN1;CAPN1;MFAP2;YBX1;FURIN;SCARB1;STAB1;HBEGF;NCL;SDC2;PTPRB;TXLNA;VCL;HLA-A;INHBB;NPNT;ANGPTL2;JAM3;LUM;LGALS3BP;LYN;STAT6;INSR;BCR;IFNAR1;FAM3C;S100A8;MYL9;ACVR2A;BMPR2;ACVR2B;S100A4;MET;IGF2R;NRAS;MPZL1;IFNGR1;MAP3K4;KRAS;MAPK3;LY96;RHBDL2;EPHB4;TFRC;PLXNB2;PIP5K1A;SMAP1;PIP4K2A;PIP5K1C;MAPK1;PTPRK;ANXA2;CD47;MMP2;PLAT;ROBO4;PDGFA;DCN;SOCS2;TNFSF12;IL13RA1;ICAM2;ICAM3;PLXDC2;TNFRSF12A;S100A9;PLK3;MCL1;RGS16;PTGS2;ODC1;EIF2AK2;CASP10;CASP6;PLK2;NR3C1;CSNK2B;HSPA1B;TAP1;HSP90AB1;ABCB1;HIPK2;NOS3;RRM2B;CTSD;BIRC3;CASP1;ATM;PTEN;SLC38A2;PRKAB1;LATS2;BDKRB2;VRK1;ULK2;PPM1D;BCL2L1;DNMT1;CCRL2;SLC7A1;KMT2E;ZFP91;VANGL1;MCFD2;AREG;BCAM;GPRC5B;GPIHBP1;CDH11;CDH5;EZR;PTAFR;PRMT5;PTPN1;PRKD2;APOE;KITLG;NPTN;TAP2;PDIA3;SCARF1;MICA;ULBP2;TGFBR3;CXCR4;TGFBR2;FSTL1;CD109;ACVRL1;PLAU;LSR;HLA-B;HLA-F;GMFB;ACKR3;RPSA;MAPK8;CSF3
</td>
<td style="text-align:left;">
TNFRSF14;TNFRSF1B;PLA2G2A;ECE1;RPS6KA1;PTAFR;TXLNA;LCK;YBX1;PTPRF;PLK3;JAK1;S1PR1;CD53;NRAS;NOTCH2;MCL1;CTSS;PIP5K1A;S100A9;S100A4;EFNA4;FCRL1;COPA;SLAMF1;FCGR2B;MPZL1;SELL;RGS16;LAMC1;ARPC5;PTPRC;IL24;CD55;CR2;CD46;LAMB3;LGALS8;ODC1;POMC;EIF2AK2;SOS1;PRKCE;MCFD2;ACTR2;CXCR4;SCN3A;ITGA6;NFE2L2;ITGA4;ITGAV;CASP10;CASP8;BMPR2;NRP2;WNT10A;TUBA4A;NCL;ITPR1;GHRL;RAF1;KAT2B;TGFBR2;RPSA;CCR1;CCR2;DAG1;PRKCD;ALCAM;CD47;FSTL1;RARRES1;TNFSF10;PIK3CA;BCL6;LPP;IL1RAP;TNK2;SPON2;CD38;TLR1;TLR6;CXCL3;CXCL10;SPP1;IL15;CASP3;IL7R;PTGER4;IL6ST;PLK2;PIK3R1;RGMB;HSPA4;HBEGF;NR3C1;ADRB2;CD74;SPARC;RIPK1;HLA-A;FLOT1;DDR1;HLA-C;LTA;TNF;LTB;NCR3;CSNK2B;HSPA1B;NOTCH4;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DOB;TAP2;TAP1;HLA-DMB;HLA-DMA;HLA-DOA;HLA-DPA1;HLA-DPB1;MAPK14;VEGFA;HSP90AB1;SMAP1;FYN;LAMA4;PTPRK;MAP3K5;IFNGR1;EZR;IGF2R;MAP3K4;CCR6;LFNG;IL6;ABCB1;CDK6;AZGP1;PILRB;PILRA;KMT2E;PIK3CG;LAMB1;FAM3C;HIPK2;EPHB6;CDK5;CD99;RPS6KA3;IL2RG;CXCR3;IL13RA1;PLXNA3;ANGPT2;TNFRSF10B;TNFRSF10A;PTK2B;FGFR1;IKBKB;TGS1;NSMAF;LY96;IL7;RRM2B;TNFRSF11B;PTK2;JAK2;TLN1;CTSL;SYK;TGFBR1;NOTCH1;CTSD;CD81;SPON1;SAA1;CD59;CD44;MDK;PTPRJ;STX3;MS4A1;VEGFB;CAPN1;CLCF1;BIRC3;BIRC2;CASP1;ATM;IL10RA;HSPA8;APLP2;JAM3;IL15RA;IL2RA;PIP4K2A;BAMBI;ITGB1;MAPK8;VCL;PTEN;FAS;CHUK;SLK;PTPRE;PTPN6;CLEC2D;CD69;KRAS;SLC38A2;TUBA1A;ITGB7;CD63;GDF11;STAT6;MDM2;LUM;DCN;IGF1;PTPN11;PRKAB1;ULK1;FGF9;SLC7A1;HMGB1;EFNB2;TNFSF13B;LAMP1;ANG;PRMT5;RIPK3;ARF6;GMFB;LGALS3;LGMN;VRK1;HSP90AA1;THBS1;PDIA3;ANXA2;SEMA7A;CSK;IL16;MFGE8;SEMA4B;FURIN;IGF1R;TNFRSF12A;IL4R;CD19;LAT;MAPK3;ITGAL;STX4;CKLF;PLCG2;SERPINF1;TNFSF12;CD68;TP53;LGALS9;UNC119;CCL5;CCL4;RARA;STAT3;GRN;MAP3K14;PPM1D;CD79B;PECAM1;BIRC5;LGALS3BP;LMAN1;BCL2;PCNA;HCK;PI3;MMP9;PTPN1;LAMA5;BSG;STK11;CD70;FCER2;ANGPTL4;DNMT1;ICAM1;TYK2;LDLR;EPOR;CALR;PRKACA;ADGRE5;JAK3;LSR;HCST;MIA;TGFB1;CD79A;PLAUR;APOE;PRKD2;DBP;CD37;SIGLEC10;LILRB1;MAPK1;MIF;DUSP18;CSF2RB;LGALS1;EP300;APP;IFNAR2;IFNAR1;ITGB2;DIP2A
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
Plasmablasts
</td>
<td style="text-align:right;">
40.09062
</td>
<td style="text-align:right;">
162
</td>
<td style="text-align:right;">
123
</td>
<td style="text-align:right;">
346
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.8230303
</td>
<td style="text-align:right;">
0.0205714
</td>
<td style="text-align:right;">
52.92623
</td>
<td style="text-align:right;">
44.75000
</td>
<td style="text-align:left;">
MIF;NOTCH1;TP53;LEPR;IL7R;LIFR;IL6ST;IL15RA;PTPN11;IL4R;TYK2;IFNAR2;HLA-C;HLA-DRA;HLA-DRB1;HLA-DRB5;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;MDM2;JAK1;BTNL9;IL6;JAK2;IL18BP;STAT3;CD74;P2RY6;CD34;VCAN;ADRB2;LDLR;FYN;LGALS9;LGALS1;JAG1;FGF2;TNC;MDK;THBS1;LAMA5;ANGPTL4;LMAN1;ACVR1B;CXCL12;TGFB1;LAMC1;LAMB2;LAMA4;LAMB1;LAMA3;DUSP18;VCAM1;SPP1;VWF;TNFSF10;TNFRSF1A;LRP5;LRP6;PRND;NRP1;NRP2;CASP8;RIPK1;TNFRSF10B;TNFRSF10D;TNFRSF10A;TNFRSF11B;PECAM1;S1PR1;ITGAV;ITGA9;PTGER4;CD44;ITGB1;ITGA5;KDR;TEK;TLR4;ARPC5;POMC;ACTR2;ADM;VEGFB;HSPA8;PTHLH;COPA;APLP2;NOTCH4;CD81;CTSS;CTSL;CTSF;CD63;LGMN;CALR;ICAM1;PCNA;RARA;LAMP1;PILRB;IL15;IL7;RXRA;HMGB1;PTK2;CAPN1;ITGB4;CD99;CD36;SCARB1;STAB1;HBEGF;HLA-A;NID1;NPNT;VEGFC;FLT4;TLN1;ANGPTL2;JAM3;LUM;TGFB3;LGALS3BP;FAM3C;S100A4;IGF2R;INSR;LY96;PIP5K1A;SMAP1;PIP4K2A;PIP5K1C;PTPRK;MCAM;ANXA2;PLAT;ROBO4;LGALS3;CASP3;TNFSF12;TNFRSF1B;EFNB2;CSK;MCFD2;GPRC5B;TAP2;TAP1;PDIA3;SCARF1;ITGA3;CD55;SDC2;LSR;APP;TNFRSF14;CXCR4;ACKR3
</td>
<td style="text-align:left;">
TNFRSF14;YBX1;PLK3;JAK1;CD53;NRAS;MCL1;CTSS;S100A4;IL6R;COPA;MPZL1;SELL;ARPC5;PTPRC;CD55;CD46;ODC1;SDC1;EIF2AK2;SOS1;MCFD2;ACTR2;CXCR4;ITGA6;NFE2L2;ITGA4;CASP10;CASP8;WNT10A;TUBA4A;NCL;RPSA;ALCAM;CD47;RARRES1;TNFSF10;CD38;SPP1;CASP3;IL6ST;PIK3R1;HSPA4;NR3C1;ADRB2;CD74;HLA-A;FLOT1;HLA-C;CSNK2B;HSPA1B;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;HLA-DQA2;TAP1;HLA-DMA;HLA-DPA1;HLA-DPB1;HSP90AB1;EZR;CDK6;KMT2E;PIK3CG;FAM3C;HIPK2;CD99;IL2RG;TGS1;LY96;TLN1;SYK;CTSD;CD81;SAA1;CD59;CD44;BIRC3;CASP1;ATM;IL10RA;HSPA8;APLP2;IL15RA;ITGB1;CLEC2D;KRAS;SLC38A2;CD63;MDM2;IGF1;HMGB1;ARF6;LGALS3;HSP90AA1;PDIA3;ANXA2;IL16;MFGE8;TNFRSF12A;CKLF;PLCG2;STAT3;GRN;CD79B;PECAM1;LMAN1;BCL2L1;PTPN1;BSG;CD70;CALR;ADGRE5;CD79A;APOE;PRKD2;CD37;MIF;LGALS1;IFNAR2;IFNAR1
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
T-cells
</td>
<td style="text-align:right;">
52.56231
</td>
<td style="text-align:right;">
375
</td>
<td style="text-align:right;">
374
</td>
<td style="text-align:right;">
1146
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.2360000
</td>
<td style="text-align:right;">
0.0205714
</td>
<td style="text-align:right;">
105.78689
</td>
<td style="text-align:right;">
49.82816
</td>
<td style="text-align:left;">
MIF;TNFSF12;BIRC2;TNFRSF1A;GRN;ITGAV;ITGB1;ITGA5;EDN1;TP53;FYN;STX3;STX4;NOTCH1;APOE;LEPR;IL7R;LIFR;IL6ST;IL15RA;PTPN11;IL4R;TYK2;IFNAR2;TGFB3;TGFB1;ITGA9;EZR;ICAM1;SPP1;PDGFB;SLC7A1;SIRPA;HLA-C;HLA-DRA;HLA-DRB1;LGALS3BP;CD58;CD59;DLL1;MDK;JAG2;DLL4;JAG1;TNFRSF14;HLA-DRB5;HLA-DQA1;HLA-DQB1;HLA-DQA2;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;ARF6;CD36;TLR4;MDM2;JAK1;BTNL9;IL6;JAK2;IL18BP;STAT3;EPHA2;EPHA4;CD74;P2RY6;INPP5D;CD34;VCAN;TNFRSF10B;TNFRSF4;ADRB2;LDLR;LGALS9;LGALS1;ITGA6;DAG1;ITGA2;CD44;ITGA3;ITGB4;FGF2;TNC;THBS1;LAMA5;ANGPTL4;RAMP3;RAMP2;LMAN1;ACVR1B;HLA-F;HLA-A;HLA-E;HLA-B;LAT;CXCL12;CXCL2;CXCL10;CXCL11;PAM;ADA;LAMC1;LAMB2;LAMA4;LAMB1;LAMA3;DUSP18;VCAM1;VWF;NID1;MFGE8;CX3CL1;CALR;TNFSF10;BMP6;BMP1;BMP4;PIK3R1;LRP5;LRP6;EFNA1;EFNB1;EFNB2;ITGA1;ADM;FAM3C;GDF15;PRND;CCL2;CCL14;CCL23;S100A4;MET;NRP1;CD200;NRP2;CASP8;RIPK1;TNFRSF10D;TNFRSF10A;TNFRSF11B;IL1R1;IL33;PECAM1;SDC3;ACKR1;KDR;SDC2;PLCE1;PRKACA;S1PR1;PTGER4;IL2RG;IL7;F2R;F2RL3;TEK;ARPC5;POMC;ACTR2;VEGFB;HSPA8;PTHLH;PDGFA;PDGFD;COPA;FGFR1;APLP2;NOTCH4;CD81;TNFRSF1B;LTBR;CSF1;SEMA4C;NFE2L2;FLT4;TNFRSF21;CHUK;TBK1;HMOX1;CD40;BAG6;CTSS;CTSL;CTSF;CD63;LGMN;FLT1;PGF;PTPRF;ADAM15;CD55;PLXNA2;SOS1;PRKCE;SCN9A;TUBA4A;ITPR1;RAF1;PIK3CA;LPP;TNK2;CASP3;FLOT1;MAPK14;CDK6;RPS6KA3;PTK2;ABL1;MCAM;THY1;SLK;PTPRE;TUBA1A;CDK2;CSK;PLCG2;PLD2;CD79B;PLCG1;BSG;STK11;PLAUR;CSF2RB;APP;IGF2;PCNA;CD99;RARA;LAMP1;SEMA6B;CCRL2;PILRB;EPHB4;IL15;SEMA3F;TIE1;FGF12;SPARC;FGF18;ANXA2;RXRA;HMGB1;CAPN1;TGS1;MFAP2;YBX1;FURIN;SCARB1;STAB1;HBEGF;NCL;PTPRB;TXLNA;ALCAM;TLN1;VCL;CD320;NPNT;VEGFC;ANGPTL2;JAM3;LUM;SELP;F8;IKBKB;NSMAF;SNCA;LGALS3;LYN;STAT6;INSR;BCR;IFNAR1;CLEC2B;S100A8;MYL9;ROBO3;ACVR2A;BMPR2;ACVR2B;PSMB8;PSMB9;IFNGR1;IFNGR2;IGF2R;NRAS;MPZL1;MAP3K4;KRAS;MAPK3;LY96;TFRC;PIP5K1A;SMAP1;PIP4K2A;PIP5K1C;MAPK1;PTPRK;CD47;MMP2;PLAT;ROBO4;IL13RA1;ICAM2;ICAM3;TNFRSF12A;PLK3;MCL1;RGS16;PTGS2;ODC1;EIF2AK2;CASP10;CASP6;PLK2;NR3C1;CSNK2B;HSPA1B;TAP1;HSP90AB1;ABCB1;HIPK2;NOS3;RRM2B;CTSD;BIRC3;CASP1;ATM;PTEN;SLC38A2;PRKAB1;LATS2;BDKRB2;VRK1;ULK2;PPM1D;BCL2L1;DNMT1;ETV5;AREG;KMT2E;ZFP91;VANGL1;MCFD2;CD46;GPRC5B;PTAFR;PRMT5;PTPN1;PRKD2;KITLG;SOCS2;NPTN;TAP2;PDIA3;SCARF1;MICA;ULBP2;TGFBR3;CXCR4;TGFBR2;FSTL1;CD109;ACVRL1;DCN;PLAU;LSR;GMFB;ACKR3;OSMR;RPSA;MAPK8;S100A9
</td>
<td style="text-align:left;">
TNFRSF14;TNFRSF25;TNFRSF1B;PLA2G2A;ECE1;RPS6KA1;FGR;TXLNA;LCK;YBX1;PLK3;LRP8;JAK1;TGFBR3;VCAM1;S1PR1;CSF1;CD53;NRAS;VANGL1;CD2;NOTCH2;CD160;MCL1;CTSS;PIP5K1A;S100A9;S100A8;S100A4;S100A1;IL6R;ADAM15;EFNA4;COPA;SLAMF1;CD247;MPZL1;SELL;FASLG;TNFSF4;RGS16;ARPC5;PTPRC;CD55;CD46;LAMB3;LGALS8;RRM2;ODC1;SDC1;POMC;EIF2AK2;SOS1;MCFD2;ACTR2;CD8A;CD8B;ZAP70;CXCR4;DPP4;ITGA6;NFE2L2;ITGA4;ITGAV;CASP10;CASP8;BMPR2;CD28;CTLA4;ICOS;WNT10A;TUBA4A;EPHA4;NCL;COL6A3;RAMP1;PDCD1;ITPR1;RAF1;KAT2B;TGFBR2;CX3CR1;RPSA;CCR1;CCR5;DAG1;PRKCD;LRIG1;ALCAM;CD47;CD200R1;RARRES1;TNFSF10;PIK3CA;BCL6;LPP;IL1RAP;APOD;TNK2;SPON2;CD38;TLR1;TXK;CXCL8;CXCL1;CXCL2;CXCL10;CXCL13;SPP1;CASP6;IL15;CASP3;IL7R;PTGER4;ITGA1;ITGA2;GZMA;IL6ST;PLK2;PIK3R1;HSPA4;HBEGF;NR3C1;ADRB2;PDGFRB;CD74;SPARC;HAVCR2;ITK;RIPK1;HLA-A;FLOT1;HLA-C;LTA;TNF;LTB;NCR3;CSNK2B;HSPA1B;HLA-DRA;HLA-DRB5;HLA-DRB1;HLA-DQA1;HLA-DQB1;HLA-DQA2;TAP2;TAP1;HLA-DMB;HLA-DMA;HLA-DPA1;HLA-DPB1;MAPK14;VEGFA;HSP90AB1;SMAP1;CD109;FYN;MAP3K5;IFNGR1;EZR;IGF2R;MAP3K4;CCR6;LFNG;ABCB1;CDK6;AZGP1;PILRB;KMT2E;PIK3CG;FAM3C;PLXNA4;HIPK2;EPHB6;RARRES2;CDK5;CD99;RPS6KA3;EFNB1;IL2RG;CXCR3;CD40LG;PLXNA3;ANGPT2;TNFRSF10B;TNFRSF10A;PTK2B;FGFR1;PLAT;IKBKB;TGS1;NSMAF;LY96;RRM2B;TNFRSF11B;JAK2;TLN1;GNAQ;NTRK2;CTSL;SYK;TGFBR1;ABL1;RXRA;NOTCH1;CTSD;CD81;SPON1;SAA1;CD59;CD44;MDK;PTPRJ;STX3;MS4A1;CD6;CD5;VEGFB;CAPN1;CTSF;BIRC3;BIRC2;CASP1;ATM;IL10RA;JAML;CD3E;CD3D;CD3G;HSPA8;CHEK1;APLP2;IL15RA;IL2RA;PRKCQ;PIP4K2A;ITGB1;MAPK8;CDK1;PLAU;VCL;PTEN;FAS;CHUK;SLK;PTPRE;VWF;TNFRSF1A;LAG3;CD4;PTPN6;KLRB1;CLEC2D;CD69;KLRF1;KLRD1;KLRC2;KLRC1;KRAS;NELL2;SLC38A2;VDR;TUBA1A;ITGB7;ITGA5;CD63;GDF11;CDK2;STAT6;IFNG;MDM2;LUM;DCN;SOCS2;IGF1;PTPN11;PRKAB1;FLT1;SLC7A1;HMGB1;TNFSF13B;LAMP1;PRMT5;RIPK3;ARF6;GMFB;LGALS3;LGMN;VRK1;HSP90AA1;THBS1;PDIA3;ANXA2;CSK;IL16;MFGE8;FURIN;TNFRSF12A;IL4R;LAT;SPN;MAPK3;ITGAL;STX4;ITGAM;ITGAX;CKLF;PLCG2;TNFSF12;TP53;LGALS9;UNC119;CCL5;CCL4;ERBB2;RARA;STAT3;GRN;MAP3K14;PPM1D;CD79B;PECAM1;PRKCA;BIRC5;LGALS3BP;LMAN1;BCL2;CD226;PCNA;JAG1;BCL2L1;PLCG1;PI3;MMP9;PTPN1;AURKA;BSG;STK11;CD70;TNFSF14;DNMT1;ICAM1;TYK2;LDLR;EPOR;CALR;PRKACA;ADGRE5;JAK3;LSR;HCST;TGFB1;CD79A;PLAUR;APOE;PRKD2;DBP;CD37;LILRB1;KIR3DL1;KIR3DL2;MAPK1;MIF;LIF;OSM;DUSP18;IL2RB;LGALS1;EP300;APP;IFNAR2;IFNAR1;ITGB2;DIP2A
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level4">
<h4 id="intercellular-cross-talk-visualization">Intercellular cross-talk visualization<a class="anchor" aria-label="anchor" href="#intercellular-cross-talk-visualization"></a>
</h4>
<p>Intercellular cross-talk results can be filtered to maintain only the
significant ones. The results can be visualized by using Ulisse package
functions. The code below can be used to plot the results as a network.
<code>filtering</code> argument is used to control which intercellular
cross-talks have to be visualized in the network. If set to
<code>TRUE</code>, then <code>p_val</code>, <code>FDR</code> and
<code>ct_val</code> are used to identify the significant communications
and plot these with a solid line, whereas all the others will be plotted
with a dashed line. Here, we enabled this visualization, by providing
p-value and FDR filtering. <code>ct_val = NULL</code> allows to consider
all intercellular cross-talk without filtering for this value.
<code>community</code> can be logical (if the community should be
calculated by using <code><a href="https://rdrr.io/pkg/igraph/man/cluster_fast_greedy.html" class="external-link">igraph::fastgreedy.community()</a></code>), or a
community object as calculated with igraph package. <code>vertex</code>
is used to color the vertices, and can be set to <code>"name"</code> to
color them according to the names of the cell-types.
<code>edge_col_by</code> and <code>edge_width</code> parameters control
the value used to color the edges and if their width should be
proportional to that value. <code>edge_adj_col</code> is used to control
the transparency of the edges: as in pathway cross-talk there can be
lots of links it can e useful to control the transparency.
<code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code> function returns the igraph cross-talk
network with the communities under “comm_id” vertex attribute (if
calculated). <code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code> function uses
<code>ggraph</code> package functions, which are ggplot2-based. If
<code>file_out</code> is set to <code>NULL</code> the function returns
also the ggplot2 network object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">expr_ccc_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_network_CT.html">plot_network_CT</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">expr_ccc</span><span class="op">$</span><span class="va">ct_res</span>, </span>
<span>                                filtering <span class="op">=</span> <span class="cn">T</span>, p_val <span class="op">=</span> <span class="fl">0.05</span>, FDR <span class="op">=</span> <span class="fl">0.2</span>, ct_val <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                                community <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                                vertex <span class="op">=</span> <span class="st">"name"</span>, vertex_size <span class="op">=</span> <span class="fl">5</span>, vertex_label <span class="op">=</span> <span class="cn">TRUE</span>, vertex_pal <span class="op">=</span> <span class="va">pal_ct</span>,</span>
<span>                                edge_col_by <span class="op">=</span> <span class="st">"ct_score"</span>, edge_pal<span class="op">=</span><span class="cn">NULL</span>, </span>
<span>                                edge_width <span class="op">=</span> <span class="cn">T</span>, edge_adj_col <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> </span>
<span><span class="va">expr_ccc_net</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH be33f36 UN-- 9 36 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), label (v/c), ct_score (e/n), ngenes_gs1 (e/n),</span></span>
<span><span class="co">#&gt; | ngenes_gs2 (e/n), nlink (e/n), p_value_link (e/n), FDR_link (e/n),</span></span>
<span><span class="co">#&gt; | p_adj_BH (e/n), weight_gs1 (e/n), weight_gs2 (e/n), genes_gs1 (e/c),</span></span>
<span><span class="co">#&gt; | genes_gs2 (e/c), filt (e/c)</span></span>
<span><span class="co">#&gt; + edges from be33f36 (vertex names):</span></span>
<span><span class="co">#&gt;  [1] Endothelial--CAFs              Endothelial--PVL              </span></span>
<span><span class="co">#&gt;  [3] Endothelial--B-cells           Endothelial--Plasmablasts     </span></span>
<span><span class="co">#&gt;  [5] Endothelial--T-cells           Endothelial--Myeloid          </span></span>
<span><span class="co">#&gt;  [7] Endothelial--Cancer Epithelial Endothelial--Normal Epithelial</span></span>
<span><span class="co">#&gt;  [9] CAFs       --PVL               CAFs       --B-cells          </span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code></pre>
<p><img src="sc_data/pct_net.jpeg" width="600"></p>
<p><code><a href="../reference/ct_heatmap.html">ct_heatmap()</a></code> function can be used to plot intercellular
cross-talk as an heatmap by selecting the variable that have to be used
to color the heatmap (<code>color_by</code>, here we used the cross-talk
score), color scale (<code>color = c("lightyellow", "red3")</code>), and
the color of cells in the heatmap of the cell types pairs without
communications (<code>no_ct_color</code>).</p>
<p>Similarly to <code><a href="../reference/plot_network_CT.html">plot_network_CT()</a></code>,
<code><a href="../reference/ct_heatmap.html">ct_heatmap()</a></code> function can select significant interactions,
which are represented with a star. This is achieved by enabling
<code>filtering = TRUE</code>, that will further require p-value, FDR
and/or cross-talk score thresholds.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ct_heatmap.html">ct_heatmap</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">clCC</span><span class="op">$</span><span class="va">ct_res</span>, </span>
<span>           color_by <span class="op">=</span> <span class="st">"ct_score"</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightyellow"</span>, <span class="st">"red3"</span><span class="op">)</span>, no_ct_color <span class="op">=</span> <span class="st">"whitesmoke"</span>, </span>
<span>           filtering <span class="op">=</span> <span class="cn">TRUE</span>, p_val <span class="op">=</span> <span class="fl">0.05</span>, FDR <span class="op">=</span> <span class="fl">0.2</span>, ct_val <span class="op">=</span> <span class="cn">NULL</span>,label_size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="sc_data/cc_ct_f_heatmap.jpeg" width="600"></p>
</div>
</div>
<div class="section level3">
<h3 id="deg-data">DEG data<a class="anchor" aria-label="anchor" href="#deg-data"></a>
</h3>
<p>We will now analyse intercellular cross-talk by using DE results.</p>
<p>Here we will use Seurat function for Differential Expression Analysis
with default parameters to calculate DEG for each cell type against the
others. Then, we will use Benjamini-Hochberg method to obtain adjusted
p-value to be used to filter DEGs to maintain only the significant ones.
We will select only the up-regulated genes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">celltype_major</span></span>
<span></span>
<span><span class="va">DEGs_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">data</span>, assay <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span> <span class="co">#this may take a while</span></span>
<span><span class="va">DEGs_data</span><span class="op">$</span><span class="va">p_val_BH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">DEGs_data</span><span class="op">$</span><span class="va">p_val</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">DEGs_data</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
p_val
</th>
<th style="text-align:right;">
avg_log2FC
</th>
<th style="text-align:right;">
pct.1
</th>
<th style="text-align:right;">
pct.2
</th>
<th style="text-align:right;">
p_val_adj
</th>
<th style="text-align:left;">
cluster
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
p_val_BH
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ACKR1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4.585262
</td>
<td style="text-align:right;">
0.615
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
ACKR1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
VWF
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.505398
</td>
<td style="text-align:right;">
0.861
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
VWF
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
PLVAP
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.213029
</td>
<td style="text-align:right;">
0.820
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
PLVAP
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
RAMP2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.016198
</td>
<td style="text-align:right;">
0.877
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
RAMP2
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
ECSCR.1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.816407
</td>
<td style="text-align:right;">
0.861
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
ECSCR.1
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<p>It is not important which package and algorithm is used for DEG
calculation, as Ulisse function is structured to handle any input
type.</p>
<p>DEGs data can now be used to build the cell-type lists by using
<code><a href="../reference/preparing_DEG_list.html">preparing_DEG_list()</a></code> function. This function does not take
as an input a table with the results of DEG calculation, but the
specific columns (genes, their log2 Fold Change, p-value and the
cluster), so that it is not important which package and algorithm is
used for the differential analysis. <code><a href="../reference/preparing_DEG_list.html">preparing_DEG_list()</a></code>
uses these inputs to calculate for each <code>gene</code> in
<code>clusters</code> the associated weights, that are the product
between <code>log2FC</code> and the -log10 of <code>p_val</code>. Then,
the function builds a vector for each <code>cluster</code>, each one
composed by the weights, named after the respective genes. Then, these
vectors are filtered to maintain only the gene that are present in
<code>universe</code> and returned as a list</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEGs_data</span> <span class="op">&lt;-</span> <span class="va">DEGs_data</span><span class="op">[</span><span class="va">DEGs_data</span><span class="op">$</span><span class="va">p_val_BH</span> <span class="op">&lt;=</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">DEGs_data</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&gt;=</span> <span class="fl">0.5</span>,<span class="op">]</span></span>
<span></span>
<span></span>
<span><span class="va">DEG_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preparing_DEG_list.html">preparing_DEG_list</a></span><span class="op">(</span>cluster <span class="op">=</span> <span class="va">DEGs_data</span><span class="op">$</span><span class="va">cluster</span>, </span>
<span>                               p_val <span class="op">=</span> <span class="va">DEGs_data</span><span class="op">$</span><span class="va">p_val_BH</span>, </span>
<span>                               log2FC <span class="op">=</span> <span class="va">DEGs_data</span><span class="op">$</span><span class="va">avg_log2FC</span>, </span>
<span>                               gene <span class="op">=</span> <span class="va">DEGs_data</span><span class="op">$</span><span class="va">gene</span>, </span>
<span>                               universe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">adj.m</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DEG_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;     ACKR1       VWF     RAMP2     RAMP3   CLEC14A    ADGRL4 </span></span>
<span><span class="co">#&gt; 1397.4371 1068.3300  919.2382  839.4853  801.0583  800.9338</span></span></code></pre>
<div class="section level4">
<h4 id="communication-analysis-1">Communication Analysis<a class="anchor" aria-label="anchor" href="#communication-analysis-1"></a>
</h4>
<p>The subsequent part of the analysis is identical to the one used for
expression data: <code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code> is used to analyse
intercellular cross-talk with the same inputs and parameters. The
calculation will results again in a two-object list with the detail of
the gene-gene interaction and the cross talk results.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">DEG_ccc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gs_cross_talk.html">gs_cross_talk</a></span><span class="op">(</span>gs_list <span class="op">=</span> <span class="va">DEG_list</span>,</span>
<span>                         gene_network_adj <span class="op">=</span> <span class="va">adj.m</span>,</span>
<span>                         k <span class="op">=</span> <span class="fl">49</span>, shared <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                         ct_info <span class="op">=</span> <span class="cn">T</span>, hash <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                         mc_cores_perm <span class="op">=</span> <span class="fl">1</span>, mc_cores_ct <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">DEG_ccc</span><span class="op">$</span><span class="va">ct_info</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
gs1_gene
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:left;">
gs2_gene
</th>
<th style="text-align:right;">
score
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
KDR
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
DCN
</td>
<td style="text-align:right;">
369360.44
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
ITGB1
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
LUM
</td>
<td style="text-align:right;">
62407.07
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
ITGB1
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
COL6A3
</td>
<td style="text-align:right;">
44302.73
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
NOTCH4
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
THBS2
</td>
<td style="text-align:right;">
127874.51
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
ITGB1
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:left;">
THBS2
</td>
<td style="text-align:right;">
30321.90
</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG_ccc</span><span class="op">$</span><span class="va">ct_res</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class="table  lightable-minimal table table-striped table-condensed" style='font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
gs1
</th>
<th style="text-align:left;">
gs2
</th>
<th style="text-align:right;">
ct_score
</th>
<th style="text-align:right;">
ngenes_gs1
</th>
<th style="text-align:right;">
ngenes_gs2
</th>
<th style="text-align:right;">
nlink
</th>
<th style="text-align:right;">
p_value_link
</th>
<th style="text-align:right;">
FDR_link
</th>
<th style="text-align:right;">
p_adj_BH
</th>
<th style="text-align:right;">
weight_gs1
</th>
<th style="text-align:right;">
weight_gs2
</th>
<th style="text-align:left;">
genes_gs1
</th>
<th style="text-align:left;">
genes_gs2
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
CAFs
</td>
<td style="text-align:right;">
3706408.2
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
75
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0200000
</td>
<td style="text-align:right;">
0.0566667
</td>
<td style="text-align:right;">
10179.2059
</td>
<td style="text-align:right;">
18710.881
</td>
<td style="text-align:left;">
KDR;ITGB1;NOTCH4;ITGA6;CD36;INSR;ITGA5;ITGAV;CD200;PTPRB;NRP1;PDGFB;TIE1;BMPR2;FZD4;SPARC;MMRN2;FGFR1;VWF;ROBO4;ROBO1;THBS1;CD81;TGFBR3;CDH5;IL6;SELP;ACVRL1;TGFBR2;ACKR1;TEK;ANGPTL2;CXCL12;ANGPTL4;PGF;CD40;VEGFC;SEMA3G;NID1;LAMB1;LAMA4;VCAM1;LAMC1;LAMA5;IL6ST;LIFR;STAT3;RAMP2;RAMP3;CALCRL;CD99;ADM;TNFRSF10B;GRN;PLXNA2;PLXND1;NRP2;APP;FLT1
</td>
<td style="text-align:left;">
DCN;LUM;COL6A3;THBS2;IGF1;CXCL12;MXRA5;PTN;PDGFRA;ANGPTL2;FST;WNT2;PDGFRB;FGFR1;CD248;FBLN1;FGF7;SPARC;GREM1;SCARA5;NID1;VCAM1;SLIT3;MMP2;CD99;LRP1;INHBA;ANGPTL4;COMP;F3;PLA2G2A;BMP1;VCAN;LGALS3BP;TGFB3;LAMB1;CCL2;LGALS1;CLEC11A;SPON2;ITGA5;CDH11;IGF2;NRP1;TNFSF13B;MDK;NRP2;ITGB1;ITGA1;IL6;IL6ST;ADM;ITGAV;THBS1;PTHLH;LAMC1;LAMA4;TNC;CD81;CXCL11;RAMP1;TNFSF10;TNFRSF1A;SDC2;CXCL2;STAT3;CXCL9;CLCF1;CXCL10;SEMA3C;CXCL6;SPON1;CXCL1;CXCL3;VEGFA
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
PVL
</td>
<td style="text-align:right;">
1055689.1
</td>
<td style="text-align:right;">
51
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.0200000
</td>
<td style="text-align:right;">
0.0566667
</td>
<td style="text-align:right;">
6838.0833
</td>
<td style="text-align:right;">
3553.581
</td>
<td style="text-align:left;">
NRP1;PDGFB;CXCL12;LAMB1;LAMC1;TEK;TIE1;ROBO4;ROBO1;ACKR1;FLT1;NRP2;FGFR1;ITGAV;PAM;EDN1;ITGB1;DLL4;DLL1;JAG1;JAG2;CCL23;ITGA5;ACVRL1;TGFBR2;TGFBR3;CDH5;ITGA6;MMRN2;IL6ST;NOTCH4;VWF;VEGFC;NID1;LAMA4;VCAM1;ANGPTL2;LAMA5;THBS1;CD81;KDR;ANGPTL4;APP;PTK2;STAB1;CD36;LIFR;IL6;STAT3;PTPRB;GRN
</td>
<td style="text-align:left;">
PDGFRB;AVPR1A;ITGA1;ANGPT2;SLIT3;CCL8;PGF;SPARC;EDIL3;FXYD1;EDNRA;NID1;NOTCH3;ADRA2A;ITGA7;EDNRB;ANGPT1;VCAM1;CCL2;TGFB3;LGALS1;ANGPTL4;LAMA4;CD248;ANGPTL2;IL6;JAG1;COL6A3;ITGB1;ANGPTL1;CD99;LAMC1;CD36;PTK2;SDC2;LGALS3BP;LAMB2;APOE;FGF7;TLN1;CD59;LAMB1;THBS1;IL6ST;STAT3;ITGAV;CXCL12;CLEC11A;TNC;ECE1;TNFRSF1A;BCAM
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
B-cells
</td>
<td style="text-align:right;">
134594.1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.94
</td>
<td style="text-align:right;">
0.8782609
</td>
<td style="text-align:right;">
0.9987500
</td>
<td style="text-align:right;">
1791.6487
</td>
<td style="text-align:right;">
1769.084
</td>
<td style="text-align:left;">
TNFRSF1A;CD40;CXCL12;CD81;CD63;CD34;EFNB2;TNFSF10;VCAM1;VWF;THBS1
</td>
<td style="text-align:left;">
LTB;CXCR4;HLA-DRA;SELL;FCRL1;CCR6;EZR;HLA-DRB1;ITGA4
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
Plasmablasts
</td>
<td style="text-align:right;">
112975.8
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.92
</td>
<td style="text-align:right;">
0.9817647
</td>
<td style="text-align:right;">
0.9987500
</td>
<td style="text-align:right;">
881.9485
</td>
<td style="text-align:right;">
128.098
</td>
<td style="text-align:left;">
PECAM1
</td>
<td style="text-align:left;">
CD38
</td>
</tr>
<tr>
<td style="text-align:left;">
Endothelial
</td>
<td style="text-align:left;">
T-cells
</td>
<td style="text-align:right;">
2248603.9
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.1311111
</td>
<td style="text-align:right;">
0.0566667
</td>
<td style="text-align:right;">
5940.8172
</td>
<td style="text-align:right;">
4960.810
</td>
<td style="text-align:left;">
F2R;CLEC2B;CD59;ACKR1;PECAM1;KDR;CSF2RB;EPHA4;MCAM;PLXNA2;APP;ITGB1;ADAM15;SLK;TUBA1A;STAT3;FLOT1;PTK2;PTPRE;CD36;CD55;CXCL12;VWF;VCAM1;THBS1;TNFRSF1A;CD40;GRN;CD34;CD81;LIFR;IL6ST;ACVRL1;TGFBR2;FSTL1;TGFBR3;ITGAV;ICAM2;ITGA5;TEK
</td>
<td style="text-align:left;">
GZMA;KLRB1;CD2;CCL5;FYN;KLRF1;CXCL13;CXCR4;ITGA4;LTB;TNFRSF1B;ADGRE5;ITGB7;CD38;SELL;CD99;JAK1;TGFB1;ITGB2;TLN1;STAT3;DIP2A;PIK3R1
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level4">
<h4 id="intercellular-cross-talk-visualization-1">Intercellular cross-talk visualization<a class="anchor" aria-label="anchor" href="#intercellular-cross-talk-visualization-1"></a>
</h4>
<p>As done before, we can build intercellular cross-talk network by
filtering the results and using Ulisse function and plot it as a
network</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DEG_CCC_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_network_CT.html">plot_network_CT</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">DEGcl_ct</span><span class="op">$</span><span class="va">ct_res</span>, </span>
<span>                               filtering <span class="op">=</span> <span class="cn">T</span>, p_val <span class="op">=</span> <span class="fl">0.05</span>, FDR <span class="op">=</span> <span class="fl">0.2</span>, ct_val <span class="op">=</span> <span class="cn">NULL</span>,  </span>
<span>                               community <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                               vertex <span class="op">=</span> <span class="st">"name"</span>, vertex_size <span class="op">=</span> <span class="fl">5</span>, vertex_label <span class="op">=</span> <span class="cn">TRUE</span>, vertex_pal <span class="op">=</span> <span class="va">pal_ct</span>,</span>
<span>                               edge_col_by <span class="op">=</span> <span class="st">"ct_score"</span>, edge_pal<span class="op">=</span><span class="cn">NULL</span>, </span>
<span>                               edge_width <span class="op">=</span> <span class="cn">T</span>, edge_adj_col <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> </span>
<span><span class="va">DEG_CCC_net</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; IGRAPH 22bb420 UN-- 9 34 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), label (v/c), ct_score (e/n), ngenes_gs1 (e/n),</span></span>
<span><span class="co">#&gt; | ngenes_gs2 (e/n), nlink (e/n), p_value_link (e/n), FDR_link (e/n),</span></span>
<span><span class="co">#&gt; | p_adj_BH (e/n), weight_gs1 (e/n), weight_gs2 (e/n), genes_gs1 (e/c),</span></span>
<span><span class="co">#&gt; | genes_gs2 (e/c), filt (e/c)</span></span>
<span><span class="co">#&gt; + edges from 22bb420 (vertex names):</span></span>
<span><span class="co">#&gt;  [1] Endothelial--CAFs              Endothelial--PVL              </span></span>
<span><span class="co">#&gt;  [3] Endothelial--B-cells           Endothelial--Plasmablasts     </span></span>
<span><span class="co">#&gt;  [5] Endothelial--T-cells           Endothelial--Myeloid          </span></span>
<span><span class="co">#&gt;  [7] Endothelial--Cancer Epithelial Endothelial--Normal Epithelial</span></span>
<span><span class="co">#&gt;  [9] CAFs       --PVL               CAFs       --B-cells          </span></span>
<span><span class="co">#&gt; + ... omitted several edges</span></span></code></pre>
<p><img src="sc_data/DEG_net.jpeg" width="600"></p>
<p>or as an heatmap</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/ct_heatmap.html">ct_heatmap</a></span><span class="op">(</span>ct <span class="op">=</span> <span class="va">DEGcl_ct</span><span class="op">$</span><span class="va">ct_res</span>, </span>
<span>           color_by <span class="op">=</span> <span class="st">"ct_score"</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lightyellow"</span>, <span class="st">"red3"</span><span class="op">)</span>, no_ct_color <span class="op">=</span> <span class="st">"whitesmoke"</span>, </span>
<span>           filtering <span class="op">=</span> <span class="cn">T</span>, p_val <span class="op">=</span> <span class="fl">0.05</span>, FDR <span class="op">=</span> <span class="fl">0.2</span>, ct_val <span class="op">=</span> <span class="cn">NULL</span>,  label_size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p><img src="sc_data/DEGcc_ct_f_heatmap.jpeg" width="600"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="comparison-between-expression-and-deg-intercellular-cross-talk-results">Comparison between expression and DEG intercellular cross-talk
results<a class="anchor" aria-label="anchor" href="#comparison-between-expression-and-deg-intercellular-cross-talk-results"></a>
</h2>
<p>Ulisse provides a function to build and visualize a union network out
of multiple analyses or samples.
<code><a href="../reference/comparing_results_network.html">comparing_results_network()</a></code> function takes as an input a
list of results obtained with <code><a href="../reference/gs_cross_talk.html">gs_cross_talk()</a></code>. The function
does not align the results, so there should be correspondence among the
gene sets names of the different samples, especially when using
clusters. Here we will compare the significant results obtained with
expression and DEG intercellular cross-talk analyses on the same sample,
thus the cell-types corresponds.
<code><a href="../reference/comparing_results_network.html">comparing_results_network()</a></code> needs as an input a named list
with the cross-talk results that should be already filtered to maintain
only the significant intercellular cross-talk, if needed. This function
will create a union network where the vertices are all the cell types
present in the results in <code>res_list</code>, the links are the all
the communication present between them in the results. The vertices can
be colored by considering the number of results in which are present
(<code>vertex_number = TRUE</code>) and/or by a discrete variable (in
this case <code>vertex = "name"</code>, otherwise a two list, where the
first element is the name of the variable, the second the named vector
with, see function help for detail). If both parameters are enabled, as
in this case, the <code>vertex</code> one is used to color voronoi
tesselation (see ggrpah package for further detail). When
<code>vertex_number = TRUE</code>, a color palette can be provided,
composed by minimum and maximum value colors, or the function will use
as default “blue” to “red”. Edges can be colored by the number of
results that share them (<code>edge_color_by = "number"</code>) or, as
in this case, by which results contain them
(<code>edge_color_by = "which"</code>). Edges width can be proportional
to the number of results in which they are present
(<code>edge_width = TRUE</code>).
<code><a href="../reference/comparing_results_network.html">comparing_results_network()</a></code> will return the union network
and the plot id <code>file_out</code> argument is not provided.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expr <span class="op">=</span> <span class="va">expr_ccc</span><span class="op">$</span><span class="va">cc_communications</span>,</span>
<span>                 DEG <span class="op">=</span> <span class="va">DEG_ccc</span><span class="op">$</span><span class="va">cc_communications</span><span class="op">)</span></span>
<span><span class="va">res_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">res_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">p_value_link</span> <span class="op">&lt;=</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">x</span><span class="op">$</span><span class="va">FDR_link</span> <span class="op">&lt;=</span> <span class="fl">0.2</span>,<span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comp.out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/comparing_results_network.html">comparing_results_network</a></span><span class="op">(</span>res_list <span class="op">=</span> <span class="va">res_list</span>, </span>
<span>                                      vertex_number <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                      vertex <span class="op">=</span> <span class="st">"name"</span>, vertex_pal <span class="op">=</span> <span class="va">pal_ct</span>,</span>
<span>                                      voronoi_radius <span class="op">=</span> <span class="fl">0.3</span>, voronoi_alpha <span class="op">=</span> <span class="fl">0.3</span>, </span>
<span>                                      edge_width <span class="op">=</span> <span class="cn">TRUE</span>, edge_color_by <span class="op">=</span> <span class="st">"which"</span>,</span>
<span>                                      edge_adj_col <span class="op">=</span> <span class="fl">0.5</span>, vertex_number_adj <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comp.out</span></span></code></pre></div>
<p><img src="sc_data/comp_res_net.jpeg" width="600"></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Hao2021" class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erica Andersen-Nissen, William M. Mauck,
Shiwei Zheng, Andrew Butler, Maddie J. Lee, et al. 2021. <span>“<span class="nocase">Integrated analysis of multimodal single-cell
data</span>.”</span> <em>Cell</em> 184 (13): 3573–3587.e29. <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">https://doi.org/10.1016/j.cell.2021.04.048</a>.
</div>
<div id="ref-turei2021" class="csl-entry">
Türei, Dénes, Alberto Valdeolivas, Lejla Gul, Nicolàs Palacio-Escat,
Michal Klein, Olga Ivanova, Márton Ölbei, et al. 2021. <span>“Integrated
Intra-and Intercellular Signaling Knowledge for Multicellular Omics
Analysis.”</span> <em>Molecular Systems Biology</em> 17 (3): e9923.
</div>
<div id="ref-wu2021" class="csl-entry">
Wu, Sunny Z, Ghamdan Al-Eryani, Daniel Lee Roden, Simon Junankar, Kate
Harvey, Alma Andersson, Aatish Thennavan, et al. 2021. <span>“A
Single-Cell and Spatially Resolved Atlas of Human Breast
Cancers.”</span> <em>Nature Genetics</em> 53 (9): 1334–47.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Alice Chiodi, Ettore Mosca.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
